%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%% OXFORD THESIS TEMPLATE

% Originally by Keith A. Gillow (gillow@maths.ox.ac.uk), 1997
% Modified by Sam Evans (sam@samuelevansresearch.org), 2007
% Modified by John McManigle (john@oxfordechoes.com), 2015
% Modified by Maud Gautier (https://github.com/MaudGautier), 2019
% Modified by Thibault Latrille (https://github.com/ThibaultLatrille), 2020
%
% This version Copyright (c) 2020 Thibault Latrille
%
% Broad permissions are granted to use, modify, and distribute this software
% as specified in the MIT License included in this distribution's LICENSE file.

%%%%% CHOOSE PAGE LAYOUT
%\documentclass[a4paper,openright,twoside]{thesis}
\documentclass[a4paper,oneside,nobind]{thesis}
\usepackage{hyperref}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{bm}

\definecolor{Nonpolar}{HTML}{fef1d2}
\definecolor{Polar}{HTML}{a9fdd8}
\definecolor{Basic}{HTML}{d7f8ff}
\definecolor{Acidic}{HTML}{feceea}
\definecolor{Stop}{HTML}{D0D0D0}

\graphicspath{{figures/}{MutationSelectionDrift/artworks/}{GenotypePhenotypeFitness/artworks/}{GenotypePhenotypeFitness/}{NucleotideBias/artworks/}}
\makeatletter
\def\input@path{{MutationSelectionDrift/artworks/}{GenotypePhenotypeFitness/artworks/}{NucleotideBias/artworks/}}
\makeatother

\input{notations}
% make part like a chapter

%%%%% THE ACTUAL DOCUMENT STARTS HERE
\begin{document}
\pagenumbering{gobble}
\hypersetup{pageanchor=false}
\input{sections/frontpage}
%%%%% CHOOSE YOUR LINE SPACING HERE
\setlength{\textbaselineskip}{1.2em}
% You can set the spacing here for the roman-numbered pages (acknowledgements, table of contents, etc.)
\setlength{\frontmatterbaselineskip}{1.2em}
% Leave this line alone; it gets things started for the real document.
\setlength{\baselineskip}{\textbaselineskip}

%%%%% CHOOSE YOUR SECTION NUMBERING DEPTH HERE
% The level that gets a number:
\setcounter{secnumdepth}{2}
% The level that shows up in the ToC:
\setcounter{tocdepth}{2}

%%%% EPIGRAPH
\begin{alwayssingle}
	\thispagestyle{empty}
	\vspace*{\fill}
	\makebox[\textwidth][r] {
		\begin{minipage}[t]{8cm}
			\emph{}\\
			\hfill \slshape{L'humanité est constamment aux prises avec deux processus contradictoires dont l'un tend à instaurer l'unification, tandis que l'autre vise à maintenir ou à rétablir la diversification.}\\
			\hfill \qauthor{Claude Lévi-Strauss.}
		\end{minipage}
	}
	\vspace*{\fill}
	\vspace*{\fill}
\end{alwayssingle}


%%%%% DEDICATION
%\begin{alwayssingle}
%	\thispagestyle{empty}
%	\newgeometry{hmargin={1.6cm, 1.6cm}, vmargin={1.5cm,1.5cm}, headsep=.5cm,footskip=0.5cm}
%	\begin{center}
%		\includegraphics[valign=t,width=1\textwidth, page=1] {figures/preface}
%	\end{center}
%	\restoregeometry
%\end{alwayssingle}

% %%%%% ABSTRACT SEPARATE
% % This is used to create the separate, one-page abstract that you are required to hand into the Exam
% % Schools.  You can comment it out to generate a PDF for printing or whatnot.
% \begin{abstractseparate}
%     \input{text/abstract} % Create an abstract.tex file in the 'text' folder for your abstract.
% \end{abstractseparate}
%
% \newgeometry{inner=2cm, outer=2cm, top=2cm, bottom=2cm}
\setlength{\baselineskip}{1.4\frontmatterbaselineskip}
\begin{alwayssingle}
	\input{sections/abstract}
\end{alwayssingle}

\begin{alwayssingle}
	\input{sections/abstract_fr}
\end{alwayssingle}

%%%%% ACKNOWLEDGEMENTS -- Nothing to do here except comment out if you don't want it.
%\input{sections/acknowledgement}

\setlength{\baselineskip}{\frontmatterbaselineskip}
\hypersetup{pageanchor=true}
\begin{romanpages}
%%%%% MINI TABLES
\dominitoc % include a mini table of contents
%\dominilof  % include a mini list of figures
%\dominilot  % include a mini list of tables
% This aligns the bottom of the text of each page.  It generally makes things look better.
\flushbottom
% This is where the whole-document ToC appears:
{
	\setlength{\baselineskip}{\frontmatterbaselineskip}
	\hypersetup{linkcolor=GREYDARK}
	\tableofcontents
	\listoffigures
	\mtcaddchapter
	\listoftables
	\mtcaddchapter
}
{
	\setlength{\baselineskip}{\frontmatterbaselineskip}
	\input{sections/glossary}
}

\end{romanpages}

%%%% CHAPTERS
\setlength{\baselineskip}{1.4\frontmatterbaselineskip}
\pagenumbering{arabic}
\flushbottom
\input{sections/preamble}

\pagestyle{fancybook}

\part{Introduction}

\label{part:intro}
\input{sections/intro_historical}
\input{sections/intro_formalism}
\input{sections/intro_codon_models}
\input{sections/intro_inference}
\input{sections/intro_thermodynamic}
\input{sections/intro_goal}

\part{Studies}
\label{part:studies}

\thispagestyle{empty}
\chapter[Robustness of codon models to mutational bias]{Robustness of codon models to mutational bias}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:NucleotideBias}
\input{NucleotideBias/Nucleotide-Bias-main}

\thispagestyle{empty}
\chapter[Inferring long-term population size] {Inferring long-term demographic history with Mutation-Selection models}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:MutSelDrift}
\input{MutationSelectionDrift/MutSelDrift-main}

\section{Supplementary Materials}
Supplementary materials and figures are available in appendix, chapter~\ref{chap:MutSelDrift-SuppMat}.
The scripts and instructions necessary to reproduce the experiments are available at \url{https://github.com/ThibaultLatrille/MutationSelectionDrift}.

\section{Acknowledgments}
The authors gratefully acknowledge the help of Nicolas Rodrigue, Laurent Gueguen, Philippe Veber for their advice and review concerning this manuscript.
This work was performed using the computing facilities of the CC LBBE/PRABI.

\thispagestyle{empty}
\chapter[Substitution rate susceptibility]{Substitution rate response to changes in effective population size and protein expression level}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:GenoPhenoFit}
\input{GenotypePhenotypeFitness/GenoPhenoFit-main}

\section{Supplementary Materials}
The mathematical development of the general case and supplementary figures are available in appendix, chapter~\ref{chap:GenoPhenoFit-SuppMat}.
The scripts and instructions necessary to reproduce this study are available at \url{https://github.com/ThibaultLatrille/GenotypePhenotypeFitness}.

\section{Acknowledgments}
The authors gratefully acknowledge the help of Nicolas Rodrigue and Laurent Duret for their advice and review concerning this manuscript.
This work was performed using the computing facilities of the CC LBBE/PRABI.

\part{Conclusion}
\label{part:conclusion}
\input{sections/conclusion}

%% APPENDICES %% 
\part{Appendices}

\thispagestyle{empty}
\chapter{Inferring long-term population size - Supplementary Materials}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:MutSelDrift-SuppMat}
\input{MutationSelectionDrift/MutSelDrift-supp-mat}

\thispagestyle{empty}
\chapter{Substitution rate susceptibility - Supplementary Materials}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:GenoPhenoFit-SuppMat}
\input{GenotypePhenotypeFitness/GenoPhenoFit-supp-mat}

\chapter*{Mutation-selection-drift as bridge between phylogeny and population-genetics}
\addcontentsline{toc}{part}{Poisson random fields}
\phantomsection
\label{sec-appendix:PRF}
\input{sections/appendix_PRF}

\thispagestyle{empty}
%\chapter*{Detecting site-specific adaptation with mutation-selection models}
\addcontentsline{toc}{part}{Detecting site-specific adaptation with mutation-selection models}
\phantomsection
\label{sec-appendix:MutSelM3starMBE}
\includepdf[pages=-]{sections/MutSelM3starMBE.pdf}

%%%%% REFERENCES
\thispagestyle{empty}
\bibliographystyle{natbib}
\bibliography{references/references, references/introduction}
\end{document}

