\documentclass{article}
\usepackage{xcolor}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\definecolor{LIGHTBLUELINK}{HTML}{3366BB}
\definecolor{PURPLELINK}{HTML}{663366}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false ]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
linkcolor=DARKBLUELINK,
anchorcolor=DARKBLUELINK,
citecolor=DARKBLUELINK,
filecolor=DARKBLUELINK,
menucolor=DARKBLUELINK,
urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage[margin=60pt]{geometry}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{bm,bbold}
\usepackage{verbatim}
\usepackage{float}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{tabu}
\tabulinesep=0.6mm
\newcommand\cellwidth{\TX@col@width}
\usepackage{hhline}
\setlength{\arrayrulewidth}{1.2pt}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\AtBeginEnvironment{tabu}{\footnotesize}
\usepackage{booktabs}

\usepackage{graphicx}
\graphicspath{{artworks/}}
\pdfstringdefDisableCommands{%
\renewcommand*{\bm}[1]{#1}%
% any other necessary redefinitions
}
\usepackage{xfrac, nicefrac}
\usepackage{natbib}
\pdfinclusioncopyfonts=1
\input{notations}

\begin{document}
\part*{Supplementary materials}
\tableofcontents

\section{Simulated alignments}
\label{sec:simulated-alignments}

\subsection{Primate phylogeny - 4980 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_1_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_2_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_3_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_1_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize D\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_2_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/obs_atgc_3_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize G\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_simulations/omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize H\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa and 9960 codon sites, for 32 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value (simulations shown in the main manuscript).
Observed $\atgc$ from the alignment and predicted $\atgc$ by the models at the first (panels A \& D), second (panels B \& E) and third (panels C \& F) codon positions as a function of mutational bias for Muse \& Gaut (panels A, B \& C) and our tensor model (panel C, D \& E).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel G) and our tensor model (panel H).


\subsection{Primate phylogeny - 498 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa and 498 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons1Mu1.0_omega_pair_MF.pdf}
True versus estimated values of $\omega$ between pairs of amino-acids.
Vertical bars are the 95\% confidence intervals for the mean value.

\subsection{Primate phylogeny - 996 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa and 996 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons2Mu1.0_omega_pair_MF.pdf}
True versus estimated values of $\omega$ between pairs of amino-acids.
Vertical bars are the 95\% confidence intervals for the mean value.

\subsection{Primate phylogeny - 2490 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa and 2490 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons5Mu1.0_omega_pair_MF.pdf}
True versus estimated values of $\omega$ between pairs of amino-acids.
Vertical bars are the 95\% confidence intervals for the mean value.

\subsection{Primate phylogeny - 9960 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa and 9960 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons20Mu1.0_omega_pair_MF.pdf}
True versus estimated values of $\omega$ between pairs of amino-acids.
Vertical bars are the 95\% confidence intervals for the mean value.

\subsection{Primate phylogeny - 4980 codons - branch length \%2}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu0.5_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu0.5_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu0.5_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu0.5_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu0.5_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa with decreased branch lenght by a factor 2 and 4980 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\subsection{Primate phylogeny - 4980 codons - branch length x2}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu2.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu2.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu2.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu2.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu2.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa with increased branch lenght by a factor 2 and 4980 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\subsection{Primate phylogeny - 4980 codons - branch length x4}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu4.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu4.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu4.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu4.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu4.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa with increased branch lenght by a factor 4 and 4980 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\subsection{Primate phylogeny - 4980 codons - branch length x8}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu8.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu8.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu8.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu8.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/PrimatesExons10Mu8.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 61 primate taxa with increased branch lenght by a factor 8 and 4980 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).
Estimated $\hat{\omega}$ and simulated $\omega$ across replicates as a function of mutational bias ($\lambda$) for Muse \& Gaut (panel D) and our tensor model (panel E).

\subsection{Mammalian phylogeny - 4980 codons}

\begin{center}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/MammalsExons10Mu1.0_lambda_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize A\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/MammalsExons10Mu1.0_lambda_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize B\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/MammalsExons10Mu1.0_lambda_GTR.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize C\hspace{4.35cm}}}\hfill
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/MammalsExons10Mu1.0_omega_MG.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize E\hspace{4.35cm}}}
    \begin{minipage}{0.325\linewidth}
        \includegraphics[width=\linewidth, page=1]{inference_supp_mat/MammalsExons10Mu1.0_omega_MF.pdf}
    \end{minipage}
    \llap{\raisebox{1.25cm}{\scriptsize F\hspace{4.35cm}}}
\end{center}
Simulations for 90 mammalian taxa and 4980 codon sites, for 8 different values of mutational bias ($\lambda$) from 0.2 to 5.0 with 5 replicates per value.
Estimated versus true mutational bias, using a codon model in which $\omega$ is modeled as a scalar (Muse \& Gaut formalism, MG, panel A) or as a tensor (mean-field approach, panel B), or by applying a GTR nucleotide model to the 4-fold degenerate third-coding positions only (panel C).

\end{document}