Notations in the main manuscript and the supplementary file are identical, expect that $\EmpiricalDeltaDeltaG$ is simplified to $\DeltaDeltaG$ and $\EmpiricalDeltaGmin$ is simplified to $\DeltaGmin$ in the supplementary (hereby) for formula readability and developments.
$\EmpiricalDeltaDeltaG$ and $\EmpiricalDeltaGmin$ are kept here to refer to the empirical estimations.

\section{\texorpdfstring{$\omega$}{ω} response after a change in \texorpdfstring{$\Ne$}{Nₑ}}
\label{sec:susceptibility-after-a-change-in-Ne}

\subsection{Genotype to phenotype map}
\label{subsec:genotype-to-phenotype-map}

Define $\NbrSites$ as the number of sites in the genotype sequence.
Each site can be in one of $\Nstate \geq 2$ states, where only $1$ state is defined the stable state, and $\Nstate - 1$ states are unstable.
For a given genotype sequence, define {phenotype} $0 \leq x \leq 1$ as the current proportion of sites in the unstable state.
After a mutation, given that only one site can change at a time, the absolute change of $x$ is either $0$ or $\dx=1/\NbrSites$.
Define $\rho_{x}(\dx)$ as the probability to get a change of {phenotype} equal to $\dx$, if the current {phenotype} is $x$:
\begin{gather}
    \begin{cases}
        \dx &\text{ with probability } \rho_{x}(\dx) = 1-x, \\
        0 &\text{ with probability } \rho_{x}(0) = x \left[1 - \frac{1}{\Nstate - 1}\right], \\
        -\dx &\text{ with probability } \rho_{x}(-\dx) = \frac{x}{\Nstate - 1}.\\
    \end{cases} \label{eq:proba-pheno}
\end{gather}

\subsection{Selection coefficient}
\label{subsec:selection-coefficient}

$s(x, \dx)$ is the selection coefficient of an effect $\dx$ if the current {phenotype} is $x$:
\begin{align}
    s(x, \dx) &= \frac{ \wrightfit (x + \dx) - \wrightfit (x)}{ \wrightfit ( x)}, \\
    &\simeq \frac{1}{ \wrightfit (x)}\frac{ \partial \wrightfit (x) }{\partial x} \dx, \\
    &\simeq \frac{ \partial \ln (\wrightfit (x))}{\partial x} \dx, \\
    &\simeq \frac{ \partial \logfit (x) }{\partial x} \dx, \label{eq:s_from_fitness}
\end{align}
where $ \wrightfit ( x)$ is the Wrightian fitness of {phenotype} $x$, and $ \logfit  = \ln (\wrightfit)$ is the log-fitness (or Malthusian fitness).
And the selective effect of the opposite change ($-\dx$) is the opposite selection coefficient:
\begin{gather}
    s(x, -\dx) \simeq - s(x, \dx) \text{ from eq.~\ref{eq:s_from_fitness}} \label{eq:s_minus_deltax}, \\
    \iff S(x, -\dx) \simeq - S(x, \dx), \label{eq:S_minus_deltax}
\end{gather}
where $S(x\eq, \dx) = 4\Ne s (x\eq, \dx)$ is the scaled selection coefficient.

\subsection{Probability of fixation}
\label{subsec:probability-of-fixation}

The probability of fixation of a mutation with effect $\dx$, for a resident {phenotype} $x$ is :
\begin{align}
    \pfix(x, \dx) &= \frac{ 1 - \e^{-2 s(x, \dx)}}{1 - \e^{-4\Ne s(x, \dx)}}, \\
    &\simeq \frac{ 2 s(x, \dx)}{1 - \e^{-4\Ne s(x, \dx)}}, \\
    &= \frac{ 2 s(x, \dx)}{1 - \e^{-S(x, \dx)}}. \label{eq:pfix}
\end{align}
And in the case of {neutral} mutations, the probability of fixation is:
\begin{gather}
    \pfix(x, 0) = \frac{ 1}{2 \Ne} \label{eq:pfix0}.
\end{gather}
And the ratio of probability of fixation between selected and {neutral} mutations is:
\begin{align}
    \frac{\pfix(x, \dx)}{\pfix(x, 0)} & = \frac{ 2 \Ne 2 s(x, \dx)}{1 - \e^{-S(x, \dx)}} \text{ from eq.~\ref{eq:pfix} and~\ref{eq:pfix0}},\\
    & = \frac{S(x, \dx)}{1 - \e^{-S(x, \dx)}} \label{eq:pfixratio}.
\end{align}

\subsection{Equilibrium phenotype}
\label{subsec:equilibrium-phenotype}

At equilibrium {phenotype} $x\eq$, the expected selection coefficient of mutation that reached fixation must be $0$:
\begin{gather}
    0 = \E_{\dx} \left[ s(x\eq, \dx) \pfix(x\eq, \dx) \right], \\
    \iff 0 = \frac{ 2 s(x\eq, \dx)^2}{1 - \e^{-S(x\eq, \dx)}}   \rho_{x\eq}(\dx) + s(x\eq, 0) \frac{ \rho_{x\eq}(0)}{2\Ne} + \frac{ 2s(x\eq, -\dx)^2}{1 - \e^{-S(x\eq, -\dx)}} \rho_{x\eq}(-\dx)\text{ from eq.~\ref{eq:pfix} and~\ref{eq:pfix0}}, \\
    \Longrightarrow \frac{ 2s(x\eq, \dx)^2}{1 - \e^{-S(x\eq, \dx)}}   \rho_{x\eq}(\dx) \simeq \frac{ - 2s(x\eq, \dx)^2}{1 - \e^{S(x\eq, \dx)}}   \rho_{x\eq}(-\dx) \text{ from eq.~\ref{eq:S_minus_deltax}}, \\
    \iff \frac{ \rho_{x\eq}(\dx)}{\rho_{x\eq}(-\dx)} \simeq \e^{-S(x\eq, \dx)} \frac{ \e^{-S(x\eq, \dx)} - 1}{ \e^{-S(x\eq, \dx)} \left( 1 - \e^{S(x\eq, \dx)} \right)}, \\
    \iff \ln \left( \frac{1 - x\eq}{x\eq} \right) + \ln (\Nstate-1) \simeq -S(x\eq, \dx) \text{ from eq.~\ref{eq:proba-pheno}} \label{eq:equilibrium-pheno}, \\
    \iff \lambda_{K}(x\eq) \simeq -S(x\eq, \dx) \label{eq:equilibrium_lambda},
\end{gather}
where  $\lambda_{K}(x\eq) = \ln \left( \frac{1 - x\eq}{x\eq} \right) + \ln (\Nstate-1)$.

\subsection{Relative substitution rate (\texorpdfstring{$\omega$}{ω}) at equilibrium}
\label{subsec:mean-scaled-fixation-probability-omega-at-equilibrium}

The {substitution} rate of all selected relative to the {substitution} rate of {neutral} mutations is denoted $\omega$, which can also be interpreted as the mean fixation probability of mutations scaled by the fixation probability of {neutral} mutations $p = 1 / {2 \Ne}$.
\begin{align}
    \omega & = \E_{\dx} \left[ \frac{\pfix(x, \dx)}{\pfix(x, 0)} \right], \\
    & = (1 - x) \frac{ S(x, \dx)}{1 - \e^{-S(x, \dx)}} + x \left(\frac{\Nstate - 2}{\Nstate - 1}\right) + \frac{x}{\Nstate-1} \frac{ S(x, -\dx)}{1 - \e^{-S(x, -\dx)}} \text{ from eq.~\ref{eq:proba-pheno},~\ref{eq:pfix} and~\ref{eq:pfix0}}, \\
    & = (1 - x) \frac{ S(x, \dx)}{1 - \e^{-S(x, \dx)}} - \frac{x}{\Nstate-1}  \frac{ S(x, \dx)}{1 - \e^{S(x, \dx)}} + x \left(\frac{\Nstate - 2}{\Nstate - 1}\right) \text{ from eq.~\ref{eq:S_minus_deltax}}.
\end{align}
$\omega\eq$ at equilibrium is then determined by the {phenotype} at equilibrium $x\eq$:
\begin{align}
    \omega\eq &= (1 - x\eq) \frac{ S(x\eq, \dx)}{1 - \e^{-S(x\eq, \dx)}} - \frac{x\eq}{\Nstate-1} \frac{ S(x\eq, \dx)}{1 - \e^{S(x\eq, \dx)}} + x\eq \left(\frac{\Nstate - 2}{\Nstate - 1}\right) , \\
    &= x\eq \left[ \frac{2 (x\eq-1)  \lambda_{K}(x\eq) }{\Nstate (x\eq-1)+1} + \frac{\Nstate - 2}{\Nstate - 1}\right] \text{ from eq.~\ref{eq:equilibrium-pheno}}.
\end{align}
\begin{center}
    \includegraphics[width=0.8\textwidth, page=2] {analytical-relaxation}
\end{center}
Moreover, given that the number of state if large enough $\Nstate \gg 1$, the equilibrium $\omega$ can be approximated as:
\begin{align}
    \omega\eq  & = x\eq \left[ \frac{2 (x\eq-1)  \lambda_{K}(x\eq) }{\Nstate (x\eq-1)+1} + \frac{\Nstate - 2}{\Nstate - 1}\right], \\
    & \simeq x\eq
\end{align}
And the derivative of $\omega\eq$ w.r.t to $x\eq$ is:
\begin{gather}
    \frac{\der \omega\eq}{\der x\eq} = 2 \left[ \frac{\Nstate (x\eq- 1) + 1+\left[\Nstate (x\eq-1)^2+2 x\eq-1\right] \lambda_{K}(x\eq) }{(\Nstate (x\eq-1)+1)^2}\right] + \frac{\Nstate - 2}{\Nstate - 1} \label{eq:dw_dx}.
\end{gather}
\begin{center}
    \includegraphics[width=0.8\textwidth, page=3] {analytical-relaxation}
\end{center}
Moreover, given that the number of state if large enough $\Nstate \gg 1$, the response in equilibrium $\omega$ due to change in {phenotype} can be approximated as:
\begin{align}
    \frac{ \der \omega\eq}{\der x\eq}  & = 2 \left[ \frac{\Nstate (x\eq- 1) + 1+\left[\Nstate (x\eq-1)^2+2 x\eq-1\right] \lambda_{K}(x\eq) }{(\Nstate (x\eq-1)+1)^2}\right] + \frac{\Nstate - 2}{\Nstate - 1}, \\
    & \simeq \frac{2 \lambda_{K}(x\eq)}{\Nstate} + 1, \\
    & \simeq 1. \label{eq:approx_domega_dx}
\end{align}

\subsection{\texorpdfstring{$\omega$}{ω} response after a change in \texorpdfstring{$\Ne$}{Nₑ}}
\label{subsec:omega-susceptibility-after-a-change-in-Ne}

Define the function $G(x, \Ne )$ as:
\begin{gather}
    G(x, \Ne ) \equiv \lambda_K (x\eq) + 4 \Ne s(x, \dx),
\end{gather}
The equilibrium equation (eq.~\ref{eq:equilibrium-pheno}) states that $G(x\eq, \Ne )=0$, meaning that $x\eq$ is implicitly a function of $\Ne$:
\begin{gather}
    G(x\eq(\Ne), \Ne ) = 0, \\
    \Longrightarrow \frac{\partial G(x\eq, \Ne ) }{\partial x\eq }\frac{ \der x\eq}{\der \Ne} + \frac{ \partial G(x\eq, \Ne )}{\partial \Ne} = 0, \\
    \iff \left[  \frac{\partial \lambda_K (x\eq) }{\partial x\eq }  + 4 \Ne \frac{ \partial s(x\eq, \dx) }{\partial x\eq } \right]\frac{ \der x\eq}{\der \Ne} + 4 s(x\eq, \dx) = 0, \\
    \iff \left[  \frac{\partial \lambda_K (x\eq) }{\partial x\eq } + 4 \Ne \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}\dx \right]\frac{ \der x\eq}{\der \Ne}  = - 4 \frac{ \partial \logfit (x\eq) }{\partial {x\eq}}\dx \text{ from eq.~\ref{eq:s_from_fitness}}, \\
    \iff 4 \dx \left[ \frac{1}{4 \dx \Ne } \frac{\partial \lambda_K (x\eq) }{\partial x\eq } + \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2} \right] \Ne \frac{ \der x\eq}{\der \Ne}  = - 4 \dx \frac{ \partial \logfit (x\eq) }{\partial {x\eq}}, \\
    \iff \frac{ \der x\eq}{\der \ln (\Ne)}  = - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{1}{4 \dx \Ne} \frac{\partial \lambda_K (x\eq) }{\partial x\eq } + \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}}  \label{eq:dx_dlnNe}.
\end{gather}
Giving the equation for the response of {phenotype} at equilibrium after a change of {effective population size}.
Together, the response of {substitution} rate at equilibrium, after a change of {effective population size} can be obtained as:
\begin{align}
    \frac{\der \omega\eq}{\der \ln (\Ne)} & = \frac{\der \omega\eq}{\der x\eq} \frac{ \der x\eq}{\der \ln (\Ne)}, \\
    &= - \frac{\der \omega\eq}{\der x\eq} \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{1}{4 \dx \Ne } \frac{\partial \lambda_K (x\eq) }{\partial x\eq } + \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}} \text{ from eq.~\ref{eq:dx_dlnNe}} \label{eq:dw_dlnNe}.
\end{align}
Moreover, with the approximation that $\left| 4 \Ne \frac{ \partial s(x\eq, \dx) }{\partial x\eq } \right| \gg \left| \frac{\partial \lambda_K (x\eq) }{\partial x\eq } \right|$, meaning that a change in {phenotype} causes a higher change in scaled selection coefficient than mutational bias, we have:
\begin{gather}
    \frac{ \der x\eq}{\der \ln (\Ne)}  = - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{1}{4 \dx \Ne} \frac{\partial \lambda_K (x\eq) }{\partial x\eq } + \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}},\\
    \Longrightarrow \frac{ \der x\eq}{\der \ln (\Ne)}  \simeq - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}}. \label{eq:approx_dx_dlnNe}
\end{gather}
Together, these approximations leads to the following response in equilibrium $\omega$ after change in $\Ne$ as:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (\Ne)} & \simeq - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}}
    \label{eq:approx_dw_dlnNe}
\end{align}


\section{Models for the log-fitness function}
\label{sec:models-for-the-log-fitness-function}

\subsection{Folded fraction}
\label{subsec:folded-fraction}

All phenotype-fitness functions considered below are log-concave, and as a result, $\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}$ is a decreasing function of $x$; the less stable the protein already is, the stronger the purifying selection against additional destabilizing mutations. More precisely, fitness functions depends on the folded fraction of the protein of interest, which is given by the Fermi-Dirac distribution:
\begin{align}
    \probaFold (x) & = \frac{1}{1 + e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x)}},
\end{align}
where $x$ is the fraction of destabilizing mutations, each contributing to $\DeltaDeltaG$ in free energy of folding (also denoted empirically $\EmpiricalDeltaDeltaG$), and $\beta = 1 / kT$.
Thus, $\DeltaGmin < 0$ is the difference in free energy between folded and unfolded state when all sites are stable (also denoted empirically $\EmpiricalDeltaGmin$).
As a result, $\NbrSites \DeltaDeltaG$ is thus the expected change in $\DeltaG$ when all sites are considered unstable.
The misfolded fraction is  $\probaUnfold = 1-\probaFold$.
In addition, $\probaFold$ is typically close to 1 (or $\probaUnfold \ll 1$), so that we can use a first-order approximation:
\begin{align}
    \probaFold(x) & = 1 - \probaUnfold(x) \\
    & \simeq 1 - e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x)}
\end{align}
or equivalently
\begin{align}
    \probaUnfold(x) & \simeq e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x)}
\end{align}

\subsection{Fitness equal to folded fraction}
\label{subsec:fitness-equal-to-folded-fraction}

A first model is to assume that the fitness is equal to the folded fraction~\citep{Goldstein2013}:
\begin{gather}
    \wrightfit (x) = \frac{1}{1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}}. \label{eq:fitness}
\end{gather}
The derivative of fitness w.r.t to {phenotype} is:
\begin{align}
    \frac{\partial  \logfit (x)}{\partial x}  & = - \frac{\partial \ln \left( 1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)} \right)}{\partial x} \text{ from eq.~\ref{eq:fitness}}, \\
    & = - \beta \NbrSites \DeltaDeltaG \frac{\e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}}{1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}} \label{eq:df_dx},\\
    & \simeq - \beta \NbrSites \DeltaDeltaG \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}.
\end{align}
The equilibrium {phenotype} ($x\eq$) is :
\begin{gather}
    \lambda_{K}(x\eq) = 4\Ne \beta \DeltaDeltaG \frac{\e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x\eq)}}{1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x\eq)}}  \text{ from eq.~\ref{eq:equilibrium_lambda} and~\ref{eq:df_dx}}.
\end{gather}
Using $\Ne=10^4$, $\beta=1.686$, $\DeltaGmin = \EmpiricalDeltaGmin = -118$, $\NbrSites=300$, $\DeltaDeltaG=\EmpiricalDeltaDeltaG=1$, we have the following :
\begin{center}
    \includegraphics[width=0.8\textwidth, page=4] {analytical-relaxation}
\end{center}
Where in this example we can visually appreciate that the a change in {phenotype} causes a higher change in scaled selection coefficient than mutational bias (eq.~\ref{eq:approx_dx_dlnNe}).
And the second derivative of fitness w.r.t to {phenotype} is:
\begin{align}
    \frac{\partial^2  \logfit (x)}{\partial x^2} & = - \beta \NbrSites \DeltaDeltaG \frac{\partial}{\partial x} \left( \frac{\e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}}{1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}} \right) \text{ from eq.~\ref{eq:df_dx}}, \\
    & = - \beta \NbrSites \DeltaDeltaG  \beta \NbrSites \DeltaDeltaG \frac{\e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}}{\left( 1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}\right)^2}, \\
    & = \frac{\beta \NbrSites \DeltaDeltaG}{1 + \e^{\beta(\DeltaGmin + \NbrSites \DeltaDeltaG x)}} \frac{ \partial \logfit (x) }{\partial x} \text{ from eq.~\ref{eq:df_dx}} \label{eq:df2_dx2}, \\
    & \simeq \beta \NbrSites \DeltaDeltaG \frac{ \partial \logfit (x) }{\partial x}
\end{align}
Finally, $\omega$ response after a change in $\Ne$ is simply:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (\Ne)} & \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG} \text{from eq.~\ref{eq:df2_dx2} and~\ref{eq:approx_domega_dx}},
\end{align}
which is independent of $x\eq$, meaning $\omega$ is linearly decreasing with $\Ne$ in log space.
This model, however, does not express the fact that selection is typically stronger for proteins characterized by higher levels of expression.

\subsection{Selective cost proportional to amount of misfolded protein}
\label{subsec:selective-cost-proportional-to-amount-of-misfolded-protein}

A slight variation is to assume that the selective cost itself is proportional to the total amount of misfolded protein~\citep{Drummond2005a, Wilke2006, Drummond2008, Serohijos2012}. For a given protein with expression level $y$:
\begin{align}
    \logfit (x) = - A y \probaUnfold (x),
\end{align}
where $A$ is the cost per misfolded macromolecule.
Then,
\begin{align}
    \frac{\partial  \logfit (x)}{\partial x} & \simeq -A y \beta \DeltaDeltaG \NbrSites e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x)}.
\end{align}
Under this model, the {phenotype} at equilibrium is given by:
\begin{align}
    \lambda_{K}(x\eq) = 4\Ne y A \beta \DeltaDeltaG \NbrSites e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x\eq)} \text{ from eq.~\ref{eq:equilibrium_lambda} and~\ref{eq:df_dx}}.
\end{align}
And the response of $\omega$ after a change in $\Ne$ is the same as before:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (\Ne)} & \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG}.
\end{align}
Since $\Ne$ and $y$ are confounded factors, meaning they only appear in the equation as a product between the two, implicit derivation leads to the same result whenever the derivation is w.r.t $\Ne$ or $y$, leading to same compact equation:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (y)} = \frac{ \der \omega\eq}{\der \ln (\Ne)} \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG}.
\end{align}

\subsection{Translational errors}
\label{subsec:translational-errors}

Another variant account for translational errors.
Translational errors occur at a rate $\rho$ per residue.
These errors contribute additional destabilizing mutations, each with effect size $\dx = 1/\NbrSites$.
The total number of translational errors per macromolecule is approximately Poisson distributed:
\begin{align}
    \pi_k & = e^{-\rho \NbrSites} \frac{(\rho \NbrSites)^k}{k!}
\end{align}
and the total selective cost is now an average over all possible values of $k$:
\begin{align}
    \logfit (x) & = - A y \sum_k \pi_k e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x + \DeltaDeltaG k)}
    \\
    & = - A y e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)} \sum_k e^{-\rho \NbrSites} \frac{(\rho \NbrSites)^k}{k!} e^{\beta \DeltaDeltaG k}
    \\
    & = - A y e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x) + \rho \NbrSites (e^{\beta \DeltaDeltaG} -1 )}
    \\
    & \simeq - A y e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x) + \rho \beta \DeltaDeltaG \NbrSites}
    \\
    & =- A y e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites (x + \rho))}
\end{align}
In words, the fitness function is the same the previous model, except that the trait $x$ (fraction of destabilizing mutations) is shifted by $\rho$, the mean fraction of additional mutations contributed by translation errors. This additional factor is independent of $x$, and as a result, the scaled selection strength is essentially the same, up to a proportionality constant (contributed by the shift):
\begin{align}
    4 \Ne \frac{\partial  \logfit (x)}{\partial x} & \propto -4 \Ne y e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites (x + \rho))}
    \\ & \propto -4 \Ne y e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)}
\end{align}
Moreover, $\omega$ response after a change in $\Ne$ is again the same as before:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (\Ne)} & \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG}.
\end{align}

\subsection{Cost-benefit argument}
\label{subsec:cost-benefit-argument}

The cost-benefit argument~\citep{Beaulieu2018} is based on two assumptions
\begin{enumerate}
    \item the expression level is regulated so that the total number of \emph{functional} macromolecules is maintained at a target level $y$;
    \item the log-fitness is proportional to the ratio of the \emph{total} cost of expression over the benefit contributed by the protein.
\end{enumerate}
Specifically, the protein is assumed to be regulated so as to reach a level of expression of functional proteins of $y$, and contributes a total benefit $B$ (which depends on its specific function). Given that only a fraction $\probaFold(x) = 1-\probaUnfold(x)$ of the total amount of protein expressed by the cell is functional, the total cost of expression $C$ is then equal to:
\begin{align}
    C(x) & = \frac{y}{\probaFold(x)}
    \\ & \simeq y (1 + \probaUnfold(x))
\end{align}
Then, the log-fitness is given by:
\begin{align}
    \logfit (x) & = -A \frac{y}{B} \left( 1 + e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)} \right)
    \\
    & = -by (1 + e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)}),
\end{align}
where $b = A /B$.
Compared to models 2 (section~\ref{subsec:selective-cost-proportional-to-amount-of-misfolded-protein}) and 3 (section~\ref{subsec:translational-errors}), the log-fitness now has an additional term that depends on the target expression level $y$, but not on trait $x$. The scaled strength of selection on mutations affecting $x$ has thus the same functional form as for the two previous models:
\begin{align}
    4 \Ne \frac{\partial  \logfit (x)}{\partial x} & \propto -4 \Ne y e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)}
\end{align}
Alternative cost-expression models could also be used, allowing for a non-linear cost function for expression or for some susceptibility of the realized equilibrium expression level, as a function of the number of mutations.
Under these models, the strength of selection is still expected to be an increasing function of $y$, although not linear:
\begin{align}
    4 \Ne \frac{\partial  \logfit (x)}{\partial x} & \propto -4 \Ne g(y) e^{\beta(\DeltaGmin + \DeltaDeltaG\NbrSites x)},
\end{align}
where $g$ is some function of $y$.
Moreover, $\omega$ response after a change in $\Ne$ is again the same as before:
\begin{align}
    \frac{ \der \omega\eq}{\der \ln (\Ne)} & \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG}.
\end{align}


\section{Model of protein-protein interactions}
\label{sec:model-of-protein-protein-interactions}

The proteome is assumed to be composed of $m$ protein species, all with same abundance $C$.
Each macromolecule may either be in free form or engaged in a non-specific interaction.
Only pairwise interactions are considered, and higher-order interactions are ignored.
The equilibrium is characterized by:
\begin{align}
[ij]
    & = \frac{[i][j]}{C_0} \, e^{\beta E_{ij}},
\end{align}
where $[i]$ and $[j]$ are the concentrations of protein species $i$ and $j$, and $[ij]$ is the concentration of their (non-specific) dimer. Here, $E_{ij}$ is the interaction free energy, which can itself be decomposed as a sum of three terms:
\begin{align}
    E_{ij} & = \DeltaGmin + E_i + E_j
    \\
    & = \DeltaGmin + \DeltaDeltaG \NbrSites (x_i + x_j),
\end{align}
where we assume that each protein has $n=100$ residues at its surface, $x_i$ stands for the fraction of hydrophobic residues at the surface of protein $i$, and each hydrophobic residue makes an additive contribution of $\EmpiricalDeltaDeltaG$ to the total.

By conservation of the total number of molecules:
\begin{align}
    C & = [i] + \sum\limits_{j \neq i} [ij] \\
    & = [i] + \sum\limits_{j \neq i} \frac{[i][j]}{C_0} \, e^{\beta E_{ij}}
\end{align}
and we note:
\begin{align}
    \epsilon_i = \sum_j [ij]
\end{align}
the fraction of protein $i$ sequestered in non-specific interactions.
We assume that the log fitness is proportional to the total amount of protein sequestered in non-specific interactions:
\begin{align}
    \logfit (x) & = -b \sum_i \epsilon_i,
\end{align}
where $b>0$ is a parameter determining the overall stringency of selection against non-specific interactions.

\subsection{Mean field, weak-interaction limit}
\label{subsec:mean-field-interaction-limit}

To make the model tractable and compact, we assume that non-specific interactions are weak, i.e.~$\epsilon_i \ll 1$ for all $i$.
We then make a first-order approximation in the $\epsilon_i$'s.
In addition, we use a mean-field approximation, such that, when considering a specific protein species $i$, we assume that all other proteins have the same fraction $\bar x$ of hydrophobic residues at their surface.
The value of $\bar x$ could in principle be found using a self-consistent argument, essentially by (1) explicitly calculating the net {substitution} flux for protein $i$ with fraction $x_i$, under mean field $\bar x$, and (2) expressing the constraint that this {substitution} process for protein $i$ is stationary at $x_i = \bar x$.
This derivation is not conducted here, as it is not needed.
Using these approximations, we can re-express the conservation of total mass as:
\begin{align}
    C & = [i] + (m-1) [i] \frac{C}{C_0} \, e^{\beta (\DeltaGmin + \DeltaDeltaG \NbrSites (\bar x + x_i))}
\end{align}
Here, we have used the fact that $[j] = C(1 - \epsilon_j)$ can be approximated as $[j] \simeq C$ since it is involved in a term already of the order of $\epsilon_i$. As a result, all $m-1$ terms of the sum over $j\neq i$ are identical.
Next, solving for $[i]$ gives:
\begin{align}
[i]
    & = \frac{C} {1 + (m-1) \frac{C}{C_0} \, e^{\beta (\DeltaGmin + \DeltaDeltaG \NbrSites (\bar x + x_i))}}
    \\ & \simeq C \left( 1 - m \frac{C}{C_0} \, e^{\beta (\DeltaGmin + \DeltaDeltaG \NbrSites (\bar x + x_i))} \right)
    \\ & =
    C (1 - \epsilon_i)
\end{align}
and thus $\epsilon_i$ can be identified with:
\begin{align}
    \epsilon_i  & = m \frac{C}{C_0} \, e^{\beta (\DeltaGmin +\DeltaDeltaG \NbrSites (\bar x + x_i))}
\end{align}
Now, assume that the system is at equilibrium (thus $x_i = \bar x$).
The strength of selection acting on mutations occurring at the surface of protein $i$, of effect size $\dx = \pm 1/\NbrSites$, is given by $s = \kappa \dx$ where:

\begin{align}
    \kappa_i & = b \frac{\der\epsilon_i} {\der x_i}
    \\ & =
    b \beta \DeltaDeltaG \NbrSites m \frac{C}{C_0} \, e^{\beta (\DeltaGmin +\DeltaDeltaG \NbrSites (\bar x + x_i))}
\end{align}
and thus:
\begin{align}
    \ln (\kappa_i) & = \ln \left( b \beta \DeltaDeltaG \NbrSites m \frac{C}{C_0}  \right) \, + \, \beta (\DeltaGmin +\DeltaDeltaG \NbrSites \bar x) \, + \, \beta \DeltaDeltaG\NbrSites x_i,
\end{align}
where only the last term depends on $x_i$.
Finally, applying the main result of this work to the present case allows us to express the response of $\omega$ as a function of $\Ne$ as:
\begin{align}
    \chi & = \frac{\der\omega} {\der \ln (\Ne)}
    \\ & =  2(\lambda - 1) \, \frac{\der\ln \kappa_i}{\der x_i}
    \\ & =  2 (\lambda - 1) \, \frac{1}{\beta \DeltaDeltaG \NbrSites}
\end{align}
Note that, here, we have used $K=2$ (hydrophobic and polar residues are roughly equally likely to occur by mutation), and assumed $x^* \ll 1$.
A more accurate formula could be used without this latter assumption.
In any case, $\chi$ is now dependent on $x^*$, through $\lambda$.

\subsection{Empirical calibration}
\label{subsec:empirical-calibration}

Based on empirical estimates found in \citet{Zhang2008}.
The mean fraction of hydrophobic residues at the surface of proteins is $0.22 \pm 0.06$. With $n=100$ residues, this makes $22 \pm 6$.
The mean value for $E_{ij}$ is $7kT$, with a standard deviation of $\sigma = 1.8kT$.
Assuming that this standard deviation of $\pm 1.8kT$ is contributed by $\pm 6$ mutations gives $\EmpiricalDeltaDeltaG = 1.8 / 6 = 0.3$ kT or $0.18$ kcal per mole.
Also, with $x=0.22$, $\lambda \simeq 4$, and thus $\chi = 6 / 30 = 0.2$, thus a much stronger response than under the model based on conformational stability.


\section{Empirical estimation}
\label{sec:empirical-estimation}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
        \begin{tabu}{|l|l|c|c|}
            \hline
            \textbf{Type} & \textbf{Specie} & $\bm\hat{\chi}{}$ & $\bm{r^2}$ \\
            \hline
            \hline Plant & Oryza sativa & -0.008 & 0.047 \\
            \hline Plant & Arabidopsis thaliana & -0.012 & 0.128 \\
            \hline Archaea & Sulfolobus solfataricus & -0.037 & 0.097 \\
            \hline Archaea & Thermococcus kodakarensis & -0.026 & 0.058 \\
            \hline Fungi & Saccharomyces cerevisiae & -0.029 & 0.211 \\
            \hline Fungi & Aspergillus nidulans & -0.034 & 0.124 \\
            \hline Bacteria & Escherichia coli & -0.021 & 0.151 \\
            \hline Bacteria & Bacillus subtilis & -0.046 & 0.151 \\
            \hline Animal & Caenorhabditis elegans & -0.026 & 0.039 \\
            \hline Animal & Drosophila melanogaster & -0.005 & 0.021 \\
            \hline Animal & Mus musculus & -0.008 & 0.085 \\
            \hline Animal & Homo sapiens & -0.004 & 0.031 \\
            \hline
        \end{tabu}}
    \caption[Substitution rate as a function of expression level]{Substitution rate as a function of expression level compiled by \citet{Zhang2015}.}
    \label{tab:table-susceptibility}
\end{table}


In \citet{Brevet2019}, the covariance matrix with $\ln(\Ne)$ and $\ln(\omega)$ as entries allows to approximate $\chi$:
\begin{align}
    \widehat{\frac{\der \ln (\omega)}{\der \ln (\Ne)}} & = \dfrac{\text{Cov}[\ln(\omega), \ln(\Ne)]}{\text{Var}[\ln(\Ne)]}, \\
    \Rightarrow \widehat{\frac{\der \omega}{\omega \der \ln (\Ne)}} & = \dfrac{\text{Cov}[\ln(\omega), \ln(\Ne)]}{\text{Var}[\ln(\Ne)]}, \\
    \Rightarrow \widehat{\chi} & \simeq \widehat{\omega} \dfrac{\text{Cov}[\ln(\omega), \ln(\Ne)]}{\text{Var}[\ln(\Ne)]}, \\
    \Rightarrow \widehat{\chi} & \simeq 0.2 \dfrac{-0.45}{4.45}, \\
    \Rightarrow \widehat{\chi} & \simeq - 0.02
\end{align}


\section{Simulation using the 3D structure of protein}
\label{sec:simulation-using-the-3d-structure-of-protein}

We simulated {substitutions} in the protein phosphatase ($\Nsite=300$ {codon} sites) as in \citet{Goldstein2017}.
From a {DNA} sequence $\Seqi$ after $t$ {substitutions}, we compute the free energy of the folded state $ \GFold \left(\Seqi\right)$, using the 3-dimensional structure of the folded state and pair-wise contact energies between neighboring amino-acid residues:
\begin{equation}
    \GFold \left(\Seqi\right) = \sum\limits\sumSetsite \sum\limits_{r \in \mathcal{V}(\site)} I \left(\Seqi(\site), \Seqi(r) \right),
\end{equation}
where $I(a,b)$ is the pair-wise contact energies between amino acid $a$ and $b$, using contact potentials estimated by \citet{Miyazawa1985}, and $\mathcal{V}(\site)$ are the neighbor residues of site $\site$ (closer than $7\angstrom$) in the 3D structure.\\

The free energy of unfolded states $\GUnfold \left(\Seqi\right)$ is approximated using $55$ decoy 3D structures that supposedly represent a sample of possible unfolded states:
\begin{equation}
    \GUnfold \left(\Seqi\right) = \left\langle G\left(\Seqi\right) \right\rangle - kT \ln (1.0\text{E}^{160}) - \dfrac{2 \left[ \left\langle G\left(\Seqi\right)^2 \right\rangle - \left\langle G\left(\Seqi\right) \right\rangle^2\right] }{kT}
\end{equation}
where the average $\left\langle . \right\rangle$ runs other the $55$ decoy 3D structures, and $k$ is the Boltzmann constant and $T$ the temperature in Kelvin.\\

From the energy of folded and unfolded states, we can compute the difference in free energy between the states:
\begin{equation}
    \phenoFold\left(\Seqi\right) = \GFold \left(\Seqi\right) - \GUnfold \left(\Seqi\right)
\end{equation}

\begin{center}
    \includegraphics[width=\textwidth] {ModelSimuFold.pdf}
\end{center}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth]{SimuFold-Elasticity-DG.pdf}
    \end{minipage}%
    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth]{SimuStab-Elasticity-DG.pdf}
    \end{minipage}
    \caption[$\DeltaG$ response to changes in $\Ne$]{
        $\DeltaG$ response to changes in $\Ne$.
        Left Panel: Model of folding free energy computed using 3D structural conformations and pairwise contact potential energies between neighbouring amino-acid residues.
        Right Panel: Additive {phenotype} model, where for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid.
        Scaling experiment simulating sequence evolution and recording the average $\DeltaG$ (y-axis) observed at equilibrium as a function of $\Ne$ (x-axis).
        Along the x-axis, $200$ replicate simulations are performed for each different $\Ne$, the average (solid lines) and $90\%$ confidence interval (shaded area) of $\omega$ are shown.
        $\DeltaG$ is linearly dependent on log-$\Ne$, with a slope equal to $1/\beta=0.593$.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth]{SimuFold-DG-DDG.pdf}
    \end{minipage}%
    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth]{SimuStab-DG-DDG.pdf}
    \end{minipage}

    \caption[$\EmpiricalDeltaDeltaG$ correlation to $\DeltaG$]{
        $\EmpiricalDeltaDeltaG$ correlation to $\DeltaG$.
        Left Panel: Model of folding free energy computed using 3D structural conformations and pairwise contact potential energies between neighbouring amino-acid residues.
        Right Panel: Additive {phenotype} model, where for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid.
        Simulations are performed for $\Ne$ varying from $10^2$ to $10^8$, where each dot is an independent simulation at equilibrium.
        Along each simulation, the average $\EmpiricalDeltaDeltaG$ of all proposed mutations is recorded (y-axis), and represented as a function of the average $\DeltaG$ (x-axis).
        $\EmpiricalDeltaDeltaG$ is negatively correlated to $\Delta$, which is expected since protein under higher $\Ne$ are more stable (lower $\DeltaG$, see above), and mutations are more destabilizing on average.
        To be more precise, the negative correlation between $\EmpiricalDeltaDeltaG$ and $\DeltaG$ is observed empirically with a linear fit of $\EmpiricalDeltaDeltaG = -0.13\DeltaG + 0.23$~\citep{Serohijos2012}.
        This correlation is a necessary condition for observing a response of $\omega$ to changes in $\Ne$~\citep{Goldstein2013} and protein expression level~\citep{Serohijos2012}.
    }
    \label{fig:DG-vs-DDG}
\end{figure}


\section{Simulated \texorpdfstring{$\omega$}{ω} response to changes in \texorpdfstring{$\Ne$}{Nₑ}}
\label{sec:simulated-omega-response-to-changes-Ne}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuDfe-Elasticity.pdf}
    \caption[$\omega$ response with gamma distributed selection coefficient]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), under a model of gamma distributed selection coefficient.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        In the model of gamma distributed fitness effect, $\omega$ at equilibrium is strongly dependent on log-$\Ne$ where the slope correlation is proportional to the inverse of the shape parameter of the gamma distribution~\citep{Welch2008}.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuProfile-Elasticity.pdf}
    \caption[$\omega$ response to changes in $\Ne$ under a model of site-specific amino-acid fitness profiles]{
        $\omega$ at equilibrium as a function of $\Ne$ (relative), under a model of amino-acid fitness profiles.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        In the model of site-wise amino-acid fitness profiles taken from~\citep{Bloom2017}, $\omega$ at equilibrium is strongly dependent on log-$\Ne$.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-Elasticity.pdf}
    \caption[$\omega$ response to changes in $\Ne$ under a model of additive free energy of folding]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), for a model of additive free energy of folding.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The fixed parameters are $\DeltaGmin=-118$, $\DeltaDeltaG=1$, $\NbrSites=300$, $\beta=1.686$.
        The simulations of our additive free energy model match the theoretical prediction that the slope of the linear relation (dashed line) is equal to $\beta \NbrSites \DeltaDeltaG)^{-1} = 0.00198 \simeq = 0.00199$.
        The non-monotony is suspected to be due to the discrete number of sites and states, such that the changes in $\DeltaG$ after a mutation is either $-1$, $0$ or $1$. Such non-monotony is not observed with the Grantham model, in which the $\omega$ is lower and the slope of the response is lower, closer to the empirical 3D model.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-Grantham-Beta-Elasticity.pdf}
    \caption[Effect of $\beta$ on the $\omega$ response to changes in $\Ne$]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), for various parameter $\beta$.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The fixed parameters are $\DeltaGmin=-118$, $\DeltaDeltaG=1$, $\NbrSites=300$, and for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid. $\beta$ are given in the legend.
        Increasing $\beta$ decreases the slope of the $\omega$-$\Ne$ relationship, as predicted in our theoretical model.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-Grantham-Nsite-Elasticity.pdf}
    \caption[Effect of sequence size on the $\omega$ response to changes in $\Ne$]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), for various sequence size.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The fixed parameters are $\DeltaGmin=-118$, $\DeltaDeltaG=1$, $\beta=1.686$, and for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid. $\NbrSites$ are given in the legend.
        Increasing $\NbrSites$ decreases the slope of the $\omega$-$\Ne$ relationship, as predicted in our theoretical model.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-Grantham-Gamma-Elasticity-ExpressionLevel.pdf}
    \caption[Effect of $\DeltaDeltaG$ on the $\omega$ response to changes in $\Ne$ with regards to expression level]{
        $\omega$ at equilibrium as a function of the expression level $y$ (log scale), for various value of $\DeltaDeltaG$.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The fixed parameters are $\DeltaGmin=-118$, $\beta=1.686$, $\NbrSites=300$, and for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid. $\DeltaDeltaG$ are given in the legend.
        Increasing $\DeltaDeltaG$ increases the slope of the $\omega$-$y$ relationship, as predicted in our theoretical model.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-Grantham-GammaStd-Elasticity.pdf}
    \caption[Effect of site variance on the $\omega$ response to changes in $\Ne$]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), for various between site variance.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The parameters are $\DeltaGmin=-118$, $\NbrSites=300$, $\beta=1.686$ and each site has its own gamma distributed $\DeltaDeltaG$ with mean $1$ and standard deviation given in the legend. $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid.
        Increasing the variance of $\DeltaDeltaG$ increases $\omega$, by shifting the equilibrium to higher $x\eq$ since more unstable sites with low $\DeltaDeltaG$ are fixed before reaching sensible deleterious selection coefficient against unstable mutations. Once many sites are unstable, the $\omega$ is higher since {non-synonymous} mutations between unstable states are effectively {neutral}. However the slope of the $\omega$-$\Ne$ relationship is not sensibly changed.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {SimuStab-GammaStd-Elasticity.pdf}
    \caption[Effect of site variance on the $\omega$ response to changes in $\Ne$ without Grantham distance]{
        $\omega$ at equilibrium as a function of $\Ne$ (log scale), for various between site variance under a model considering Grantham distances.
        For each population size, $200$ simulations were performed and the average (solid line) and $90\%$ confidence interval (shaded area) are shown.
        The parameters are $\DeltaGmin=-118$, $\NbrSites=300$, $\beta=1.686$ and each site has its own gamma distributed $\DeltaDeltaG$ with mean $1$ and standard deviation given in the legend.
        Increasing the variance of $\DeltaDeltaG$ increases $\omega$, by shifting the equilibrium to higher $x\eq$ since more unstable sites with low $\DeltaDeltaG$ are fixed before reaching sensible deleterious selection coefficient against unstable mutations. Once many sites are unstable, the $\omega$ is higher since {non-synonymous} mutations between unstable states are effectively {neutral}. However the slope of the $\omega$-$\Ne$ relationship is not sensibly changed.
    }
\end{figure}


\section{Simulated relaxation time of \texorpdfstring{$\omega$}{ω}}
\label{sec:simulated-relaxation-time-of-omega}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {Relaxation-Stability-Alpha.pdf}
    \caption[Relaxation time of $\omega$ dependence on $\NbrSites$, while correction for $\DeltaGmin$]{
        $\omega$ Relaxation after a brutal change in $\Ne$, for various $\NbrSites$ while correcting for $\DeltaGmin$.
        The left and right panel correspond to low $\Ne$ ($1e^{5}$) and the middle panel corresponds to high $\Ne$ ($2e^{6}$).
        Solid line corresponds to the average over replicates ($r$) and the shaded area correspond to the $90\%$ interval among replicates.
        The mutation rate ($\mu$) is $1e{-8}$ per year per site, and the total time of the computation is $900$ million years.
        $\beta=1.686$, $\DeltaDeltaG=0.2$ for all simulations. The number of sites is changed from $\NbrSites=15$ to $\NbrSites=158$, and the number of replicates is changed accordingly such that the total number of sites ($\NbrSites*r$) is kept constant.
        Moreover, $\DeltaGmin$ is changed according to $\NbrSites$ and $\DeltaDeltaG$ such that the equilibrium value $x\eq$ is kept constant, by solving numerically equation~\ref{eq:equilibrium-pheno}.
        Increasing $\NbrSites$ implies a higher rate of relaxation.}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {Relaxation-Stability.pdf}
    \caption[Relaxation time of $\omega$ dependence on $\NbrSites$]{
        $\omega$ Relaxation after a brutal change in $\Ne$, for various $\NbrSites$.
        The left and right panel correspond to low $\Ne$ ($1e^{5}$) and the middle panel corresponds to high $\Ne$ ($2e^{6}$).
        Solid line corresponds to the average over replicates ($r$) and the shaded area correspond to the $90\%$ interval among replicates.
        The mutation rate ($\mu$) is $1e{-8}$ per year per site, and the total time of the computation is $900$ million years.
        $\beta=1.686$, $\DeltaDeltaG=0.2$ and $\DeltaGmin=-10$ for all simulations. The number of sites is changed from $\NbrSites=15$ to $\NbrSites=158$, and the number of replicates is changed accordingly such that the total number of sites ($\NbrSites*r$) is kept constant.
        Increasing $\NbrSites$ implies a higher $\omega$ at equilibrium, a lower response of the $\omega$ to changes in $\Ne$ and a higher rate of relaxation.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {Relaxation-Stability-Gamma.pdf}
    \caption[Relaxation time of $\omega$ dependence on $\NbrSites$, while correction for $\DeltaDeltaG$]{
        $\omega$ Relaxation after a brutal change in $\Ne$, for various $\NbrSites$ while correcting for $\DeltaDeltaG$.
        The left and right panel correspond to low $\Ne$ ($1e^{5}$) and the middle panel corresponds to high $\Ne$ ($2e^{6}$).
        Solid line corresponds to the average over replicates ($r$) and the shaded area correspond to the $90\%$ interval among replicates.
        The mutation rate ($\mu$) is $1e{-8}$ per year per site, and the total time of the computation is $900$ million years.
        $\beta=1.686$, $\DeltaGmin=-10$ for all simulations. The number of sites is changed from $\NbrSites=15$ to $\NbrSites=158$, and the number of replicates is changed accordingly such that the total number of sites ($\NbrSites*r$) is kept constant.
        Moreover, $\DeltaDeltaG$ is changed according to $\NbrSites$ such that the product $\DeltaDeltaG\NbrSites$ is kept constant, thus the response of the $\omega$ to changes in $\Ne$ is kept constant.
        Increasing $\NbrSites$ implies a higher $\omega$ at equilibrium, and a higher rate of relaxation.
    }
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {Relaxation-Stability-Grantham-Alpha-Gamma.pdf}
    \caption[Relaxation time of $\omega$ for the Grantham model]{
        $\omega$ Relaxation after a brutal change in $\Ne$, under a Grantham model.
        The left and right panel correspond to low $\Ne$ ($1e^{5}$) and the middle panel corresponds to high $\Ne$ ($2e^{6}$).
        Solid line corresponds to the average over replicates ($r$) and the shaded area correspond to the $90\%$ interval among replicates.
        The mutation rate ($\mu$) is $1e{-8}$ per year per site, and the total time of the computation is $900$ million years.
        $\beta=1.686$, $\DeltaDeltaG=-10$ for all simulations. The number of sites is changed from $\NbrSites=15$ to $\NbrSites=158$, and the number of replicates is changed accordingly such that the total number of sites ($\NbrSites*r$) is kept constant.
        Moreover, $\DeltaDeltaG$ is changed according to $\NbrSites$ such that the product $\DeltaDeltaG\NbrSites$ is kept constant, thus the response of the $\omega$ to changes in $\Ne$ is kept constant.
        Finally, $\DeltaGmin$ is changed according to $\NbrSites$ and $\DeltaDeltaG$ such that the equilibrium value $x\eq$ is kept constant, by solving numerically equation~\ref{eq:equilibrium-pheno}.
        Increasing $\NbrSites$ implies a higher rate of relaxation.}
\end{figure}


\section{Distribution of fitness effects}
\label{sec:distribution-of-fitness-effects}

DNA mutations changing a genotype can result in a change of {phenotype}, and ultimately a change in fitness.
From a specific genotype, all the possible mutations thus result in a distribution of phenotypic effect (DPE) and fitness effects ({DFE}).
The DPE and {DFE} are not known a priori, but are the resulting consequence of the mutation-selection-drift balance.
Empirically, these distributions are of particular importance since they can be obtained experimentally or inferred with other data.
As an example, {DFE} can be inferred from polymorphism dataset~\citep{Eyre-walker2007, Galtier2016}.
Moreover, the distribution of $\EmpiricalDeltaDeltaG$ for novel mutations can be obtained experimentally.

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth] {DPE-DFE.pdf}
    \caption[Distribution of fitness effects and phenotypic effect]{
        Distribution of fitness effects and phenotypic effect for novels {non-synonymous} mutations observed along a simulation at the mutation-selection balance.
        $\DeltaGmin=-118$, $\DeltaDeltaG=1$, $\NbrSites=300$, $\beta=1.686$, and for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid.
        Each side of the distribution is fitted to a gamma distribution, shown in dotted line.
        Panel A. Distribution of observed $\EmpiricalDeltaDeltaG$, which fit adequately the gamma distribution for negative $\EmpiricalDeltaDeltaG$ (stabilizing mutations).
        Panel A. Distribution of observed selection coefficient, which fit adequately the gamma distribution for both positive and negative selection coefficient. However the shape parameter estimated is not the same for positive and negative selection coefficients.
    }
\end{figure}