\section{Introduction}

Molecular sequences differ across species due to the particular history of nucleotide substitutions along their respective lineages.
These substitutions in turn are the result of the interplay between evolutionary forces such as mutation and selection, whose relative forces are determined by the amount of random genetic drift.
These forces have effects at different levels: mutations are carried by molecular sequences, selection is mediated at the level of individuals, while random genetic drift is a population sampling effect.
Yet, they jointly contribute to the long-term molecular evolutionary process.
Thus, the challenge of molecular evolution is to tease out their respective contributions, based on comparative analyses.

One main aspect of this challenge is to correctly evaluate the role of random drift in the long term evolutionary process.
Population genetics theory implies that the strength of drift, due to the stochastic sampling of mutations, is less pronounced in lineage with large effective population size ($\Ne$), and as a consequence, the purification by selection of weakly deleterious mutations is more effective.
This fundamental idea is at the core of the nearly-neutral theory of evolution.
This theory posits that a substantial fraction of mutations are deleterious or weakly deleterious, and as result predicts that the substitution rate (relative to the neutral expectation), called $\omega$, decreases along lineages with higher $\Ne$~\citep{Ohta1972, Ohta1992}.

This prediction has been more quantitatively examined under the assumption that the selective effects of mutations are drawn from a fixed distribution of fitness effects (\acrshort{DFE})~\citep{Kimura1979, Welch2008}.
Assuming a gamma \acrshort{DFE}, a key result obtained in this context is an approximate allometric scaling of $\omega$ as a function of $\Ne$ (i.e. $\omega \sim \Ne^{-k}$), where $k$ is the shape parameter of the \acrshort{DFE}.
In practice, DFEs are strongly leptokurtic, which thus predicts a weak negative relation between $\omega$ and $\Ne$.

The quantity $\omega$ is an observable of the molecular evolutionary process, at least if we can separately estimate the rate of neutral and selected substitutions in a phylogenetic or comparative context.
Practically, in the case of protein-coding \acrshort{DNA} sequences, and assuming that synonymous mutations are neutral, while non-synonymous changes affecting the amino-acid sequence are under selection, $\omega$ can be identified with the ratio of non-synonymous over synonymous substitution rates , called $\dnds$.
Thus the nearly-neutral argument translates into a predicted decrease in $\dnds$ as a function of long-term changes in $\Ne$.

The context of protein-coding sequences fostered another modelling approach, based on genotype-fitness maps instead of distribution of fitness effects.
In this alternative approach, the selective effect of a mutation depends on the fitness of both the source and the target amino acids involved in the mutation event~\citep{Halpern1998, Rodrigue2010, Tamuri2012}.
More fundamentally, the fitness depends solely on the current genotype, not on the trajectory of mutations leading to this sequence.
For example, if the target amino acid has a higher fitness than the current amino acid, the selective effect of the mutation is positive, and reciprocally negative for a lower fitness.
Such a modelling approach predicts higher substitution rates between amino acid with similar biochemical properties.
Even though this modelling approach differs substantially from the one assuming a fixed \acrshort{DFE}, it also predicts a negative correlation between $\omega$ and $\Ne$, at least when the process is at equilibrium~\citep{Spielman2015, DosReis2015}.

Conversely, one striking theoretical result was the proof that $\omega$ is in fact predicted to be independent of $\Ne$ under relatively very general circumstances, namely, whenever (i) the fitness is a log-concave function of a phenotype and (ii) the phenotype itself is equimutable.
Equimutability states that the distribution of phenotypic changes due to mutations is independent the current phenotype of individuals~\citep{Cherry1998}.
This general theoretical argument has been invoked in the context of \textit{in silico} experiments of protein sequence evolution, assuming that proteins are under selection for their thermodynamic stability, with fitness being proportional to folding probability of the protein~\citep{Goldstein2013}.
Thermodynamic stability is itself computed using a 3D structural model of the protein.
These computational experiments have led to the observation that $\omega$ is essentially independent of $\Ne$.
An explanation proposed for this result is that the distribution of changes in free energy of folding ($\EmpiricalDeltaDeltaG$) due to mutations is approximately independent of the current free energy ($\DeltaG$), thus making the free energy of folding essentially equimutable.

However, the equimutability assumption is a relatively strong one, which also conflicts with combinatorial considerations about the relation between sequence and phenotype~\citep{Serohijos2012}.
For example, if a protein sequence is already maximally stable, only destabilizing (or neutral) mutations can occur.
More generally, assuming that the stability of a protein sequence reflects an underlying fraction of positions having already accepted destabilizing amino acids, then the probability of destabilizing mutational events is in turn expected to directly depend on the current stability of the protein.

Altogether, depending on the theoretical model mapping sequence to fitness, $\omega$ can be either independent or negatively correlated to $\Ne$, or even positively if considering adaptive evolution and environmental changes~\citep{Lanfear2014}.

Empirically, variation in $\omega$ between lineages has been inferred using phylogenetic codon models applied to empirical sequences~\citep{Yang1998,Zhang2004}.
Confronting branch-specific $\omega$ estimates to life-history traits such as body mass or generation time uncovered a positive correlation~\citep{Popadin2007, Nikolaev2007}.
Subsequently, integrative inference methods combining molecular sequences and life-history traits have also found that $\omega$ correlates positively with traits such as longevity and body mass~\citep{Lartillot2011, Figuet2017}.
Since lineage with a large body size and extended longevity typically correspond to species with low $\Ne$~\citep{Romiguier2014}, these empirical correlations suggest a negative correlation between $\omega$ and $\Ne$, thus confirming the theoretical prediction of the nearly-neutral theory of evolution.
However, the universality and robustness of the correlation between $\omega$ and life-history traits is still debated.
Results have not been entirely consistent across independent studies, the correlation was found to be either not statistically significant~\citep{Lartillot2012}, or even in the opposite direction depending on the specific clade under study or the potential biases taken into account~\citep{Lanfear2010, Nabholz2013, Lanfear2014, Figuet2016}.

If empirical evidence for a negative correlation of $\omega$ with $\Ne$ is still not totally convincing, another empirical correlation is known to be much more robust.
Indeed, expression level or protein abundance is one of the best predictors of $\omega$, with highly expressed proteins typically having lower $\omega$ values, a correlation clearly significant although relatively weak~\citep{Duret2000, Rocha2004, Drummond2005a, Zhang2015, Song2017}.
Theoretical models, also based on protein stability, have been invoked to explain this negative correlation between $\omega$ and expression level~\citep{Wilke2006, Drummond2008}.
According to this argument, selection against protein misfolding due to toxicity, which is stronger for more abundant proteins, induces abundant proteins to evolve toward greater stability, resulting in a more constrained and more slowly evolving protein coding sequence~\citep{Serohijos2012}.

The possibility that expression level and $\Ne$ might play similar roles in the evolution of proteins has already been noticed.
More precisely, under models of selection against protein misfolding, the free energy of folding $\DeltaG$ is predicted to vary similarly along a gradient of either $\Ne$ or expression level~\citep{Serohijos2013}.
As a corollary, under strict equimutability of $\DeltaG$, these computational models imply that $\omega$ should also be independent of expression level~\citep{Serohijos2012}, thus like what is predicted with regards to changes in $\Ne$.

Altogether, both theoretical results and empirical analyses are not yet conclusive about the question of how $\omega$ depends on $\Ne$ and expression level.
In particular, the theoretical response of $\omega$ to changes in both $\Ne$ and expression level has not been quantified and, most importantly, has not been related to the specific map between genotype, phenotype and fitness.
Such an analytical development would be useful to more decisively confront the theoretical predictions relating $\omega$ to both $\Ne$ and expression level to empirical data.
Ultimately, relating proteins structural parameters to the response of $\omega$ would help to bridge the gap between protein thermodynamics on one side and comparative genomics on the other side.

Lastly, the theoretical results discussed so far are valid only at the mutation-selection-drift balance.
In a non-equilibrium regime, however, and at least under a model assuming a site-independent genotype-fitness map, an increase in $\Ne$ first leads to an increase in $\omega$ caused by adaptive substitutions, and subsequently a decrease in $\omega$ due to stronger purifying selection in the long term~\citep{Jones2016}.
Studying only equilibrium properties can thus be misleading.
For that reason, the dynamic response of $\omega$ to changes in $\Ne$ must also be addressed, quantified, and its connection with the underlying selective landscape better characterized.
Dynamic properties of $\omega$ to changes in $\Ne$ are of theoretical interest but are also empirically relevant, such that, if overlooked they could thwart the relation between theoretical expectations and empirical estimates.

In this context, the aim of the present study is to characterize the dynamic and equilibrium response of $\omega$ to changes in $\Ne$ and expression level, and to relate this response to structural parameters of the model.
To this effect, we develop a general mathematical approach to derive a quantitative approximation of the response of $\omega$ to changes in $\Ne$ and expression level, in the context of a given genotype-phenotype-fitness map, as depicted in figure~\ref{fig:Summary}.
In the light of previously published empirical estimates from protein thermodynamic and comparative genomics, we discuss the articulation between empirical data and our mechanistic model.
We also discuss some of the alternative biophysical mechanisms that could determine the selective landscape on protein-coding sequences, and how they would modulate the response of $\omega$ to changes in $\Ne$ and expression level.


\section{Results}

\subsection{Models of evolution}

The results that are presented below are valid for a general category of models of sequence evolution, based on an additive trait $x$, such that the coding positions of the sequence contribute additively to the trait.
The trait is under directional selection specified by a decreasing and log-concave fitness function $ \wrightfit (x)$.
As a specific example, we more specifically consider a model of protein evolution under the constraint of thermodynamic stability, as depicted in the left panel of figure~\ref{fig:Summary}.
This model is inspired from previous work~\citep{Williams2006, Goldstein2011, Pollock2012}, except that we make several simplifying assumptions, allowing us to derive analytical equations.

\begin{figure}[htbp]
    \centering
    \includegraphics[width=\textwidth, page=1] {summary.pdf}
    \caption[Outline of the theoretical results]{
    Outline of the theoretical results.
    The genotype to fitness relationship is depicted in the left panel.
    The genotype is defined as the complete state of the amino-acid sequence.
    Subsequently, the phenotype ($x$) is a real-valued summary of the genotype, and is defined in our model as the fraction of destabilizing sites in the sequence.
    Finally, fitness is a decreasing log-concave function of the phenotype, depending on structural parameters of the model.
    Once defined the relation from genotype to fitness at the individual level, the evolutionary process is presented in the middle panel.
    The process consists of point substitutions of the amino-acid representative sequence of the population.
    Substitutions are mutation reaching fixation, thus the rate of substitution depends mechanistically on the interplay between mutation, selection and genetic drift.
    For a given effective population size $\Ne$, the evolutionary process result in an average value of the phenotype $\smash{x\eq}$ and also an average substitution rate (relative to the neutral rate) $\omega$.
    Averaging over time is equivalent to determining the statistical equilibrium, by ergodicity of the stochastic process, given we can derive the probabilities of transitions between phenotypes.
    The scaling of this equilibrium $\omega$ under a range of $\Ne$, presented in the right panel, gives us the slope of this relationship, called the susceptibility $\chi$, as a function of the structural parameters defined in the phenotype-fitness map.
    }
    \label{fig:Summary}
\end{figure}
In the original biophysical model, protein stability is determined by the difference in free energy between the folded and unfolded conformations, called $\DeltaG$ and measured in kcal/mol.
Technically, free energy is computed based on the 3D conformation of the protein and using statistical potentials.
As a result, the stabilizing or destabilizing effect of an amino acid at a particular site depends on amino acids present in the vicinity in 3D conformation, thus implementing what has been called specific epistasis~\citep{Dasmeh2018}.

Here, we approximate this model such that the (de-)stabilizing effect at a particular site, such as measured by the $\EmpiricalDeltaDeltaG$ of the mutation, does not depend on other neighbouring residues, thus disregarding specific epistasis~\citep{Dasmeh2014}.
Instead, each site contributes independently and additively to $\DeltaG$.
In addition, we assume that, for each site of the sequence, only one amino acid is stabilizing the protein.
All $19$ other amino acids are equally destabilizing.
Each site bearing a destabilizing amino acid contributes an excess of $\DeltaDeltaG > 0$ (in kcal/mol) to the total $\DeltaG$.
To note, $\DeltaDeltaG$ and $\EmpiricalDeltaDeltaG$ are equivalent, however $\DeltaDeltaG$ is preferred for mathematical readability, and $\EmpiricalDeltaDeltaG$ will refer to the empirical estimate.
The smallest achievable value of $\DeltaG$, obtained when all amino acids of the sequence are stabilizing, is noted $ \DeltaGmin < 0$.
Similarly, $\EmpiricalDeltaGmin$ will refer to the empirical estimate of $\DeltaGmin$.
In this model, the most succinct phenotype of a given genotype (i.e. sequence) is just the proportion of destabilizing amino acids in the sequence, defined as $0 \leq x \leq 1$.
Thus $\DeltaG$ is a linear function of $x$:
\begin{align}
    \DeltaG (x) = \DeltaGmin + \NbrSites \DeltaDeltaG x,
\end{align}
where $\NbrSites$ is the number of sites in the sequence.

For a given $\DeltaG$, thermodynamic equations allows one to derive the proportion of protein molecules that are in the native (folded) conformation in the cytoplasm.
This fraction is assumed to be a proxy for fitness, motivated in part by the fact that a protein must be folded to perform its function.
A slightly different model will be considered below, in order to take into account protein expression level (see section~\ref{sec:expression}).

Analytically, the fitness function is given by the Fermi Dirac distribution and is typically close to $1$, leading to a first-order approximation~\citep{Goldstein2011}:
\begin{gather}
    \wrightfit (x) = \dfrac{1}{1 + e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x)}},\\
    \Rightarrow \wrightfit (x) \simeq 1 - e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x)},\\
    \Rightarrow \logfit (x) = \ln ( \wrightfit (x) ) \simeq e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x)},
\end{gather}
where $\wrightfit$ is the Wrightian fitness for a given phenotype and $\logfit $ is the Malthusian fitness (or log-fitness).
Here, $\DeltaGmin$ and $\DeltaDeltaG$ are defined as above, and the parameter $\beta$ is $1.686$ mol/kcal at $25 \degree C$ (or $298.2 \degree K$).

Of note, even though the phenotypic effect of a mutation at a given site does not depend on the amino-acids that are present at other sites (i.e. the trait is additive), the fitness effect of a mutation still depends on other sites (i.e. the log-fitness is not additive).
As a result, the molecular evolutionary process is site-interdependent, a property referred to as non-specific epistasis~\citep{Dasmeh2018}.

\subsection{Response of \texorpdfstring{$\omega$}{ω} to changes in \texorpdfstring{$\Ne$}{Nₑ}. Analytical approximation}

For a given effective population size $\Ne$, the evolutionary process reaches an equilibrium (figure~\ref{fig:Summary}, middle panel).
This substitution rate at this equilibrium, normalized by the substitution rate of neutral of mutations to discard the influence of the underlying mutation rate, is denoted $\omega$.
This relative rate can also be interpreted as the mean fixation probability of mutations scaled by the fixation probability of neutral alleles $p = 1 / {2 \Ne}$, the mean being weighted by the probability of occurrence of mutations in the population.
As a result, an $\omega < 1$ indicates that mutations are negatively selected on average, and $\omega$ decreases with increasing strength of purifying selection.

In this section we present an analytical approximate solution for the response of $\omega$ after a change in $\Ne$ (in log space), as depicted in the right panel of figure~\ref{fig:Summary}.
We call this response the susceptibility of $\omega$ to changes in $\Ne$, and denote it as $\chi$:
\begin{align}
    \chi = \frac{ \der \omega}{\der \ln (\Ne)} \label{eq:chi}
\end{align}
Deriving $\chi$ is done in two steps.
First, we determine the mean phenotype at equilibrium, when evolutionary forces of mutation, selection and genetic drift compensate each other.
Subsequently, differential calculus is used to compute the response of the equilibrium phenotype to a change in $\Ne$, which allows us to ultimately derive an equation for $\chi$.
The main results of our derivation are given both in the general case of any (log-concave) phenotype-fitness map, and in the specific case of the biophysical model introduced above.
A more detailed derivation is available in the supplementary materials (section~\ref{sec:susceptibility-after-a-change-in-Ne}).

For a given genotype, mutations can have various effects: they can increase or decrease the proportion of destabilizing amino acids, or do nothing if the mutation is between two destabilizing amino acids.
To derive the probabilities of such events to occur, we also make the simplifying assumption that all transitions between amino acids are equiprobable.
Altogether, any mutation in the sequence can then have a phenotypic effect of $0$ or $\dx=1 /\NbrSites$, with probabilities of transitions equal to:
\begin{gather}
    \begin{cases}
        \dx &\text{ with probability } 1-x, \\
        0 &\text{ with probability } \frac{18 x }{19}, \\
        -\dx &\text{ with probability } \frac{x}{19}.\\
    \end{cases} \label{eq:proba}
\end{gather}
In the extreme case of an optimal phenotype ($x = 0$), only destabilizing mutations are proposed.
Moreover, the probability to propose a stabilizing mutation (effect $-\dx$), or a neutral mutation (effect $0$), is proportional to $x$.
Conversely, the probability to propose a destabilizing mutation is equal to $(1-x)$.
As a result, the mutation bias is proportional to $(1-x)/x$.
This mutation bias fundamentally reflects a combinatorial effect, due to the number of mutational opportunities available in either direction.

Second, we need to determine the strength of selection acting on mutations.
Destabilizing mutations are selected against with a negative selection coefficient which can be approximated by:
\begin{align}
    s & \simeq \frac{1}{\NbrSites}\frac{ \partial \logfit (x) }{\partial x} \label{eq:s} \\
    \Rightarrow s & \simeq - \beta \DeltaDeltaG e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x)}, \label{eq:s-unfolded}
\end{align}
where $ \logfit = \ln (\wrightfit)$ is the log-fitness (or malthusian fitness).
Conversely, stabilizing mutations will be under positive selection with opposite sign but same absolute value.
It is important to realize that the selective effect is dependent on $x$.
Furthermore, because the fitness function is log-concave, the absolute value of $s$ increases with $x$.

Based on these expressions for the mutational and selective pressures, one can then study the trajectory followed by the evolutionary process.
Starting from an optimal sequence, mostly destabilizing mutations will occur, some of which may reach fixation and accumulate until selection coefficients against new deleterious mutations is too strong, at which point the protein will reach a point of equilibrium called marginal stability~\citep{Taverna2002, Bloom2007}.
Most importantly, the probability of fixation of mutations is affected by genetic drift, and thus depends on the effective population size ($\Ne$).
At the equilibrium between mutation, selection and drift, the process fluctuates through the occurrence of advantageous and deleterious substitutions compensating each other.
This equilibrium can be determined by expressing the constraint that the selection coefficient of substitutions is expected to be null on average~\citep{Goldstein2013}.
Formally, and after simplification, the equilibrium phenotype denoted $x\eq$ is given in the general case by:
\begin{align}
    \ln \left( \frac{1 - x\eq}{x\eq} \right) + \ln (19) & \simeq - \frac{4\Ne}{\NbrSites} \frac{ \partial \logfit (x\eq) }{\partial {x\eq}} \\
    \Rightarrow \ln \left( \frac{1 - x\eq}{x\eq} \right) + \ln (19) & \simeq 4\Ne \beta \DeltaDeltaG e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x\eq)} \text{,} \label{eq:equilibrium}
\end{align}
in the more specific case of the biophysical model.
This equation essentially expresses the mutation-selection equilibrium: the left-hand side of the equation is the log of the mutation bias at $x$, while the right-hand side is simply $4 \Ne s$, the scaled selection coefficient.

This equation cannot be solved explicitly for $x\eq$, but a qualitative intuition on the consequences of change in $\Ne$ to the equilibrium phenotype $x\eq$ is given in figure~\ref{fig:NeChangeInfluence}.
Intuitively, an increase in $\Ne$ results in a more optimal phenotype, closer to $0$.
The mutation bias (left-hand side of equation~\ref{eq:equilibrium}) decreases with $x$ while the strength of selection (right-hand side of equation~\ref{eq:equilibrium}) increases with $x$, and the equilibrium phenotype is obtained at their intersection.
An increase in $\Ne$ leads to shifting the selective response upward, which then results in a leftward shift of the equilibrium phenotype (i.e. closer to $0$).
The leftward shift is smaller for selective strengths characterized by a steeper curve, resulting in qualitatively weaker susceptibility of the equilibrium phenotype to changes in $\Ne$

\begin{figure}[htbp]
    \centering
    \includegraphics[width=\textwidth, page=1] {theoretical.pdf}

    \caption[Phenotype susceptibility after a change in $\Ne$]{
    Phenotype susceptibility after a change in $\Ne$.
    The equilibrium phenotype $x\eq$ is determined by the equation $\smash{\ln(\frac{1 - x\eq}{x\eq}) + \ln(19)=4\Ne \beta \DeltaDeltaG e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x\eq)}}$.
    The right-hand side of the equation increases exponentially with $x$ where $\NbrSites \DeltaDeltaG$ is the exponential growth rate ({\color{RED}red} and {\color{GREEN}green} curves).
    The parameter $\DeltaDeltaG$ is chosen such that $\DeltaDeltaG$ is increased by an order of magnitude in the red compared to the green curve, moreover, the value of $\DeltaGmin$ is chosen such that the equilibrium value $\smash{x\eq}$ is kept constant by solving numerically equation~\ref{eq:equilibrium}.
    Whenever $\NbrSites \DeltaDeltaG$ is large ({\color{RED}red solid} curve), the exponential growth is steep and increasing $\Ne$ (dotted curve) only changes subtly the equilibrium phenotype by a factor $\smash{\color{RED}\Delta x\eq}$.
    On the other hand, whenever $\NbrSites \DeltaDeltaG$ is low ({\color{GREEN}green} solid curve), the exponential growth is steadier and increasing $\Ne$ (dotted curve) involve a stronger susceptibility of the equilibrium phenotype by a factor $\smash{\color{GREEN}\Delta x\eq}$.
    Moreover, changes in $\smash{x\eq}$ reflects the changes in $\omega$ since both are approximately equal (equation~\ref{eq:dnds}).
    }
    \label{fig:NeChangeInfluence}
\end{figure}

The results obtained thus far only relate the equilibrium phenotype ($x\eq$) to $\Ne$.
To capture how $\omega$ varies with $\Ne$, we also need to obtain an expression for $\omega$ as a function of $x\eq$.
At equilibrium we can derive (section~\ref{subsec:mean-scaled-fixation-probability-omega-at-equilibrium} in supplementary materials) the expected substitution rate of mutations, and thus $\omega$, which simply approximates to:
\begin{gather}
    \omega \simeq x\eq \label{eq:dnds}
\end{gather}
This simple approximation is due to the fact that the substitutions between two destabilizing amino acids (which are neutral) compose the largest proportion of proposed mutations having a substantial probability of fixation (equation~\ref{eq:proba}).
In contrast, stabilizing mutations are rare, while destabilizing mutations have a low probability of fixation.
Since there is a fraction $x\eq$ of sites already occupied by a destabilizing amino-acid, these neutral substitutions occur at rate $x\eq$.

Combined together, these analytical approximations yield the susceptibility (equation~\ref{eq:chi}) of $\omega$ to a change in $\Ne$:
\begin{gather}
    \chi = \frac{ \der \omega}{\der \ln (\Ne)} \simeq - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{\NbrSites}{4 \Ne} \frac{\partial \ln [ (1 - x\eq) / x\eq ]}{\partial x\eq } + \frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}}.
\end{gather}
The two terms of the denominator correspond to the derivative of respectively the mutational bias and the scaled selection coefficient, respectively.
However, the mutational bias decreases weakly with $x$ (blue curve on figure~\ref{fig:NeChangeInfluence}) while the strength of selection increases sharply with $x$ (red and green curves).
As a consequence, the derivative of the mutational bias is much lower than the derivative of the selection coefficient around the equilibrium point (i.e. the phenotype is \textit{nearly} equimutable).
The second term can therefore be ignored, which leads to a very compact equation for susceptibility $\chi$ in the general case:
\begin{gather}
    \chi \simeq - \frac{\frac{ \partial \logfit (x\eq) }{\partial {x\eq}}}{\frac{ \partial^2 \logfit (x\eq) }{\partial {x\eq}^2}}
\end{gather}
The susceptibility is thus equal to the inverse of the relative curvature, i.e. the ratio of the second to the first derivatives, of the log-fitness function, taken at the equilibrium phenotype.
Of note, this susceptibility is strictly negative for decreasing log-concave fitness functions, asserting that $\omega$ is a decreasing function of $\Ne$.
In addition, the susceptibility itself is low in absolute value (i.e. $\omega$ responds more weakly) for strongly concave log-fitness functions.
This equation quantitatively captures the intuition developed in figure~\ref{fig:NeChangeInfluence}, namely that the response of $\omega$ is very weak if the selection curve is very steep around the equilibrium set point (red curve compared to green curve).

In the specific case of the biophysical model, the susceptibility ($\chi$) further simplifies to:
\begin{gather}
    \chi \simeq -\dfrac{1}{\beta \NbrSites \DeltaDeltaG}, \label{eq:chi_application}
\end{gather}
meaning that $\omega$ is linearly decreasing with $\Ne$ (in log scale) since $\chi$ is independent of $x\eq$, or, in other words, that the exact value of the equilibrium phenotype has no impact on the slope.
Moreover, only the compound parameter $\beta \DeltaDeltaG \NbrSites$ has an impact on the slope of the linear relationship.
Thus, in particular, the slope of the linear relationship between $\omega$ and $\Ne$ is affected by $\DeltaDeltaG$ but not by $\DeltaGmin$.
Of note, empirically, only relative values of $\Ne$ (up to a multiplicative constant) are required to obtain an estimate of $\chi$.
% Although the general equation seems to suggest in appearance that elasticity only depends on the fitness function, in fact, what the biophysical model shows is that the elasticity can actually be modulated in two ways: by playing on the genotype / phenotype relationship (here, $\DeltaDeltaG$), or by playing on the fitness phenotype relationship (here, beta). All the more reason to consider beta as a potentially free parameter too.

\subsection{Response of \texorpdfstring{$\omega$}{ω} to changes in protein expression level}
\label{sec:expression}

Effective population size is not the sole predictor of $\omega$, and expression level (or protein abundance) is also negatively correlated to $\omega$.
However, our previous model, which assumes that fitness is proportional to the folded fraction, and is thus independent of protein abundance, does not express the fact that selection is typically stronger for proteins characterized by higher levels of expression.
An alternative biophysical model is to assume that each misfolded protein molecule has the same relative effect on fitness, caused by its toxicity for the cell~\citep{Drummond2005a, Wilke2006, Drummond2008, Serohijos2012}.

Ou general derivation can be directly applied to this case.
For a given protein with expression level $y$ and a cost $c$ representing the selective cost per misfolded molecule (positive constant), the fitness and selection coefficient can be defined as follows:
\begin{gather}
    \logfit (x) \simeq - c y e^{\beta (\DeltaGmin + \NbrSites \DeltaDeltaG x)} \\
    \Rightarrow s \simeq - \beta \DeltaDeltaG c y e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x)}.
\end{gather}
Under this model, the total selective cost of a destabilizing mutation is now directly proportional to the total amount of misfolded proteins.
This fitness function leads to the following expression for the mutation-selection-drift equilibrium:
\begin{equation}
    \ln \left( \frac{1 - x\eq}{x\eq} \right) + \ln (19) = 4\Ne y A \beta \DeltaDeltaG \NbrSites e^{\beta(\DeltaGmin + \DeltaDeltaG \NbrSites x\eq)}.
\end{equation}
Importantly, in this equation, $\Ne$ and $y$ are confounded factors appearing only as a product.
This means that increasing either $\Ne$ or $y$ leads to same change in equilibrium phenotype, and hence the same change in $\omega$.
In other words, the susceptibility of the response to changes in either $\Ne$ or expression level is the same:
\begin{align}
    \chi = \frac{ \der \omega\eq}{\der \ln (\Ne)} = \frac{ \der \omega\eq}{\der \ln (y)} \simeq - \frac{1}{\beta \NbrSites \DeltaDeltaG}. \label{eq:chi_application_y}
\end{align}

A similar result can be obtained under other models relating phenotype to fitness, for example if the selective cost is due to translational errors (section~\ref{subsec:translational-errors} in supplementary materials).
Alternatively if the protein is assumed to be regulated such as to reach a specific level of functional protein abundance under a general cost-benefit argument~\citep{Cherry2010,Gout2010}, a multiplicative factor depending solely on the expression level is prefixed (section~\ref{subsec:selective-cost-proportional-to-amount-of-misfolded-protein} in supplementary materials).
Altogether, we theoretically obtain the same linear decrease of $\omega$ with regards to either effective population size or expression level (in log space) under a broad variety of hypotheses.

\subsection{Simulation experiments}

Our theoretical derivation of the susceptibility of $\omega$ to changes in $\Ne$ (and expression level) is based on several simplifying assumptions about the evolutionary model and makes multiple approximations.
In order to test the robustness of our main result, we therefore conducted systematic simulation experiments, relaxing several of these assumptions.
In each case, simulations were conducted under a broad range of values of $\Ne$, monitoring the average $\omega$ observed at equilibrium and plotting the scaling of these measured equilibrium $\omega$ as a function of $\Ne$.

Specifically, with respect to mutations, our derivation assumes that all amino-acid transitions are equiprobable, or in other words, the complexity of the genetic code is not taken into account.
Simulating evolution of \acrshort{DNA} sequence and invoking a matrix of mutation rates between nucleotide allows us to test the robustness of our results to this assumption.
Furthermore, with regard to the phenotypic effects of amino-acid changes, in our derivation, we assumed that all destabilizing amino acids have an identical impact on protein stability.
In reality, one would expect conservative amino-acid replacements to be less destabilizing than radical changes.
This assumption is relaxed in our simulation, such that destabilizing mutations in each position are now proportional to the Grantham distance~\citep{Grantham1974} between the optimal amino acid in this position and the amino acid proposed by the non-synonymous mutation.
Finally, our derivation assumes that the number of sites in the sequence ($\NbrSites$) is large, such that the selection coefficient is well approximated by the fitness derivative (equation~\ref{eq:s}).
The robustness of this approximation was tested by conducting simulations with finite sequences of realistic length ($\NbrSites=300$ coding positions).
% Finally, the substitution rate is known to vary among sites within a protein, and each site does not contribute equally to the stability of the protein~\citep{Echave2016}.
% To mimic this effect, the destabilizing effect of mutations is scaled by a factor specific to each site, drawn from gamma distribution (see supplementary materials).

These simulation experiments demonstrate, first, that the relation between $\omega$ and log-$\Ne$ is indeed linear, at least in the range explored here, and that the slope of the linear regression matches the expected theoretical value (figure~\ref{fig:GoldsteinVsToy}A).
Secondly, we observe that the parameter $\DeltaGmin$ has virtually no effect on the slope of the linear regression, as also expected theoretically (figure~\ref{fig:GoldsteinVsToy}B).
Instead, decreasing $\DeltaGmin$ (to more negative values) merely results in an overall increase in $\omega$ over the whole range of $\Ne$ (i.e. has an impact on the intercept, not on the slope of the relation).
This is due to the fact that decreasing $\DeltaGmin$ shifts the equilibrium to higher $x\eq$, since more destabilizing sites can then reach fixation before reaching the point of marginal stability.

Finally, we relaxed our assumption that each site of the sequence contributes independently to $\DeltaG$, by taking into account the 3D structure of protein and using a statistical potential to estimate $\DeltaG$ (section~\ref{sec:simulation-using-the-3d-structure-of-protein} in supplementary materials).
We implemented the original model considered in \citet{Williams2006}, \citet{Goldstein2011} and \citet{Pollock2012}, in which the free energy is computed based on the 3D conformation using pairwise contact potential energies between neighbouring amino-acid residues~\citep{Miyazawa1985}.
The original works showed that under this model, $\omega$ is approximately independent of $\Ne$~\citep{Goldstein2013}.
Using extensive simulations in order to obtain sufficient resolution, we observe that $\omega$ is in fact weakly dependent on $\Ne$, being again approximately linear with log-$\Ne$ (figure~\ref{fig:GoldsteinVsToy}C).
Moreover, the observed slope ($\widehat{\chi}=-0.00117$) matches the slope obtained under the model of additive $\DeltaG$ ($\widehat{\chi}=-0.00125$, figure~\ref{fig:GoldsteinVsToy}D), considering $\EmpiricalDeltaDeltaG = 1.0$ kcal/mol for destabilizing mutations and $\NbrSites=300$.
In this experiment (figure~\ref{fig:GoldsteinVsToy}D), $\DeltaGmin$ was set to $-118$ kcal/mol, which is the $\DeltaG$ of the optimal (maximally stable) sequence of $300$ sites~\citep{Goldstein2011}.
\begin{figure}[htbp]
    \centering
    \includegraphics[width=\textwidth] {elasticity-low.pdf}

    \caption[$\omega$ susceptibility to change in $\Ne$]
    {$\omega$ susceptibility to change in $\Ne$.
    Scaling experiment simulating sequence evolution and recording the average $\omega$ in the vertical axis observed at equilibrium as a function of $\Ne$ in the horizontal axis.
    Along the x-axis, $200$ replicate simulations are performed for each different $\Ne$, the average (solid lines) and $90\%$ confidence interval (shaded area) of $\omega$ are shown.
    Linear regression (dashed lines) allows to estimate the susceptibility ($\hat{\chi}$), which is compared to theoretical expectation.
    The fixed parameters are $\NbrSites=300$, $\beta=1.686$.
    Simulations are conducted under our additive phenotype model (expect panel C), and for each non-optimal amino acid, $\DeltaDeltaG$ is scaled by the Grantham distance to the optimal amino acid.
    Panel A; $\DeltaGmin$ and $\DeltaDeltaG$ are given in the legend.
    $\DeltaDeltaG$ is increased and $\DeltaGmin$ is changed accordingly such that the equilibrium value $\smash{x\eq}$ is kept constant, by solving numerically equation~\ref{eq:equilibrium}.
    The estimated susceptibility ($\hat{\chi}$) decreases proportionally to the inverse of $\DeltaDeltaG$, as predicted by our theoretical model.
    Panel B; $\DeltaDeltaG=1$ and $\DeltaGmin$ are given in the legend.
    Decreasing $\DeltaGmin$ (to more negative values) increases $\omega$ but changes hardly the slope of the linear regression, as expected theoretically.
    Panel C; Folding free energy that determines the stability of the folded native state is computed using 3D structural conformations and pairwise contact potential energies between neighbouring amino-acid residues.
    The estimated susceptibility $\hat{\chi}$ is $-0.00124$, and as a result $\omega$ at equilibrium is weakly dependent on log-$\Ne$.
    Panel D; Simulation under the approximation of an additive free energy model, with empirical estimate of $\DeltaGmin = -118$ kcal/mol and $\DeltaDeltaG = 1$ kcal/mol.
    Our theoretical prediction of the susceptibility is $\chi = 0.00198$, close to the observation using 3D structural conformations.
    Under such parametrization, the scaling behaviour is similar to panel C, although with less variance.
    \label{fig:GoldsteinVsToy}
    }
\end{figure}

\subsection{Time to relaxation}

Although the equilibrium value of $\omega$ after changes in $\Ne$ is an important feature of the $\omega$-$\Ne$ relationship, another characteristic that is scarcely studied is the dynamic aspect~\citep{Jones2016}, particularly the relaxation time to reach the new equilibrium $\omega$.
We observed in our simulations that the determining factor of the relaxation time is the number of sites $\NbrSites$ (figure~\ref{fig:relaxStability}A), such that the return to equilibrium is faster for longer sequences.
This observation matches the theoretical prediction that more mutational opportunities are available for longer sequences, driving the trait close to equilibrium at a faster rate.

It may be useful to compare the relaxation pattern observed here with the predictions under two alternative models of sequence evolution, representing two extreme scenarios.
On one hand, having fitness modelled at the level of sites, such as contemplated by many phylogenetic mutation-selection models~\citep{Halpern1998, Rodrigue2010, Tamuri2012}, leads to a situation where every site has to adapt on its own to the new change in $\Ne$.
The relaxation time is then very long, on the order of the inverse of the per-site substitution rate.
On the other hand, assuming a fixed distribution of fitness effect (\acrshort{DFE}) as in \citet{Welch2008}, the response of $\omega$ is instantaneous (figure~\ref{fig:relaxStability}B).
Our model is effectively in between these two extreme scenarios.

Another characteristic observed in these non-equilibrium experiments is the discontinuity of $\omega$ after a change in $\Ne$.
Most importantly, both an increase and decrease in $\Ne$ lead to a discontinuity (figure~\ref{fig:relaxStability}A \&~\ref{fig:relaxStability}B).
These non-equilibrium behaviors can both be explained mechanistically.
Under low $\Ne$, the phenotype is far away from the optimal phenotype because the efficacy of selection is weaker.
A sudden increase in $\Ne$ results first in a short traction toward a more optimal phenotype, which results in a suddenly higher $\omega$, caused by a transient adaptation of the protein toward a higher stability.
Conversely, under high $\Ne$ the phenotype is closer to optimal and the purification of deleterious mutations is stronger.
The reaction to a decrease in $\Ne$ is a relaxation of the purification and thus an $\omega$ closer to the neutral case, which results into higher $\omega$ until reaching the point of marginal stability.
To note, an increase in $\Ne$ can theoretically and possibly lead to an $\omega$ that is temporarily greater than 1 due to adaptive evolution~\citep{Jones2016}, while a decrease in $\Ne$ always imply an $\omega < 1$, as it gives at most a neutral regime of relaxed selection.
\begin{figure}[htbp]
    \centering
    \includegraphics[width=\textwidth] {Relaxation.pdf}
    \caption[ $\omega$ relaxation after a change in $\Ne$]{
    $\omega$ relaxation after a change in $\Ne$.
    Solid line corresponds to the average over $1000$ replicates and the shaded area corresponds to the $90\%$ interval among replicates.
    The mutation rate ($\mu$) is $1e{-8}$ per year per site, and the total time of the computation is $700$ million years.
    Panel A.
    $\beta=1.686$ for all simulations.
    The \acrshort{DNA} sequence of $500$ sites is divided into exons of equal size.
    However the number of sites per exon changes between simulations from $\NbrSites=5$ to $\NbrSites=500$.
    Moreover, $\DeltaDeltaG$ is changed according to the exon size such that the product is kept constant, and as a consequence the susceptibility of $\omega$ to changes in $\Ne$ is kept constant.
    Finally, $\DeltaGmin$ is changed accordingly such that the equilibrium value $\smash{x\eq}$ is kept constant, by solving numerically equation~\ref{eq:equilibrium}.
    As such, whatever the exon size, $\smash{x\eq}$ and $\chi$ are kept constant and thus the observed effect is due to the number of sites in the exon.
    We observe that increasing the number of sites implies a reduced time to reach the new equilibrium.
    Panel B.
    In context of a time-independent fitness landscape, where each amino acid has different fitness (site-specific profile), the time taken to reach the new equilibrium value of $\omega$ after a change in $\Ne$ is long, such that relaxation rate is on the order of the mutation rate.
    In the context of a fixed distribution of fitness effects (\acrshort{DFE}), the relaxation time is non-existent and the new equilibrium value of $\omega$ is reached instantaneously.
    }
    \label{fig:relaxStability}
\end{figure}


\section{Discussion}
% Goal and theoretical results

We provide a compact analytical result for the equilibrium response (which, by analogy with thermodynamics, we call the susceptibility) of $\omega$ to changes in $\Ne$, and we relate this response to the parameterization of the genotype-phenotype-fitness map.
An application to a model of selection against protein misfolding shows that the susceptibility of $\omega$ to variation in $\Ne$ (in log space) is linear with a negative slope.
Furthermore, this application demonstrates that effective population size and protein expression level are interchangeable in respect to their impact on the response of $\omega$.
Our compact theoretical results, which were obtained by making several simplifying assumptions, are supported by more complex simulations of protein evolution relaxing these assumptions.
In particular, our theoretical predictions are verified under a numerical model of protein evolution in which the free energy is computed based on the 3D structure.

Overall, the susceptibility ($\chi$) is a function of structural parameters of the protein and takes a very simple analytical form, being is inversely proportional to the product of three terms;
namely the sequence size, the inverse temperature ($\beta$), and the average change in conformational energy of destabilizing mutations ($\EmpiricalDeltaDeltaG$).
Quantitatively, this product can be several orders of magnitude greater than $1$ in practice, such that the susceptibility of $\omega$, which is its inverse, is typically small.
Previous studies using this model presented an apparent lack of response of $\omega$ to changes in $\Ne$~\citep{Goldstein2013}.
We refine this result, by observing that there is in fact a very subtle and weak relation, which requires extensive computation to be detected, but which is well predicted by our theoretical derivation.
Based on empirical estimates of the structural parameters $\beta = 1.686$, $\NbrSites=300$ sites and $\EmpiricalDeltaDeltaG = 1.0$ kcal/mol~\citep{Zeldovich2007}, the estimated susceptibility is $\hat{\chi} \simeq -0.002$.
In other words, for a relative increase in $\Ne$ or expression level of $6$ orders of magnitude, a factor approximately equal $0.01$ is subtracted to $\omega$, a subtle relationship that requires laborious effort to be detected in simulated data.

\subsection{Adequacy to empirical data}
Empirically, variation in $\omega$ along the branches of phylogenetic trees has been inferred and correlated to proxies of $\Ne$, such as body size or other life-history traits.
These analyses showed mitigated support for a negative relation between $\omega$ and $\Ne$~\citep{Lanfear2014}.
More recently, phylogenetic integrative methods refined the estimate of covariation between $\omega$ and $\Ne$ along lineages by leveraging polymorphism data~\citep{Brevet2019}.
This approach gives an estimate of $\hat{\chi} \simeq 0.02$ in primates (section~\ref{sec:empirical-estimation-of-susceptibility} in supplementary materials) at least one order of magnitude greater than the quantitative estimate obtained above from the biophysical model.
More empirical data across different clades would be required to robustly consolidate such empirical estimates, but as of yet, these are challenging the idea of a very weak response.

% Stress the importance that susceptibility is valid for Ne and protein abondance
The relation between $\omega$ and expression level provides an independent, and potentially more robust, source of empirical observation.
Our theoretical results suggest that, under relatively general conditions, the response of $\omega$ to expression level should be of the same magnitude than the response to $\Ne$.
Empirically, protein expression level is one of the best predictors of $\omega$ and the empirical estimation of $\chi$ in fungi, archaea and bacteria varies in the range $[-0.046;-0.021]$ (table~\ref{tab:table-susceptibility} in supplementary materials) extracted from \citet{Zhang2015}.
Estimation in animals and plants gives somewhat lower estimates, in the range of $[-0.026;-0.004]$, although still higher (in absolute value) than $-0.002$.

Additionally, another empirical observation is the negative relation between the mean destabilizing effect of mutations (mean $\EmpiricalDeltaDeltaG$) and the $\DeltaG$ of the protein.
Such a relation is empirically observed in \citet{Serohijos2012}, with the slope of the linear regression ($\widehat{\rho}_{\EmpiricalDeltaDeltaG, \DeltaG}$) equal to $ = -0.13$.
The slope of the linear correlation observed in our simulations is weaker, with a slope of $\widehat{\rho}_{\EmpiricalDeltaDeltaG, \DeltaG} = -0.01$ under the 3D biophysical model, and $\widehat{\rho}_{\EmpiricalDeltaDeltaG, \DeltaG} = -0.003$ under the model of additive phenotype parameterized by $\DeltaDeltaG = 1$ and $\NbrSites=300$ (figure \label{fig:DG-vs-DDG} in supplementary materials).
This observation also sheds light on the correlation between $\omega$ and $\Ne$ in empirical data and in our model.
Indeed, equimutability, or namely that the distribution of $\EmpiricalDeltaDeltaG$ is independent of $\DeltaG$ is a necessary condition to observe independence between $\omega$ and $\Ne$~\citep{Cherry1998}.
In our model, $\EmpiricalDeltaDeltaG$ depends on $\DeltaG$ due to combinatorial considerations, but this dependence is weaker than empirically observed, which also translates into a weaker susceptibility of $\omega$ to changes in $\Ne$ or expression level than empirically observed.

Thus, overall, the response of $\omega$ to either $\Ne$ or expression level predicted by the biophysical model considered above seems lower than what is empirically observed.
There are several possible explanations for this discrepancy.
First, the biophysical model might be valid, but the numerical estimates used for $\NbrSites$ or $\EmpiricalDeltaDeltaG$ could be inadequate.
A $\EmpiricalDeltaDeltaG$ of $1.0$ kcal/mol seems to correspond to empirical estimates~\citep{Zeldovich2007}.
On the other hand, the effective number of positions implicated in the trait might be smaller than the total number of residues in the protein.
In our model, all positions in the protein can in principle compensate for the destabilizing effect of a mutation at a particular position.
In practice, the number of sites susceptible to compensate each other is probably smaller, resulting in a stronger departure from equimutability.

Alternatively, the biophysical model considered here might be too restrictive.
Recent empirical studies have provided evidence against the hypothesis that the rate of sequence evolution is driven solely by the toxicity effect of unfolded proteins~\citep{Plata2017,Razban2019,Biesiadecka2020}.
Notably, the response of $\omega$ to changes in expression level has also theoretically been found to arise as a consequence of protein-protein interactions, where protein may either be in free form or engaged in non-specific interactions~\citep{Yang2012, Zhang2013}.
In non-specific interactions at the protein surface, stabilizing amino acids are hydrophilic and destabilizing amino acids are hydrophobic, sticking to hydrophobic residues at the surface of other proteins~\citep{Dixit2013,Manhart2015}.

Our theoretical results can be applied more broadly to protein-protein interactions using a mean-field argument (section~\ref{sec:model-of-protein-protein-interactions} in supplementary materials).
Fitting this model with empirical structural estimates~\citep{Janin1995a, Zhang2008}, we obtain a susceptibility of $\chi \simeq -0.2$ thus a much stronger response than under the model based on conformational stability.
This much stronger response is due to fewer sites in the protein being involved in protein-protein interaction than for conformational stability, in addition to a lower free energy engaged in contact between residues.

Altogether, fitness based on protein stability is a compelling model of molecular evolution, but may not be a sufficiently comprehensive model to explain the amplitude of variation of $\omega$ empirically observed along a gradient of either effective population size or protein expression level.
The net response of $\omega$ to changes in $\Ne$ or expression level could have several biophysical causes, which in the end would imply a weak but still empirically measurable response.

\subsection{Physically grounded modelling}
% Analogy to physics
This study describes the signature imprinted on \acrshort{DNA} sequences by an evolutionary process by merging equations from population genetics and from structural physicochemical first principles.
Hence, physicochemical vocabulary and theory are recruited to describe the molecular details.
More widely, our modelling is imprinted with vocabulary and analogy to physical systems, with wording in the likes of forces, susceptibility and dynamic.
We argue that the analogy between evolutionary and physical systems is not restricted to wording, but allows one to visualize and intuit the causality chain and which approximation can be made in order to derive analytically tractable results~\citep{Sella2005, Mustonen2009, Bastolla2012, Bastolla2017}.
The robustness of results can be assessed by computational implementations and simulations.
Computational models offer a means to test the validity and robustness, while mathematical models offer an intuitive mechanistic mental analogy.

% Thermodynamic analogy
% Analogy to Clapeyron formula ? TdP/dT = d²H/d²V, where T is analogous to Ne
A first analogy between evolutionary biology and physics comes from statistical mechanics, describing how observable macroscopic properties are emerging from microscopic states.
As an example, in the Ising model, the magnetization is a macroscopic outcome of all microscopic atomic spins that can be in one of two states~\citep{Brush1967}.
At equilibrium when the system reaches minimum potential energy, the observable magnetization is controlled by variables such as temperature, and by external forces such as the magnetic field forcing the system.
By analogy, our atoms are sites of the sequence which can be in stabilizing or destabilizing state, and the macroscopic emerging observable is the relative substitution rate of selected mutation ($\omega$), which is controlled by effective population size ($\Ne$) analogous to temperature~\citep{Sella2005}.
Ultimately, the susceptibility of macroscopic observable ($\omega$) to a control variable ($\Ne$) can be tied to the microscopic parameters.
Another example of observable macroscopic properties, though not considered in this manuscript, could be for example site entropy, i.e. the effective number of observed amino acids per site at equilibrium~\citep{Goldstein2016, Jimenez2018, Jiang2018}.

% Mechanical analogy
% Elasticity with springs ? Surface tension with molecule and radius of the
% Elasticity ? Susceptibility (dimensionless) ? Compliance (units of metres per newton) ? Flexibility (units of metres per newton) ?
A second analogy between evolutionary biology physics comes from mechanics.
The phenotypic space is analogous to a spatial coordinate of the system, which is determined by external evolutionary forces; namely mutation, selection and drift.
These evolutionary forces are analogous to springs attached and stretching the phenotype in various directions and with various stiffness, resulting in an equilibrium point.
Once the equilibrium point is determined, an elastic (or reversible) deformation of the phenotype can be studied whenever applying a differential stress to the system.
In this analogy, changing the free energy of the reference phenotype ($\EmpiricalDeltaGmin$) leads in first approximation to changes of reference and equilibrium points but not a change in the stiffness of the springs, meaning the elastic coefficient (analog to susceptibility) is approximately conserved.
Conversely, stiffness ot the springs increases with regards to $\EmpiricalDeltaDeltaG$, meaning a that change in the stress applied to the system ($\Ne$) result in a weak deformation at the new equilibrium.

% Fundation to integrate other forces
Altogether, these analogies could be further formalized and used as a framework in which to integrate other external evolutionary forces.
Additional molecular processes could be included, for instance mutational bias, or GC-biased gene conversion inducing a bias in nucleic composition of genomes.
The change in equilibrium composition due to GC bias, and the magnitude of the response, could be investigated, such as to describe their signatures observable on molecular sequence, factoring the effects of mutation, selection and drift.
Finally, we hope that the results presented in this work, and more broadly the framework outlined here, could foster a better understanding of observable signatures of the long-term evolutionary process emerging from ecological parameters and molecular physicochemical first principles.

\subsection{Building block of inference models}

% Categories of models
At the crossroad between empirical data and theoretical models, inference models can compute the likelihood of an observation given a model, and also extract interpretable signature from molecular dataset.
Models of inference are classified broadly into phenomenological and mechanistic~\citep{Rodrigue2010a}.
Mechanistic models dissect the causation chains and construct a model from first principles, while phenomenological models aims to capture statistical aggregates (such as mean substitution rates) from the data.

% Phenomenological models
Example of such phenomenological models are the models parameterized directly in $\omega = \dnds$.
They will for example tell us which branch of the tree, or which site of the sequence has a decrease or increase in $\omega$ but not the underlying reason for such changes, which can be either mutation, selection, drift or another evolutionary force.
For example, in such models, a mutation and its reverse mutation do not have the opposite selection coefficient.
As such they cannot disentangle the potentially confounding effect of mutation, selection and drift.
Neither can they predict the nature of the functional relationship between population genetics or structural physicochemical parameters to $\omega$.

% Site specific mechanistic models of evolution
In contrast, mechanistic models relate structural, population-genetics and ecological parameters to the likelihood function.
As such, mechanistic inference models are suitable to construct an integrative framework relating the signal available in molecular sequences to structural parameters ($\EmpiricalDeltaDeltaG$), expression level across genes and effective population size across lineages.
Once such models are fitted to the data, the estimated parameters can be confronted to their independent empirical estimate ($\EmpiricalDeltaDeltaG, \Ne, \hdots$), which allows one to robustly test the model since orthogonal estimation of biological and ecological parameters should be congruent~\citep{Dasmeh2014}.
Additionally, once model robustness has been assessed, this inference framework could allow one to estimate unknown variables, for example ancestral effective population sizes, with enough signal and calibration of the other parameters.
Altogether, we argue that our theoretical results are a building block to construct an integrated inference framework of molecular evolution, since such models are bridging intrinsic biological parameters and the signal extractable from molecular sequences.


\section{Materials \& Methods}
Protein sequence evolution is simulated under an origin-fixation model~\citep{McCandlish2014}, where one sequence represents the whole population.
From the resident \acrshort{DNA} sequence $\Seqi$, we define $\setSeqNeighbors$ as the set of all possible mutant that are one nucleotide away from $\Seqi$, and were mutant sequences containing a stop codon are excluded.
For a protein of $\NbrSites$ amino-acid sites, $\left| \setSeqNeighbors \right| \leq 9 \NbrSites$, since each codon has a maximum of $9$ possible mutant codons that are one mutation away and that are not stop codon.
For each mutant sequence $\Seqj \in \setSeqNeighbors$, we compute its fitness and subsequently the selection coefficient of the mutant:
\begin{align}
    s \left( \Seqi,\Seqj\right) & = \dfrac{ \wrightfit \left(\Seqj \right) - \wrightfit \left(\Seqi\right) }{ \wrightfit \left( \Seqi\right)}, \\
    \Rightarrow s \left( \Seqi,\Seqj\right) & \simeq  \logfit \left(\Seqj \right) -   \logfit \left(\Seqj \right),
\end{align}
where $\wrightfit$ is the Wrightian fitness for a given phenotype and $\logfit $ is the Malthusian fitness (or log-fitness).

The next event of mutant invading the population, and the time to reach such event is chosen using Gillespie's algorithm~\citep{Gillespie1977}, according to the rates of substitution between sequences:
\begin{equation}
    \submatrix_{\Seqitoj} = \mu_{\Seqitoj} \dfrac{4 \Ne s \left( \Seqi,\Seqj\right)}{1 - \e^{-4 \Ne s \left( \Seqi,\Seqj\right)}},
\end{equation}
where $\mu_{\Seqitoj}$ is the mutation rate between $\Seqi$ and $\Seqj$, determined by the underlying $4x4$ nucleotide mutation rate matrix, and ${\submatrix_{\Seqitoj}} = \mu_{\Seqitoj}$ in the case of synonymous substitutions.
Various optimizations are implemented to reduce the computation time of mutant fitness.
The simulation starts with a burn-in period to reach mutation-selection-drift equilibrium.

\subsection{Models of fitness function}
\label{MatMet:folding}

Under a simulation of protein folding with an additive model of free energy, the protein's difference in free energy between folded and unfolded state is given by:
\begin{equation*}
    \DeltaG\left(\Seqi\right) = \DeltaGmin + \NbrSites \DeltaDeltaG * x\left(\Seqi\right),
\end{equation*}
where $0 \leq x\left(\Seqi\right) \leq 1$ is the distance of $\Seqi$ to the optimal sequence.
For each site of sequence, the optimal amino acids are chosen randomly at initialization, and the distance between the current amino acid and the optimal is scaled by the Grantham amino-acid distance~\citep{Grantham1974}.
Wrightian fitness is defined as the probability of our protein to be in the folded state, given by the Fermi-Dirac distribution:
\begin{equation}
    \wrightfit (\Seqi) = \dfrac{e^{-\beta \DeltaG\left(\Seqi\right) }}{1 + e^{-\beta \DeltaG\left(\Seqi\right) }} = \dfrac{1}{1 + e^{\beta \DeltaG\left(\Seqi\right) }},
\end{equation}
where $\beta$ is the inverse of the temperature ($\beta=1/kT$).

Under a simulation with site-specific fitness profiles, the protein log-fitness is computed as the sum of amino-acid log-fitnesses along the sequence.
In this model, each codon site $\i$ has its own fitness profile, denoted $\Profile^{(i)} = \{ \Profile^{(i)}_{\aaSource}, \forall 1 \leq \aaSource \leq 20 \}$, a vector of 20 amino-acid (Wrightian) fitnesses.
Since $\Seqi[ \i ]$ is the codon at site $\i$, the encoded amino acid is $\aaMap \left( \Seqi[ \i ] \right)$, hence the fitness at site $\i$ is $\Profile^{(i)}_{\aaMap \left( \Seqi[ \i ] \right)}$.
Altogether, the selection coefficient of the mutant $\Seqj$ is:
\begin{equation}
    s \left( \Seqi,\Seqj\right) = \sum\limits_{1 \leq \i \leq \Nsite} \ln \left( \dfrac{\Profile^{(i)}_{\aaMap \left( \Seqj[ \i ] \right)}}{\Profile^{(i)}_{\aaMap \left( \Seqi[ \i ] \right)}} \right),
\end{equation}
where $\Profile_{\i}$ used in this study are extracted from \citet{Bloom2017}, empirically determined in the context of deep mutational scanning.

Under simulation with a fixed distribution of fitness effects (\acrshort{DFE}), the selection coefficient of the mutant $\Seqj$ is gamma distributed (shape $k > 0$):
\begin{equation}
    - s \left( \Seqi,\Seqj\right) \sim \text{Gamma} \left( \bar{|s|}, k \right)
\end{equation}

\subsection{\texorpdfstring{$\omega$}{ω} along the simulation}
From the set of mutants $\setSeqNeighbors$ that is one nucleotide away from $\Seqi$, we define the subsets $\setSeqNonSynNeighbors$ and $\setSeqSynNeighbors$ that are respectively the set of non-synonymous and synonymous mutants, where $\setSeqNonSynNeighbors \cup \setSeqSynNeighbors = \setSeqNeighbors$.
As in previous works~\citep{Spielman2015, DosReis2015, Jones2016}, the ratio of non-synonymous over synonymous substitution rates of the sequence is defined as :
\begin{align}
    \omega(t) &= \dfrac{\sum\limits_{\Seqj \in \setSeqNonSynNeighbors} \submatrix_{\Seqitoj}}{\sum\limits_{\Seqj \in \setSeqNonSynNeighbors} \mu_{\Seqitoj}} \left( \dfrac{\sum\limits_{\Seqj \in \setSeqSynNeighbors} \submatrix_{\Seqitoj}}{\sum\limits_{\Seqj \in \setSeqSynNeighbors} \mu_{\Seqitoj}} \right)^{-1}\\
    &= \dfrac{\sum\limits_{\Seqj \in \setSeqNonSynNeighbors} \mu_{\Seqitoj} \dfrac{4 \Ne s \left( \Seqi,\Seqj\right)}{{1 - \e^{-4 \Ne \left( \Seqi,\Seqj\right)} }}}{\sum\limits_{\Seqj \in \setSeqNonSynNeighbors} \mu_{\Seqitoj}}
\end{align}
Averaged over all branches of the tree, the average $\omega$ is :
\begin{align}
    \omega & = \left\langle \omega(t) \right\rangle, \\
    & = \int_{t} \omega (t) \der t,
\end{align}
where the integral is taken over all branches of the tree, while the integrand $\omega (t)$ is a piece-wise function changing after every point substitution event.
