\documentclass{article}

\usepackage{xcolor}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\definecolor{LIGHTBLUELINK}{HTML}{3366BB}
\definecolor{PURPLELINK}{HTML}{663366}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false,pagebackref=true]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
linkcolor=DARKBLUELINK,
anchorcolor=DARKBLUELINK,
citecolor=DARKBLUELINK,
filecolor=DARKBLUELINK,
menucolor=DARKBLUELINK,
urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage{biorxiv} % github.com/alexsbaldwin/biorxiv-inspired-latex-style

\usepackage{url}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{textcomp}
\usepackage{gensymb}
\usepackage{cancel}
\usepackage{lmodern}
\usepackage{xfrac, nicefrac}
\usepackage{blkarray}
\usepackage{pgf,tikz}
\usetikzlibrary{positioning,arrows,automata,calc}
\usepackage{bm}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{tabu}
\tabulinesep=0.6mm
\newcommand\cellwidth{\TX@col@width}
\usepackage{hhline}
\setlength{\arrayrulewidth}{1.2pt}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\AtBeginEnvironment{tabu}{\footnotesize}
\usepackage{booktabs}

\usepackage[flushleft] {threeparttable}
\usepackage{adjustbox}
\usepackage{bbold}
\usepackage{pdfpages}
\usetikzlibrary{positioning}
\usetikzlibrary{arrows,automata}
\pdfinclusioncopyfonts=1
\usepackage{nicefrac}       % compact symbols for 1/2, etc.
\usepackage{microtype}      % microtypography
\usepackage{lineno}

\graphicspath{{artworks/}}
\makeatletter
\def\input@path{{artworks/}}
\makeatother

\renewcommand{\baselinestretch}{1.5}

\include{notations}

\linenumbers

\title{Inferring long-term effective population size with Mutation-Selection models}

%\date{September 9, 1985}	% Here you can change the date presented in the paper title
%\date{} 					% Or removing it

\author{
    \large
    T. {Latrille}$^{1,2}$, V. {Lanore}$^{1}$, N. {Lartillot}$^{1}$ \\
    \normalsize
	$^{1}$Université de Lyon, Université Lyon 1, CNRS, Laboratoire de Biométrie et Biologie Évolutive UMR 5558, F-69622 Villeurbanne, France.\\
	$^{2}$École Normale Supérieure de Lyon, Université de Lyon, Université Lyon 1, Lyon, France\\
	\texttt{\href{mailto:thibault.latrille@ens-lyon.org}{thibault.latrille@ens-lyon.org}} \\
}

% Uncomment to override  the `A preprint' in the header
%\renewcommand{\headeright}{}

\begin{document}
\maketitle

\begin{abstract}
    Mutation-selection phylogenetic codon models are grounded on population genetics first principles and represent a principled approach for investigating the intricate interplay between mutation, selection and drift.
    In their current form, mutation-selection codon models are entirely characterized by the collection of site-specific amino-acid fitness profiles.
    However, thus far, they have relied on the assumption of a constant genetic drift, translating into a unique effective population size ($\Ne$) across the phylogeny, clearly an unreasonable hypothesis.
    This assumption can be alleviated by introducing variation in $\Ne$ between lineages.
    In addition to $\Ne$, the mutation rate ($\mu$) is susceptible to vary between lineages, and both should co-vary with life-history traits (LHTs).
    This suggests that the model should more globally account for the joint evolutionary process followed by all of these lineage-specific variables ($\Ne$, $\mu$, and LHTs).
    In this direction, we introduce an extended mutation-selection model jointly reconstructing in a Bayesian Monte Carlo framework the fitness landscape across sites and long-term trends in $\Ne$, $\mu$ and LHTs along the phylogeny, from an alignment of DNA coding sequences and a matrix of observed LHTs in extant species.
    The model was tested against simulated data and applied to empirical data in mammals, isopods and primates.
    The reconstructed history of $\Ne$ in these groups appears to correlate with LHTs or ecological variables in a way that suggests that the reconstruction is reasonable, at least in its global trends.
    On the other hand, the range of variation in Ne inferred across species is surprisingly narrow.
    This last point suggests that some of the assumptions of the model, in particular concerning the assumed absence of epistatic interactions between sites, are potentially problematic.

\end{abstract}

% keywords can be removed
\keywords{Phylogenetic \and codon models \and mutation-selection models \and population genetic \and population size \and mutation rate \and life history traits.}

\section{Introduction}
\label{sec:Introduction}

Since the realization, by \citet{Zuckerkandl1965} that genetic sequences are informative about the evolutionary history of the species, molecular phylogenetics has developed into a mature and very active field.
A broad array of models and inference methods have been developed, using {DNA} sequences for reconstructing the phylogenetic relationships among species~\citep{Felsenstein1981}, for estimating divergence times~\citep{Thorne2002}, or for reconstructing the genetic sequences of remote ancestors~\citep{Liberles2007}.
However, genetic sequences might contain information about other aspects of the evolutionary history and, in particular, about past population-genetic regimes.

Interspecific divergence is the long-term outcome of population-genetic processes, in which point mutations at the level of individuals are then subjected to selection and {genetic drift}, leading to substitutions at the level of the population.
As a result, the {substitution} patterns that can be reconstructed along phylogenies are modulated by the underlying population-genetic parameters (mutation biases, selective landscapes, effective population size), suggesting the possibility to infer the past variation of these parameters over the phylogeny.
Independently, ecological properties such as phenotypic characters or life-history traits can be observed in extinct or in present-day species.
Using the comparative method~\citep{Felsenstein1985}, these traits can be reconstructed for the unobserved ancestral species.
Combined together, genetic and phenotypic ancestral reconstructions can then be used to unravel the interplay between evolutionary and ecological mechanisms.

Practically, in order to disentangle mutation, selection and {genetic drift}, we need to classify individual substitutions into different categories, differing in the strength of mutation, selection or {genetic drift}.
In protein-coding {DNA} sequences, the mutational process occurs at the nucleotide level.
Assuming that {synonymous} mutations are selectively {neutral} and that selection mostly acts at the protein level, {synonymous} substitutions can be used to infer the patterns of mutation, without any interference contributed by selection.
Then, by comparing the {non-synonymous} {substitution} rate relative to the {synonymous} {substitution} rate (the ratio $\dnds$), one can estimate the global strength of selection acting on proteins.
This idea was formalized using phylogenetic {codon} models~\citep{Muse1994,Goldman1994}.
This led to a broad range of applications, either to detect proteins under adaptive selection~\citep{Kosiol2008}, or to measure the modulations of the strength of purifying selection between sites~\citep{Echave2016}, genes~\citep{Zhang2015}, or lineages~\citep{Lartillot2011}.

Concerning variation in $\dnds$ between lineages, and in a context mostly characterized by purifying selection, the {nearly-neutral} theory predicts that changes in the global strength of selection (measured as $\dnds$) is related to changes in the relative strength of {genetic drift}, which is in turn mediated by changes in {effective population size} ($\Ne$)~\citep{Ohta1992}.
Mechanistically, populations with high $\Ne$ are characterized by more efficient purifying selection against mildly deleterious mutations, resulting in lower $\dnds$~\citep{Kimura1979, Welch2008}.

Codon models have been used to empirically measure such changes in the efficacy of purifying selection along phylogenies, either by allowing for different $\dnds$ values in different parts of the tree~\citep{Dutheil2012}, or by estimating $\dnds$ independently for every branch of the tree~\citep{Popadin2007}.
Alternatively, $\dnds$ can be modelled as a continuous trait, varying along the phylogeny as a stochastic process, splitting at each node of the tree into independent processes~\citep{Seo2004}.
Once empirical estimates of the variation in $\dnds$ between lineages or groups has been obtained, these can be compared to changes in $\Ne$ across lineages, so as to test the validity of the predictions of the {nearly-neutral} theory.
Independent empirical estimation of $\Ne$ is usually done vie proxies, such as the {neutral} diversity within species~\citep{Galtier2016}, or life-history traits.
For instance, animal species characterized by a large body size or an extended longevity are typically expected to also have a low $\Ne$~\citep{Romiguier2014}.
Alternatively, a Bayesian integrative framework has been proposed~\citep{Lartillot2011}, extending the approach of \citet{Seo2004}, in which the joint variation in $\ds$, $\dnds$ and in life-history traits or other proxies of $\Ne$ is modelled as a multivariate Brownian process, with a variance-covariance matrix capturing the signal of their correlated evolution.

Analyses using these approaches and these proxies of $\Ne$ have suggested a negative correlation between $\dnds$ and $\Ne$~\citep{Popadin2007, Lanfear2010, Lartillot2011, Lartillot2012, Romiguier2014, Figuet2017}, thus confirming the theoretical prediction of the {nearly-neutral} theory.
However, the universality and robustness of the correlation between $\dnds$ and $\Ne$ is still debated~\citep{Nabholz2013,Lanfear2014,Figuet2016, Bolivar2019}, and further investigation might be required.
Moreover, these analyses do not explicitly formalize the quantitative relationship between $\Ne$ and $\dnds$.
This relation is in principle dependent on the underlying fitness landscape~\citep{Welch2008, Cherry1998, Goldstein2011}, and can show complicated behavior due to non-equilibrium properties~\citep{Jones2016}.
These questions could be addressed in the context of a mechanistic modelling approach.

As an alternative to classical $\dnds$-based {codon} models, mechanistic {codon} models explicitly introduce population genetic equations into the {codon} {substitution} process~\citep{Halpern1998}.
Specifically, these so-called mutation-selection {codon} models explicitly assign a fitness parameter to each amino acid.
As a result, the {substitution} rate between each pair of codons can be predicted, as the product of the mutation rate and the fixation probability of the new {codon}, which is in turn dependent on the fitness of the initial and the final codons.
Since the strength of selection is typically not homogeneous along the protein sequence, and depends on the local physicochemical requirements~\citep{Echave2016, Goldstein2016,Goldstein2017}, local changes in selective strength are usually taken into account by allowing for site-specific amino-acid fitness profiles.
Site-specific amino-acid preferences are typically estimated either by penalized maximum likelihood~\citep{Tamuri2012,Tamuri2014}, or in a Bayesian context, using an infinite mixture based on a {Dirichlet process} prior~\citep{Rodrigue2010,Rodrigue2014}.
This second approach is further considered below.

Although not directly expressed in terms of this variable, the mutation-selection formalism induces an equilibrium $\dnds$, which is theoretically lower than $1$, thus explicitly modelling purifying selection~\citep{Spielman2015, DosReis2015}.
As a result, the mutation-selection codon framework proved to be a valuable null (nearly-neutral) model, against which to compare the observed $\dnds$ by classical {codon} models, so as to test for the presence of adaptation~\citep{Rodrigue2016, Bloom2017}.

However, these mutation-selection methods have so far assumed the strength of {genetic drift}, or equivalently $\Ne$, to be constant across the phylogeny.
This assumption is clearly not realistic, as attested by the empirically measured variation in $\dnds$ between lineages using classical {codon} models or, more directly, by the broad range of {synonymous} {neutral} diversity observed across species~\citep{Galtier2016}.
The impact of this assumption on the estimation of the fitness landscape across sites~\citep{Tamuri2014, Rodrigue2014}, or on the tests for the presence of adaptation~\citep{Rodrigue2016, Bloom2017} is totally unknown.
Relaxing this assumption of a constant $\Ne$ is thus necessary.

Conversely, since the mutation-selection formalism explicitly incorporates $\Ne$ as a parameter of the model, extending the model so as to let $\Ne$ vary across lineages is relatively straightforward, at least conceptually.
Doing this would then provide an occasion to address several important questions: do we have enough signal in empirical sequence alignments, to estimate the evolutionary history of $\Ne$ along a phylogeny?
Can we more generally revisit the question of the empirical correlations between $\Ne$ and ecological life-history traits (longevity, maturity, weight, size, $\hdots$), previously explored using classical $\dnds$ based models, but now in the context of this mechanistic framework?

\begin{figure*}[t]
    \begin{center}
        \includegraphics[width=\textwidth] {model_summary.pdf}
    \end{center}
    \caption[Model summary]{
        Model summary.
        Panel A.
        Our method requires a (given) rooted tree topology, an alignment of protein-coding DNA and (optionally) quantitative life-history trait for the extant species.
        Panel B.
        Relying on a codon model based on the mutation-selection formalism, assuming an auto-correlated log-Brownian process for the variation through time in effective population size ($\Ne$), mutation rate ($\mu$) and life-history traits, our Bayesian inference method estimates amino-acid fitness profiles across sites, variation in mutation rate and effective population size along the tree, as well as the node ages and the nucleotide mutation rates.}
    \label{fig:modelSummary}
\end{figure*}

\section{New approaches}
\label{sec:NewApproaches}
To address these questions, here we introduce a variant of the mutation-selection {codon} model, in which selection is modulated along the sequence (using site-specific amino-acid profiles), while the mutation rate ($\mu$), the {effective population size} ($\Ne$) and life-history traits are allowed to vary along the phylogeny (figure~\ref{fig:modelSummary}).
Methodologically, our model is fundamentally an integration between the Bayesian non-parametric version of the \citet{Halpern1998} mutation-selection model~\citep{Rodrigue2014}, and the molecular comparative framework modelling the joint evolution of life-history and molecular traits~\citep{Lartillot2011}.

Formally, the {substitution} rate (per unit of time) from {codon} $\ci$ to $\cj$, denoted $\submatrix_{\itoj}$, is equal to the total rate of mutation (per unit of time) at the level of the population~($2\Ne\mu_{\itoj}$) multiplied by the probability of fixation of the mutation $\pfix(\itoj)$:
\begin{equation}
{\submatrix_{\itoj}}
    = 2 \Ne \mu_{\itoj} \pfix(\itoj)
\end{equation}
In the case of {synonymous} mutations, which we assumed are {neutral}, the probability of fixation is independent of the original and target {codon}, and equals $1/2 \Ne$, such that ${\submatrix_{\itoj}}$ simplifies to:
\begin{equation}
    \submatrix_{\itoj} = \mu_{\itoj}
\end{equation}
In the case of {non-synonymous} mutations, the probability of fixation depends on the difference in fitness between the amino acid encoded by the initial and final codons:
\begin{equation}
    \label{eq:mutationSelection}
    {\submatrix_{\itoj}} = \mu_{\itoj} \dfrac{4 \Ne \left({\fitj - \fiti}\right)}{{1 - \e^{4 \Ne \left({\fiti - \fitj}\right)} }}
\end{equation}
where $\Fit$ is a $20$-dimensional vector specifying the log-fitness for each amino acid, and $\aai$ is the amino acid encoded by {codon} $i$.

In the model introduced here, $\Ne$ and $\mu$ are allowed to vary between species (across branches) as a multivariate log-Brownian process, but are assumed constant along the {DNA} sequence.
Conversely, amino-acid fitness profiles $\Fit$ are considered constant along the tree but are assumed to vary across sites, being modelled as independent and identically distributed random-effects from an unknown distribution estimated using a {Dirichlet process} {prior}.

This model was implemented in a {Markov chain Monte Carlo} framework, allowing for joint inference of site-specific selection profiles and reconstruction of life-history traits and population-genetic regimes along the phylogeny.
After validating our model and our inference framework against simulated data, we apply it to several cases of interest across metazoans (placental mammals, primates and isopods), for which some proxies of $\Ne$ are available.


\section{Results}
\label{sec:Results}

\begin{figure*}[t]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize A\hspace{0.35cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize B\hspace{0.35cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize C\hspace{0.35cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize D\hspace{0.35cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize E\hspace{0.35cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \llap{\raisebox{-1.0cm}{\scriptsize F\hspace{0.35cm}}}\hfill
    \caption[Inferred and simulated branch length and $\Ne$]{
        A-C: branch lengths in expected number of substitutions per site.
        D-F: $\Ne$ values across nodes (including the leaves) relative to $\Ne$ at the root.
        From left to right: simulation under the mutation-selection approximation (A,D), under a Wright-Fisher model accounting for small population size effects (5000 individuals at the root), site linkage and short term fluctuation of $\Ne$ (B,E) and accounting for site epistasis in the context of selection for protein stability.
        The tree root is $150$ million years old, where the initial population start with a mutation rate of $\smash{1e^{-8}}$ per site per generation, and generation time of $10$ years.
        These experiments confirm that signal in the placental mammalian tree can allow to reliably infer the direction of change in $\Ne$, even if linkage disequilibrium, short term fluctuation of $\Ne$ and finite population size effects are not accounted for in the inference framework.
        However, the presence of epistasis between sites is a serious threat to the inference of $\Ne$.
    }
    \label{fig:simulations}
\end{figure*}

\subsection{Validation using simulations}
\label{sec:ResultsSimulated}
The inference framework was first tested on independently simulated multiple sequence alignments (see methods).
With the aim of applying the inference method to empirical datasets, the simulation parameters were chosen so as to match an empirically relevant empirical regime.
Thus, the tree topology and the branch lengths were chosen based on a tree estimated on the mammalian dataset further considered below.
The other aspects of the simulation model (fitness landscape, variation in $\Ne$) were then varied along a gradient of increasing complexity, so as to test the inference framework under increasingly challenging conditions.

A first series of simulations was meant to test the soundness of our inference framework, by simulating essentially under the model used for inference, although with an independently developed software.
Thus, the mutation-selection approximation was assumed to be valid, and sites were simulated under different fitness profiles empirically determined~\citep{Bloom2017}, and finally, $\Ne$ was assumed to undergo discrete shifts at the tree nodes but otherwise to remain constant along each branch.
In this context, branch lengths and branch-specific values of $\Ne$ were accurately estimated by our inference method (figure~\ref{fig:simulations}, panel A \& D).
Concerning $\Ne$, the slope of the linear regression between true and estimated branch-specific $\Ne$ is $0.794$ ($r^2=0.915$)

However, the assumptions made for this first round of simulations are almost certainly violated in practice.
First, $\Ne$ is expected to undergo continuous changes along the lineages of the phylogeny.
Second, the diffusion approximation for the probability of fixation (equation~\ref{eq:mutationSelection}) may not hold in small finite populations.
Third, assuming a separate {substitution} process for each site is equivalent to assuming no linkage between sites (free recombination).
In practice, however, there is limited {recombination}, at least within exons, and this could induce deviations from the mutation-selection approximation, due to Hill-Robertson effects.

The finite population was now modelled explicitly, using a Wright-Fisher simulator, tracking the frequency of each {allele} at the gene level and at each generation along the phylogeny.
No {recombination} was implemented within genes.
These more complex simulation settings account for small population size effects, for hitchhiking of weakly deleterious mutations during selective sweep and for background selection due to linkage disequilibrium.
In addition, the {effective population size} $\Ne$ and the mutation rate were allowed to fluctuate continuously along the branches of the tree (changing by a small amount after each generation of the underlying Wright-Fisher process).
Finally, short-term fluctuations of $\Ne$, of the order of 20\% per generation, were accounted for by adding a random noise to the Brownian process describing the long-term evolution of $\Ne$.
In spite of these deviations between the simulation and the inference models, branch lengths and branch-specific {effective population sizes} could again be robustly recovered by the inference framework (slope of $0.868$, $r^2=0.919$, figure~\ref{fig:simulations}, panel B \& E).

These results are encouraging.
However, they still rely on the assumption of a site-independent fitness landscape, which is equivalent to assuming no epistasis.
Yet this assumption is almost certainly violated in practice~\citep{Pollock2014,Shah2015}.
Accordingly, we implemented a more complex, site-dependent fitness landscape accounting for the selective interactions between sites induced by the 3-dimensional structure of protein.
In this model, the conformational stability of the protein determines its probability of being in the folded state, which is in turn taken as a proxy for fitness~\citep{Williams2006, Goldstein2011, Pollock2012}.
Under this evolutionary model, and at any gven time, the fitness landscape at a particular {codon} site is dependent on the amino acids that are currently present at those sites that are in the vicinity of the focal site in 3D space (see supplementary).
When applied to data simulated using this model, our inference framework could accurately recover the simulated branch lengths (figure~\ref{fig:simulations}, panel D).
On the other hand, the distribution of $\Ne$ across the tree could not be accurately recovered (slope of $0.0196$, $r^2 = 0.0122$, figure~\ref{fig:simulations}, panel F).
In fact, no meaningful variation in $\Ne$ is detected, and the little variation in $\Ne$ that is inferred shows no correlation with the true branch-specific mean $\Ne$ values.
This effect can be explained by the predicted independence of $\dnds$, and more generally of the scaled selection coefficients associated with {non-synonymous} mutations, to changes in $\Ne$ in this specific model of protein stability, as shown theoretically by \citet{Goldstein2013}.

As an alternative model of epistasis between sites, a Fisher geometric model was also considered for the simulations (see supplementary).
The results under this model are intermediate between simulations without epistasis and simulations under the biophysically-inspired model considered above.
More specifically, under data simulated using Fisher’s geometric model, the true and estimated branch-specific $\Ne$ are strongly correlated with each other ($r^2 = 0.73$).
On the other hand, the slope of the correlation is substantially less than 1 ($0.571$).
In other words, the trends in $\Ne$ across the tree are correctly recovered, but the range of the variation in {effective population size} over the tree is substantially under-estimated.
As for the branch lengths, they are again correctly estimated.
In summary, our simulation experiments show that our inference framework is reliable in the absence of model mis-specification and is robust to violations concerning short-versus long-term variation in $\Ne$ or to the presence of empirically reasonable levels of Hill-Robertson interference.
On the other hand, and very importantly, epistasis, which is ignored by the inference model, appears to lead to a general underestimation of the true variation in Ne, to an extent that depends on the exact epistatic model but can go as far as completely obliterating any signal about the true variation in $\Ne$ across the tree in the most extreme situations.

\subsection{Empirical experiments}
\label{sec:ResultsEmpirical}
\begin{figure*}[t]
    \centering
    \includegraphics[width=\linewidth, page=1]{mammals_mirrors_low.pdf}
    \caption[Example of inferred $\Ne$ and $\mu$ on placental mammals dataset]{
        Inferred phylogenetic history of $\Ne$ (left) and $\mu$  (right) across placental mammals.
        Inference was conducted on a randomly chosen set of 18 out of 226 highly conserved CDS (< 1\% of gaps).
        Only highly conserved CDS were retained such that the assumption of constant fitness landscape is not incautiously broken by protein with changing function and/or adaptive selection.
        $\Ne$ values are relative to the root, which is arbitrarily set to one.
        Mean values of {MCMC} (after burn-in) are obtained at each node of the tree, hence a gradient can be extrapolated along each branch.
        $\mu$ spanned almost $2$ order of magnitude, and if we assume the root to be $105$My old~\citep{Kumar2017}, the rescaled mutation rate per site per year in extant species is between $\smash{1.1e^{-10}}$ and $\smash{7.8e^{-9}}$.
        $\Ne$ at the root of the tree is arbitrarily set to $1$, and all values are relative to the root, which spans at most an order of magnitude.
    }
    \label{fig:mammals_popsize_and_mutrate}
\end{figure*}

We next applied our inference framework to a series of 4 empirical datasets spanning different taxnonomic groups within metazoans.
As a first empirical case, we considered a dataset of 77 placental mammals, for which complete genome sequences and information about life-history traits is available.
Placental mammals offer an interesting example, for which {effective population size} is likely to show substantial variation across lineages.
This variation in $\Ne$ is expected to covary with life-history traits ({LHT}s), such that large-bodied species are expected to have smaller effective population sizes, compared to small-bodied species.

For computational reasons, we restricted our analyses to small concatenates made of 18 randomly sampled alignments of orthologous genes.
Since the mutation-selection model considered here assumes a mostly nearly-neutral regime, genes for which positive selection was detected using a site codon model were excluded.
To assess the reproducibility of our inference and check that the signal about variation in $\Ne$ is not driven by particular genes, we analysed 4 concatenated random samples of 18 genes.
The different concatenate showed similar trends in the change of $\mu$ ($r^2=[0.92,0.95]$) and $\Ne$ ($r^2=[0.51,0.68]$) between pairs of experiments (see supplementary).

The reconstructed long-term changes in {effective population size} ($\Ne$) is displayed in figure~\ref{fig:mammals_popsize_and_mutrate}.
We visually observe a global trend of increasing $\Ne$ throughout the tree around 90 and 60 My.
We also observe $\Ne$ to be lower in some clades, such as Cetacea and Camelidae, while being higher in other clades, such as Rodentia and Pecora.
In some cases, a decrease in $\Ne$ can be observed along an isolated branch of the tree, for example on the branches leading to the Alpaca (\textit{Vicugna pacos}) or the cheetah (\textit{Acinonyx jubatus}).

The estimated covariance matrix (table~\ref{tab:mammals_correlation}) gives a global synthetic picture about the patterns of covariation between the mutation rate per unit of time $\mu$, the {effective population size} $\Ne$ and the three {LHT}s.
First, the variation in $\mu$ across species is negatively correlated with variation in body mass, age at sexual maturity and longevity ($\rho=[-0.84, -0.83]$, table~\ref{tab:mammals_correlation}).
These correlations, which were previously reported~\citep{Lartillot2012,Nabholz2013} probably reflect generation time effects~\citep{Lanfear2010, Gao2016}.
Similarly, and more interestingly in the present context, the variation in $\Ne$ between species is also negatively correlated with {LHT}s ($\rho=[-0.54, -0.47]$, table~\ref{tab:mammals_correlation}).
This is consistent with the expectation that small-sized and short-lived species tend to be characterized by larger effective population sizes~\citep{Romiguier2014}.
Of note, these results mirror previous findings, based on classical {codon} models, showing that $\dnds$ tends to be positively correlated with {LHT}s~\citep{Lartillot2012,Nabholz2013,Figuet2017}.
Result which was also recovered on the present dataset, using a classical $\dnds$ based codon model (supplementary materials).
Interestingly, the correlation of $\dnds$ with LHTs is weaker than that of our inferred $\Ne$ with LHTs, as expected if the variation in $\dnds$ indirectly (and imperfectly) reflects the underlying variation in Ne.
Finally, $\Ne$ and $\mu$ are positively correlated in their variation ($\rho = 0.44$), which might simply reflect the fact that both negatively correlate with {LHT}s.
The partial-correlation coefficients (see supplementary) between $\Ne$ and {LHT}s are not significantly different from 0.
However, this might simply be due to the very strong correlation between the three {LHT}s considered here ($\rho=[0.81, 0.85]$), such that controlling for any one of them removes most of the signal contributed by the available empirical variation between species.

\begin{table*}[t]
    \input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsCorrelation.tex}
    \caption[Traits correlation in mammals]{
        Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity).
        Asterisks indicate strength of support of the {posterior} probability to be different than $0$ (pp) as $\smash{^{*}} pp > 0.95$ and $\smash{^{**}} pp > 0.975$.
        Observed correlations are compatible with the interpretation that large populations are composed of small, short-lived individuals.
        Moreover if the mutation rate per generation is considered constant in first approximation, the mutation rate per unit of time is positively correlated to generation rate, hence to population size.
    }
    \label{tab:mammals_correlation}
\end{table*}

Thus, altogether, the inferred trends in $\Ne$ across species appear to be as expected, based on considerations about life-history evolution.
On the other hand, the total range of the inferred variation in $\Ne$ across the entire extant taxa is surprisingly narrow, with one order of magnitude ($9.2$) at most between high and low $\Ne$ (see supplementary).
This almost certainly represents an underestimate of the true range of variation across placental mammals.

As another case study, we analysed a group of isopod species that have made multiple independent transitions to subterranean environments.
The transition from a terrestrial to a subterranean lifestyle is typically associated with a global life-history and ecological syndrome characterized by a loss of vision, longer generation times and, most interestingly, smaller population sizes, due to a lower carrying capacity of the subterranean environment~\citep{Capderrey2013}.
Protein coding {DNA} sequence alignments and qualitative life-history traits such as habitat (surface or underground), pigmentation (depigmented, partially depigmented or pigmented) and ocular structure (anophthalmia, microphthalmia, or ocular) are available for these species~\citep{Eme2013,Saclier2018}.
The assumption of a Brownian auto-correlated process for describing the changes in $\Ne$ along the tree may not be so well adapted to the present case, since the changes in $\Ne$ associated with the transition to a subterranean environment are likely to correspond to relatively sudden shifts, rather than continuous variation, and the ecological correlate (subterranean versus terrestrial) is not a quantitative trait.
However, the dataset considered here contains independent transitions to a subterranean lifestyle, thus offering an opportunity to test for a potential correlation between inferred $\Ne$ variation and terrestrial versus subterranean lifestyles over the terminal branches.
In our analysis across 4 concatenated random samples of 12 genes, we observe a reproducible (see supplementary) and statistically significant reduction in $\Ne$ for underground or depigmented species, or for species with visual impairment (see figure~\ref{fig:isopods_correlation}).
Of note, the species that dit not undergo a transition to subterranean environments feature a relative $\Ne$ close to $1$, meaning that $\Ne$ has not changed much along the lineages (since the root of the tree).
Again, the total range of the inferred variation in $\Ne$ across the entire extant taxa is surprisingly narrow, with ratio of $3.3$ at most between high and low $\Ne$ (see supplementary).

% redonner ici une evaluation du range de variation à travers la phylogénie.

Next, our empirical framework was also applied on a set of genes sampled across primates, taken from \citet{Perelman2011} and reanalysed in \citet{Brevet2019}.
In addition to LHTs (mass, female maturity, generation time and longevity), information about nuclear {synonymous} diversity ($\ps$) and {non-synonymous} over {synonymous} diversity ($\pnps$), are available for 10 species across the dataset and are expected to correlate with $\Ne$ according to population genetics~\citep{Eyre-walker2007, Galtier2016}.
However, the correlation coefficient between our inferred $\Ne$ and $\ps$ or $\pnps$ and LHTs are not statistically significant, nor with {LHT}s (see supplementary).
Again, the total range of the inferred variation in $\Ne$ across the entire tree is narrow, with ratio of $6.4$ at most between high and low $\Ne$.
This results contrasts with the finding of \citet{Brevet2019} on the same dataset based on $\dnds$-based {codon} models, where the estimated $\Ne$ was found to span several orders of magnitude, and correlated positively with $\ps$.


\begin{figure*}[t]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eco_merged}
    \end{minipage} \hfill
    \llap{\raisebox{-1.2cm}{$p_{\text{value}}< 10^{-16}$, $r^2=0.18$\hspace{0.65cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_pig_merged}
    \end{minipage} \hfill
    \llap{\raisebox{-1.2cm}{$p_{\text{value}}< 10^{-16}$, $r^2=0.20$\hspace{0.64cm}}}\hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eye_merged}
    \end{minipage}
	\llap{\raisebox{-1.2cm}{$p_{\text{value}}< 10^{-16}$, $r^2=0.20$\hspace{0.63cm}}}\hfill
    \caption[$\Ne$ as a function of traits in isopods]{
        $\Ne$ estimation for extant isopods species, sorted according to their habitat (left), pigmentation (middle), and ocular structure (right).
        All three qualitative trait statistically correlates with changes in $\Ne$.
        Underground, or depigmented species, or species with visual impairment are characteristic of low $\Ne$ species.}
    \label{fig:isopods_correlation}
\end{figure*}


\section{Discussion}
\label{sec:Discussion}
% Summary
Mechanistic phylogenetic {codon} models express the {substitution} rates between codons as a function of the mutation rates at the nucleotide level, selection over amino-acid sequences and {effective population size}.
Thus far, the development of mutation-selection models of the {HB} family~\citep{Rodrigue2010, Tamuri2012} has mostly focused on the question of fully accounting for the fine-scale modulations of selection between amino-acids and across sites~\citep{Rodrigue2010, Tamuri2012}.
However, the issue of the variation in the global population-genetic regime between species has received much less attention.
In particular, {effective population size} ($\Ne$) is expected to vary substantially over the species of a given clade, yet current mutation-selection models all invariably assume $\Ne$ to be constant across the phylogeny.

Here, we have introduced an extension of the mutation-selection model that accounts for this variation.
When applied to an alignment of protein coding sequences, this mechanistic model returns an estimate of the modulations of amino-acid preferences across sites.
Simultaneously, it reconstructs the joint evolution of life-history traits and molecular and population-genetic parameters (mutation rate $\mu$ and {effective population size} $\Ne$) along the phylogeny, while estimating the correlation matrix between these variables, intrinsically accounting for phylogenetic inertia.

\subsection{Reliability of the inference of the phylogenetic history of $\Ne$}

The reconstructions obtained on several empirical datasets, in particular in mammals and in isopods, suggest that the method is able to correctly infer the directional trends of the changes in $\Ne$ across species.
In particular, in mammals, the inferred variation in $\Ne$ correlates negatively with body size and, more generally, with life-history traits, as expected under the reasonable assumption that large-bodied mammals would tend to have smaller effective population sizes~\cite{Popadin2007,Lartillot2012,Nabholz2013,Figuet2017}.
Similarly, in isopods, smaller effective population sizes are inferred in subterranean species, again, as expected~\citep{Capderrey2013}.

However, if the trends are in right direction, the magnitude of the changes inferred across the phylogeny is surprisingly narrow and does not match independent empirical estimates of the variation in those clades.
In particular, in mammals, {synonymous} diversity varies by a factor at least 10 between species~\citep{Galtier2016}.
In animals, the {synonymous} diversity roughly spans two orders of magnitude, whereas $\Ne$ varies considerably more across species, by a factor of $10^3$~\citep{Galtier2019}.
For instance, effective population sizes estimated based on population genomic data are of the order of 10~000 in humans~\citep{Li2011}, and 100~000 in mice~\citep{Geraldes2008}.
Thus, clearly, our approach underestimates the true variation.
Different mechanisms not accounted for by the model could explain this result.

First, genetic hitchhiking, Hill-Robertson interference, and short-term fluctuations of $\Ne$ could generate this effect.
However, inference conducted on alignments simulated under a Wright-Fisher model accounting for linkage and for short-term variation in $\Ne$ suggests that empirically reasonable levels of Hill-Robertson interferences are not strong enough to explain this observation, at least in the regimes explored.
Second, $\mu$ and $\Ne$ could also be fluctuating along the genome~\citep{Gossmann2011,Ellegren2003,Eyre-Walker2014}.
This assumption needs to be tested, though we expect that relaxing this assumption would not change drastically the magnitude of inferred $\Ne$ since some of this fluctuation should be absorbed by the inferred site-specific fitness profiles.
Third, the {DNA} sequences could also be misaligned at some sites.
However we observe the same magnitude of inferred $\Ne$ for different sets of genes indicating this might not be the primary reason.
Fourth, the genes selected in our alignments could be under adaptive evolution, or their function could have changed.
However, at least in mammals, the impact of this potential problem was minimized by the use of genes for which no positive selection was detected using standard phylogenetic {codon} site models.

Finally, one key assumption of the mutation-selection model that is likely to be violated in practice is the assumption of site-independence.
In reality, epistasis might be prevalent in protein coding sequence evolution~\citep{Pollock2014,Shah2015}.
Our simulations under an epistatic landscape point to epistasis being a major factor to be investigated.
Indeed, $\Ne$ could not be appropriately estimated under these simulation settings, although the outcome more specifically depends on the exact model for the fitness landscape.
An extreme case is obtained using a biophysically-inspired model, assuming purifying selection for conformational stability.
This model was previously explored using simulations and theoretical developments \citet{Goldstein2013}, and it was shown that, under this model, $\dnds$ and more generally the {substitution} process is virtually insensitive to $\Ne$.
This is confirmed by our experiments, showing that the mutation-selection approach explored here cannot infer the true variation in $\Ne$ under this model.

A less extreme outcome is obtained under an alternative model also implementing epistatic interactions between sites via Fisher’s geometric model~\citep{Tenaillon2014,Blanquart2016}.
Interestingly, under this model, our inference framework is able to infer the correct trends of $\Ne$, although with a substantially underestimated range of inferred variation, thus mirroring the results obtained on placental mammals.
Of note, these results do not necessarily imply that models based on biophysics are empirically less relevant than Fisher’s geometric model.
Instead, they might just betray that the response of the {substitution} process to changes in $\Ne$ may be sensitive to the exact quantitative details of the underlying fitness landscape.
More work is probably needed here to characterize these exact conditions.
Nevertheless, our simulation experiments suggest a global pattern: epistatic interactions induce a buffering of the response of the {substitution} process to changes in $\Ne$.
The meaningful correlation patterns observed with {LHT}s in the case of placental mammals suggest that this buffering is not complete.
Nevertheless, ignoring epistatic interactions at the inference level appears to result in a substantial underestimation of the range over which $\Ne$ varies across species.

Interestingly, the magnitude of the inferred range of $\Ne$ variation is similar for the placental and the primate datasets (with a 9-fold and 6-fold variation in mammals and primates, respectively), whereas one would have expected a much larger range of variation over the broader phylogenetic scale of placental mammals, compared to primates.
An explanation could be that the effects of epistasis are more apparent at longer time-scales.
Indeed, the total number of substitutions from root to leaves is greater, and as a result, the local environment, and therefore the fitness landscape at the level of each site, has been less stable across the phylogeny.

Although modelling epistasis in an inference framework is a complex biological, mathematical and computational problem, our work points to a potential signal of epistasis that could be retrieved in a phylogenetic context.
More specifically, since the slope of the response of the {substitution} process to changes in $\Ne$ appears to be informative about the epistatic regime, then, conversely, by relying on independent estimates of $\Ne$ (e.g. using polymorphism), this effect could be used to leverage a quantitative estimate of the statistical distribution of epistatic effects.

Other methods have recently been developed to reconstruct phylogenetic changes in $\Ne$.
For example, a method recently developed uses polymorphism and generation time for some present-day species to reconstruct $\Ne$ along the phylogeny, based on a classical ($\dnds$-based) {codon} model~\citep{Brevet2019}.
This method implicitly relies on a {nearly-neutral} model, assuming a fixed and gamma-shaped distribution of fitness effects across {non-synonymous} mutations.
The approach is calibrated using fossils, and as a result, returns estimates of the absolute value of $\Ne$ and of its phylogenetic variation.
Here, in contrast, our method requires neither generation times nor polymorphism data, and the fitness effects are not constrained to a specific distribution.
On the other hand, the inferred effective population sizes are only relative.
In addition, the empirical fitting of the model requires more computing resources.

\subsection{Potential applications and future developments}

Apart from reconstructing the phylogenetic history of $\Ne$ and investigating its causes and covariates, another potentially interesting application of our approach is in detecting adaptation.
In this direction, mutation-selection models represent a useful null {nearly-neutral} model, explicitly modelling the background of purifying selection acting over protein coding genes.
Adaptation can then be detected by measuring the deviation from this null model~\citep{Rodrigue2016, Bloom2017}.

However, by assuming a constant $\Ne$ along a phylogeny, the statistical power of this approach to detect sites under adaptive evolution may not be optimal.
In particular, the site-specific fitness profiles inferred by the model are averaged along the phylogeny and are seemingly more diffuse than those estimated profiles under our present framework (see supplementary materials).
Thus, our method should provide a better null model of purifying selection against which to test for the presence of adaptive evolution.

This approach can be further extended in other directions.
First, currently, our model also assumes no selection on {codon} usage.
In the case of primates or placental mammals, this assumption is probably reasonable~\citep{Yang2008}, although it is more questionable for other groups, in particular Drosophila~\citep{Duret1999,Plotkin2011}.
In principle, this assumption can be relaxed by implementing selective {codon} preferences that are shared across all sites.
Such an implementation would provide the advantage of estimating {codon} usage biases, while simultaneously accounting for its confounding effect when estimating selection on amino-acids and inter-specific variation in $\Ne$.

Second, the Bayesian analysis conducted here was based on relatively small alignments (20~000 sites at most), and with strong limits on the parametrization of the underlying mixture model (allowing for at most 50 distinct profile categories).
Profiling of the program (not shown) shows that the number of components of the profile mixture is the limiting step of the computation.
Yet, a larger number of components might be required, in order to achieve more accurate inference of the site-specific profiles.
One possible development, leading to statistically more stable genome-wide estimates of $\Ne$, would be to develop a multi-gene parallelized version of the model, in which each coding sequence would have its own mixture model, and would run on a separate thread, while the history of $\Ne$ would be shared by all computing processes.

Finally, estimating $\Ne$ in a mutation-selection phylogenetic model relies on the relation between $\Ne$ and the relative strength of drift, in a context where, ultimately, the signal about the intensity of drift comes from the relative rate of {non-synonymous} substitutions.
However, this purely phylogenetic approach does not leverage a second aspect of $\Ne$ at the population level, namely, the fact that $\Ne$ also determines the levels of {neutral} genetic diversity that can be maintained ~($\pi=4\Ne u$, where $u$ is the mutation rate per generation).
Hence, {neutral} diversity yields an independent empirical estimate of $\Ne$.
In principle, our mechanistic model could be extended so as to incorporate polymorphism data within species at the tips of the phylogeny.
A similar method has been previously pioneered in the case of 3 species and using a distribution of fitness effect\citep{Wilson2011}.
More generally, the {nearly-neutral} theory of evolution defines a long-term $\Ne$, which might be different from the short-term definition of $\Ne$~\citep{Platt2018}.
Thus we could ask if empirical independent estimations of $\Ne$ from within species (based on genetic diversity) and between species (based on the {substitution} process) are congruent, and if not, what are the mechanisms responsible for this discrepancy.

Notwithstanding theoretical considerations on the {nearly-neutral} theory of evolution, empirical clues about the long-term trends in the modulations of the intensity of {genetic drift} opens up a large diversity of ecological and evolutionary questions.
Spatial and temporal changes of {genetic drift} along ecological niches and events can now be investigated, so as to disentangle the underlying evolutionary and ecological pressures.


\section{Materials and Methods}
\label{sec:MatMet}

In the model presented here, $\Ne$ and $\mu$ (and quantitative traits) are allowed to vary between species (across branches) as a multivariate log-Brownian process, but assumed constant along the DNA sequence.
Conversely, amino-acid fitness profiles are assumed to vary across sites, but are considered constant along the tree.
The model makes several assumptions about the evolutionary process generating the observed alignment.
First, the species tree topology is supposed to be known, and each gene should match the species tree, meaning genes are strict orthologs (no paralogs and no horizontal transfers).
Second, there is no epistasis (interaction between sites), such that any position of the sequence has its own independent evolutionary process and a substitution at one position does not affect the substitution process at other positions.
Third, from a population genetics perspective, we assumed sites of the protein to be unlinked, or equivalently the mutation rate is low enough such that there is no Hill-Robertson interference nor genetic hitchhiking.
Fourth, polymorphism is ignored in extant species.

The parameterization of the models is described as a Bayesian hierarchical model, including the prior distributions and the parameters of the model.
This hierarchical model is formally represented as directed acyclic graph, depicted in figure~\ref{fig:DAG-MutSelNe}.

\subsection{Nucleotide mutation rates}
The generalized time-reversible nucleotide mutation rate matrix $\Mutmatrix$ is a function of the nucleotide frequencies $\Mutequi$ and the symmetric exchangeability rates $\Exchan$~\citep{Tavare1986}.
$\Mutequi = (\mutequi_A , \mutequi_C , \mutequi_G , \mutequi_T)$ is the equilibrium base frequency vector, giving the frequency at which each base occurs at each site.
$\Exchan = \left( \exchan_{AC}, \exchan_{AG}, \exchan_{AT}, \exchan_{CG}, \exchan_{CT}, \exchan_{GT}\right)$ is the vector of exchangeabilities between nucleotides.
Altogether, the rate matrix is:
\begin{equation}
    \label{eq:gtr-mutrates}
    \Mutmatrix =
    \begin{blockarray}{ccccc}
        & A & C & G & T \\
        \begin{block}{c(cccc)}
            A & - & {\exchan_{AC}\mutequi_C} & {\exchan_{AG}\mutequi_G} & {\exchan_{AT}\mutequi_T} \\
            C & {\exchan_{AC}\mutequi_A} &                        - & {\exchan_{CG}\mutequi_G} & {\exchan_{CT}\mutequi_T} \\
            G & {\exchan_{AG}\mutequi_A} & {\exchan_{CG}\mutequi_C} &                        - & {\exchan_{GT}\mutequi_T} \\
            T & {\exchan_{AT}\mutequi_A} & {\exchan_{CT}\mutequi_C} & {\exchan_{GT}\mutequi_G} & - \\
        \end{block}
    \end{blockarray}
\end{equation}
By definition, the sum of the entries in each row of the nucleotide rate matrix $\Mutmatrix$ is equal to $0$, giving the diagonal entries:
\begin{equation}
    \mutmatrix_{a,a} = - \sum\limits_{ b \neq a, b \in \{A, C, G, T\} } \mutmatrix_{a,b}
\end{equation}
The {prior} on the exchangeabilities $\Exchan$ is a uniform Dirichlet distribution of dimension $6$:
\begin{equation}
    \label{eq:DistribExchan}
    \Exchan \sim \text{Dir}\left( \dfrac{1}{6} , 6\right).
\end{equation}
The {prior} on the equilibrium base frequencies $\Mutequi$ is a uniform Dirichlet distribution of dimension $4$:
\begin{equation}
    \label{eq:DistribMutequi}
    \Mutequi \sim \text{Dir}\left( \dfrac{1}{4} , 4\right)
\end{equation}
The general time-reversible nucleotide matrix is normalized such that the total flow equals to $1$:
\begin{equation}
    \sum\limits_{a \in \{A, C, G, T\}} - \mutequi_a \mutmatrix_{a,a} = 1.
\end{equation}

\subsection{Site-dependent selection}
\label{sec:profiles}
Site-specific amino-acid fitness profiles are assumed i.i.d. from a mixture model, itself endowed with a truncated Dirichlet process prior.
Specifically, the mixture has $\Ncat$ components ($\Ncat = 50$ by default).
The prior on component weights~($\StickBreaking$) is modeled using a stick-breaking process, truncated at $\Ncat$ and of parameter $\stickbreakinghyper$:
\begin{align}
    \label{eq:DistribStickBreaking}
    \begin{split}
        & \StickBreaking \sim \text{StickBreaking}\left( \Ncat, \stickbreakinghyper \right)\\
        \iff & \stickbreaking_{\cat} = \stick_{\cat}\cdot \prod _{{\indice=1}}^{{\cat-1}}\left(1-\stick_{\indice}\right),\ \Setcat,
    \end{split}
\end{align}
where $\stick_{\cat}$ are i.i.d. from a beta distribution
\begin{equation}
    \label{eq:Beta}
    \stick_{\cat} \sim \text{Beta}\left( 1, \stickbreakinghyper \right),\ \Setcat.
\end{equation}
Of note, the weights decrease geometrically in expectation, at rate $\stickbreakinghyper$, such that lower values of $\stickbreakinghyper$ induce more heterogeneous distributions of weights.

Each component of the mixture defines a 20-dimensional fitness profile $\Profile\catexp$ (summing to $1$), for $ \Setcat$.
These fitness profiles are i.i.d. from a Dirichlet of center $\centerProfile$ and concentration $\concentrationProfile$:
\begin{equation}
    \label{eq:DistribBase}
    \Profile\catexp \sim \text{Dir}\left( \centerProfile,\ \concentrationProfile \right),\ \Setcat.
\end{equation}

Site allocations to the mixture components $\catsite \in \catInterval $, for $\Setsite$ running over the $\Nsite$ sites of the alignment, are i.i.d. multinomial of parameter $\StickBreaking$:
\begin{align}
    \label{eq:DistribMultinomial}
    \catMultiVar \sim \text{Multinomial}\left( \StickBreaking \right).
\end{align}

For a given parameter configuration for the mixture, the Malthusian fitness selection coefficients $\Fit\siteexp$ at site $\site$, are obtained by taking the logarithm of the fitness profile assigned to this site:
\begin{equation}
    \label{eq:sitefitness}
    \Fit\siteexp = \ln \left( \Profile^{\left( \catsite \right)} \right),\ \Setsite.
\end{equation}

\subsection{Dated tree}
The topology of the rooted phylogenetic tree is supposed to be known and is not estimated by the model.
The model estimates the dates at which branches split, thus the dated tree requires $\Ntaxa - 2$ internal node ages that are free parameters, where $\Ntaxa$ is the number of extant taxa (leaves of the tree).
By definition, leaf ages are all set to $0$.
The root age is set arbitrarily to $1$, but if fossils data are also available the dated tree can be rescaled into absolute time using cross-multiplication.
A uniform prior is assumed over internal node ages $\age\nodeexp,\ \Setinternal$.

The duration $\branchtime\branchexp$ represented by a given branch $\branch$, for $\Setbranch$ is defined as the difference in ages between the oldest node at the tip of the branch $\age^{(\nodeUp)}$, and the youngest node $\age^{(\nodeDown)}$:
\begin{equation}
    \label{eq:ageTobranchtime}
    \branchtime\branchexp = \age^{(\nodeUp)} - \age^{(\nodeDown)}.
\end{equation}

\begin{figure*}[t]
    \centering
    \begin{tikzpicture}[->,>=stealth',shorten >=1pt,auto,node distance=0.6cm and 1.2cm,semithick]
        \tikzstyle{every state}=[]

        \node[state] (P) {$\Probmatrix\branchsiteexp$};
        \node[state] (Q) [below right=of P] {$\Submatrix\branchsiteexp$};
        \node[state] (R) [below right=of Q] {$\Mutmatrix$};
        \node[state] (BL) [above right=of P] {$\branchlength\branchexp$};
        \node[state] (Ne) [above right=of Q] {$\Ne\branchexp$};
        \node[state] (Bb) [above right=of Ne] {$\Brownian\nodeexp $};
        \node[state] (Mu) [left=of Bb] {$\mu\branchexp$};
        \node[state] (f) [right=of Q] {$\Fit\siteexp $};
        \node[state] (Ex) [BLUE, below right=of R] {$\Exchan$};
        \node[state] (Equi) [BLUE, right=of R] {$\Mutequi$};
        \node[state] (dT) [above left=of Bb] {$\branchtime\branchexp $};
        \node[state] (T) [BLUE, right=of dT] {$\age\nodeexp$};
        \node[state] (Base) [BLUE, above right=of f] {$\Profile\catexp$};
        \node[state] (cat) [BLUE, right=of f] {$\catsite$};
        \node[state] (ExH) [RED, right=of Ex] {$\dfrac{1}{6}, 6$};
        \node[state] (EquiH) [RED, right=of Equi] {$\dfrac{1}{4}, 4$};
        \node[state] (Unif) [RED, right=of T] {$\uniform$};
        \node[state] (Cov) [BLUE, right=of Bb] {$\CovarianceMatrix$};
        \node[state] (baseH) [RED, right=of Base] {$\concentrationProfile, \centerProfile $};
        \node[state] (sb) [BLUE, right=of cat] {$\StickBreaking$};
        \node[state] (CovH) [RED, right=of Cov] {$\covariancekappa, \covariancedf$};
        \node[state] (sbH) [RED, right=of sb] {$\stickbreakinghyper$};

        \path
        (Q) edge [black] node [above right] {\ref{eq:Probmatrix}} (P)
        (dT) edge [black] node [above left] {\ref{eq:branchlength}} (BL)
        (Mu) edge [black] node [] {} (BL)
        (BL) edge [black] node [] {} (P)
        (Ne) edge [black] node {} (Q)
        (R) edge [black] node {} (Q)
        (Bb) edge [black] node {} (Ne)
        (Bb) edge [black] node [below] {\ref{eq:branchNemu}} (Mu)
        (f) edge [black] node [above] {\ref{eq:subrates}} (Q)
        (Ex) edge [black] node [] {} (R)
        (Equi) edge [black] node [below] {\ref{eq:gtr-mutrates}} (R)
        (T) edge [black] node [above] {\ref{eq:ageTobranchtime}} (dT)
        (dT) edge [black] node {} (Bb)
        (Base) edge [black] node {} (f)
        (cat) edge [black] node [above] {\ref{eq:sitefitness}} (f)
        (ExH) edge [dashed, BLUE] node [above] {\ref{eq:DistribExchan}} (Ex)
        (EquiH) edge [dashed, BLUE] node [above] {\ref{eq:DistribMutequi}} (Equi)
        (Unif) edge [dashed, BLUE] node [above] {} (T)
        (Cov) edge [dashed, BLUE] node [above] {\ref{eq:DistribBrownian}} (Bb)
        (baseH) edge [dashed, BLUE] node [above] {\ref{eq:DistribBase}} (Base)
        (sb) edge [dashed, BLUE] node [above] {\ref{eq:DistribMultinomial}} (cat)
        (CovH) edge [dashed, BLUE] node [above] {\ref{eq:Distribcovariance}} (Cov)
        (sbH) edge [dashed, BLUE] node [above] {\ref{eq:DistribStickBreaking},\ref{eq:Beta}} (sb);
    \end{tikzpicture}

    \caption[Directed acyclic graph of dependencies between variables]{
        Directed acyclic graph (DAG) of dependencies between variables.
        Nodes of the directed acyclic graph are the variables, and edges are the functions.
        Hyper-parameters are depicted in {\color{RED}{red}} circles, random variables in {\color{BLUE}{blue}} circles, and transformed variables in black.
            {\color{BLUE}{Blue}} dashed line denotes a drawing from a random distribution, and black solid lines denote a function.
        For a given node, all the nodes pointing toward him (upstream) are its dependencies which determines its distribution.
        The other way around, following the arrows in the DAG (downstream), simple {prior} distributions are combined together to form more complex joint {prior} distribution which ultimately defines the {prior} distribution of the model.
    }\label{fig:DAG-MutSelNe}
\end{figure*}

\subsection{Branch dependent traits}
The {effective population size} $\Ne$ and mutation rate per unit of time $\mu$ are assumed to evolve along the phylogeny, and to be correlated.
If quantitative life-history traits ({LHT}s) are also available for some nodes of the tree (leaves and/or internal nodes), they are also assumed to evolve along the phylogeny and to be correlated between them, and with $\Ne$ and $\mu$.
The total number of traits is noted $\Ntrait$, when counting $\Ne$, $\mu$ and all user-defined LHT (denoted $\Trait$).
Their variation through time is modelled by an $\Ntrait$-dimensional log-Brownian process $\Brownian$.
By convention, the first component of the log-brownian corresponds to $\Ne$, and the second component to $\mu$.
Thus:
\begin{equation}
    \begin{dcases}
        \brownian_1 (t) = \ln \Ne (t) \\
        \brownian_2 (t) = \ln \mu (t) \\
        \brownian_{k+2}(t) = \ln \trait_k(t), k \in \traitInterval
    \end{dcases}
\end{equation}
The effective population size at the root is set to $1$ for identifiability of the fitness profiles.

Along a branch $\Setbranch$ of the tree, a log-Brownian process starts at the oldest node at the tip of the branch~($\nodeUp$), and ends at the youngest node~($\nodeDown$).
The rate of change of the log-Brownian process per unit of time is constant and determined by the positive semi-definite and symmetric covariance matrix $\CovarianceMatrix$. Thus the distribution at node $\nodeDown$ of $\Brownian^{(\nodeDown)}$ is multivariate Gaussian, with mean equals to the Brownian process sampled at the oldest node $\Brownian^{(\nodeUp)}$, and variance $\branchtime\branchexp \CovarianceMatrix$:
\begin{equation}
    \label{eq:DistribBrownian}
    \Brownian^{(\nodeDown)} \sim \mathcal{N}\left(\Brownian^{(\nodeUp)}, \branchtime\branchexp \CovarianceMatrix \right),\ \Setbranch.
\end{equation}
The Brownian process at the root of the tree is uniformly distributed, except for the first component fixed to $0$ for identifiability (see above).
The {prior} on the covariance matrix is an inverse Wishart distribution, parameterized by $\covariancekappa=1$ and with $\covariancedf=\Ntrait + 1$ degrees of freedom:
\begin{equation}
    \label{eq:Distribcovariance}
    \CovarianceMatrix \sim \text{Wishart}^{-1} (\covariancekappa \Identitymatrix, \covariancedf).
\end{equation}

We are interested in approximating the expected substitution rates between codons over the branch.
Ideally, under the Brownian process just described, the rates of substitution between codons are continuously changing through time.
Also, even conditional on the value of $\Ne$ at both ends, the Brownian path along the branch entails a random component, leading to complicated integral expressions for substitution rates~\citep{Horvilleur2014}.
Here, a branchwise approximation is used~\citep{Lartillot2011}, which consists of first deriving an approximation for the mean $\Ne$ along the branch, conditional on the values of $\Ne$ at both ends, and then using this mean branchwise $\Ne$ to define the codon substitution rates.

In the case of log-Brownian process, the most likely path (or geodesic) from $\Brownian^{(\nodeUp)}$ to $\Brownian^{(\nodeDown)}$ is the straight line, and therefore, it would make sense to take the mean value of $\e^{\Brownian\nodeexp}$ along this geodesic.
We then have $\Ne\branchexp$ and $\mu\branchexp$ for each branch $\Setbranch$ of the tree:
\begin{equation}
    \label{eq:branchNemu}
    \begin{dcases}
        \Ne\branchexp = \dfrac{\e^{\brownian_{1}^{(\nodeDown)}} - \e^{\brownian_{1}^{(\nodeUp)}}}{\brownian_{1}^{(\nodeDown)} - \brownian_{1}^{(\nodeUp)}} \\
        \mu\branchexp = \dfrac{\e^{\brownian_{2}^{(\nodeDown)}} - \e^{\brownian_{2}^{(\nodeUp)}}}{\brownian_{2}^{(\nodeDown)} - \brownian_{2}^{(\nodeUp)}}.
    \end{dcases}
\end{equation}

\subsection{Codon substitution rates}

The mutation rate between codons $\ci$ and $\cj$, denoted $\mu_{\itoj}$ depends on the underlying nucleotide change between the codons.
First, if codons $\ci$ and $\cj$ are not nearest-neighbours, $\mu_{\itoj}$ is equal to $0$.
Second, if codons $\ci$ and $\cj$ are only one mutation away, $\nucitoj$ denotes the nucleotide change (e.g.~$\nuc(AAT, AAG) = TG$), and $\mu_{\itoj}$ is given by the underlying nucleotide relative rate~(${\mutmatrix_{\nucitoj}}$) scaled by the mutation rate per time~($\mu$).
Technically, the $4$-dimensional nucleotide relative rate matrix~($\Mutmatrix$) is normalized such that we expect $1$ {substitution} per unit of time, hence the scaling by $\mu$.

For a given branch $\branch$ and a given site $\site$, the {codon} {substitution} rate (per unit of time) matrix $\Submatrix\branchsiteexp$ is given by:
\begin{equation}
    \label{eq:subrates}
    \begin{dcases}
        \submatrix\branchsiteexp_{\itoj} = 0\text{ if codons $\ci$ and $\cj$ are not neighbors,} \\
        \submatrix\branchsiteexp_{\itoj} = \mutmatrix_{\nucitoj}\text{ if codons $\ci$ and $\cj$ are {synonymous},} \\
        \submatrix\branchsiteexp_{\itoj} = \mutmatrix_{\nucitoj} \dfrac{4\Ne\branchexp \left({\fitj\siteexp - \fiti\siteexp}\right)}{{1 - \e^{4\Ne\branchexp\left({\fiti\siteexp - \fitj\siteexp}\right)} }} \text{ if $\ci$ and $\cj$ are non-synonymous,}\\
        \submatrix\branchsiteexp_{\ci, \ci} = - \sum\limits_{ \cj \neq \ci, \cj=1}^{61} \submatrix\branchsiteexp_{\itoj}.
    \end{dcases}
\end{equation}
We see from this equation that, $\fit$ and $\Ne$ are confounded, such that increasing the {effective population size} while decreasing the fitnesses by the same factor leads to the same {substitution} rate.

The branch lengths $\branchlength\branchexp$ are defined as the expected number of {neutral} substitutions per {DNA} site along a branch:
\begin{equation}
    \label{eq:branchlength}
    \branchlength\branchexp = \mu\branchexp \branchtime\branchexp.
\end{equation}
Together, the probability of transition between codons for a given branch $\branch$ and site $\site$ is:
\begin{equation}
    \label{eq:Probmatrix}
    \Probmatrix\branchsiteexp = \e^{\branchlength\branchexp \Submatrix\branchsiteexp},
\end{equation}
which are the matrices necessary to compute the {likelihood} of the data ($\data$) given the parameters of the model using the pruning algorithm.

\subsection{Bayesian implementation}
\label{sec:Bayesian}
Bayesian inference was conducted using Markov Chain Monte Carlo (MCMC).
Most phylogenetic {MCMC} samplers target the distribution over the model parameters given the sequence alignment, which means that they have to repeatedly invoke the pruning algorithm to recalculate the {likelihood} which is most often the limiting step of the {MCMC}.
An alternative, which is used here, is to do the {MCMC} conditionally on the detailed {substitution} history $\subhistory$, thus doing the {MCMC} over the augmented configuration~($\subhistory$, $\data$), under the target distribution obtained by combining the mapping-based {likelihood} with the {prior} over model parameters.

The key idea that makes this strategy efficient is that the mapping-based {likelihood} depends on compact summary statistics of $\subhistory$, leading to very fast evaluation of the {likelihood}.
On the other hand, this requires to implement more complex {MCMC} procedures that have to alternate between:
\begin{enumerate}
    \item sampling $\subhistory$ conditionally on the data and the current parameter configuration.
    \item re-sampling the parameters conditionally on $\subhistory$.
\end{enumerate}

To implement the mapping-based {MCMC} sampling strategy, we first sample the detailed {substitution} history $\subhistory$ for all sites along the tree.
Several methods exist for doing this~\citep{Nielsen2002,Rodrigue2008}, which are used here in combination (first trying the accept-reject method of Nielsen, then switching to the uniformization approach of Rodrigue \textit{et al} if the first round has failed).

Then, we write down the probability of $\subhistory$ given the parameters, and finally, we collect all factors that depend on some parameter of interest and make some simplifications.
This ultimately leads to relatively compact sufficient statistics (see supplementary) allowing for fast numerical evaluation of the likelihood~\citep{Irvahn2014,Davydov2016}.
As an example, making an {MCMC} move on the $\Ne$ at a given node of the tree is faster since only the mapping-based {likelihood} (using path sufficient statistics) at the neighbouring branches of the node is necessary, instead of computing the likelihood for the entire tree.

Markov chain Monte Carlo (MCMC) are run for 4000 points and the first 1000 points are discarded as burn-in.
Convergence is then assessed (see supplementary) by comparing two independent chains, checking that both site-specific fitness and branch $\Ne$ have the same posterior mean.

\subsection{Correlation between traits}
\label{sec:Correlation}
The correlation between trait $\traiti$ and trait $\traitj \in \traitInterval$ can be obtained from the covariance matrix $\Covariancematrix$:
\begin{equation}
    \rho_{\traiti, \traitj} = \dfrac{\Covariancematrix_{\traiti, \traitj}}{\sqrt{\Covariancematrix_{\traiti, \traiti} \Covariancematrix_{\traitj, \traitj}}}.
\end{equation}
This correlation coefficient is then averaged over the posterior distribution, and statistical support is assessed based on the posterior probability of having a positive (or negative) value for the coefficient.

\subsection{Simulations}
\label{sec:Simulation}
To test the robustness of the model, four parameterized simulators were developed: \texttt{SimuDiv}, \texttt{SimuPoly}, \texttt{SimuFold} \& \texttt{SimuGeo}.
All four simulators use a log-Brownian multivariate process to model the changes in the mutation rate per generation, the generation time and $\Ne$ along the lineages.
\texttt{SimuDiv}, \texttt{SimuFold} \& \texttt{SimuGeo} all simulate point substitutions along the phylogenetic tree.
The simulator starts from an initial sequence at equilibrium.
The change in fitness is computed for all possible mutant, hence computing all strictly positive {substitution} rates.
At each point, the next {substitution} is chosen proportional to these rates using in Gillespie's algorithm~\citep{Gillespie1977}.
At each node, the process is split, and finally stopped at the leaves of the tree.
\texttt{SimuPoly} simulates explicitly each generation along the phylogeny under a Wright-Fisher population, consisting of three steps: mutation, selection and {genetic drift} of currently segregating alleles.
Mutations are drawn randomly based on mutation rates.
Drift is induced by the multinomial resampling of the currently segregating alleles.
We assume that the {DNA} sequence is composed of exons, with no linkage between exons, and total linkage of sites within an exon.
Moreover, in \texttt{SimuPoly}, the instant value of log-$\Ne$ can also be modelled as a sum of a log-Brownian process and an Ornstein-Uhlenbeck process.
The log-Brownian motion accounts for long-term fluctuations, while the Ornstein-Uhlenbeck introduces short-term fluctuations.
In \texttt{SimuDiv} and \texttt{SimuPoly}, each {codon} site contributes independently to the fitness depending on the encoded amino acids, through site-specific amino-acid fitness profiles experimentally determined~\citep{Bloom2017}.
In \texttt{SimuFold}, the fitness of a sequence is computed as the probability of the protein to be in the folded state.
\texttt{SimuFold} is a \texttt{C++} adaptation of a Java code previously published~\citep{Goldstein2016, Goldstein2017}, where we also allow for changes in $\Ne$ and $\mu$ along a phylogenetic tree.
Supplementary materials describe the models in more details, as well as performance of the inference model against them.

\subsection{Empirical data}
For placental mammals, alignments were extracted from OrthoMam database~\citep{Ranwez2007,Scornavacca2019}.
Only highly conserved coding sequences are kept for the analysis, representing 226 {CDS} with $\leq 1\%$ of gaps in the alignment.
Life-history traits ({LHT}s) for longevity, age at maturity and weight were obtained from AnAge database~\citep{DEMAGALHAES2009,Tacutu2012}.
We focused our analysis on 77 taxa for which information is available for at least one {LHT}.

\section{Reproducibility - Supplementary Materials}
The simulators written in \texttt{C++} are publicly available under MIT license at \url{https://github.com/ThibaultLatrille/SimuEvol}.
The Bayesian inference model, written in \texttt{C++} in the component based~\citep{Lanore2019} software \texttt{BayesCode}, is publicly available at \url{https://github.com/ThibaultLatrille/bayescode}.
Supplementary materials and figures are available in appendix supplementary materials.
The scripts and instructions necessary to reproduce the simulated and empirical experiments are available at \url{https://github.com/ThibaultLatrille/MutationSelectionDrift}.

\section{Author contributions}
TL gathered and formatted the data, developed the new models in \texttt{BayesCode} and \texttt{SimuEvol} and conducted all analyses, in the context of a PhD work (Ecole Normale Superieure de Lyon).
VL restructured and refactored the code sustaining the branch and site heterogeneous Bayesian Monte Carlo in \texttt{BayesCode}.
TL and NL both contributed to the writing of the manuscript.

\section{Acknowledgements}
We wish to thank Tristan Lefébure for sharing the isopods phylogeny, alignments and life-history traits.
We thank Philippe Veber for insightful discussion on mutation-selection models and software development.
We gratefully also acknowledge the help of Nicolas Rodrigue, Laurent Gueguen, Benoit Nahbolz and Laurent Duret for their advice and review concerning this manuscript.
This work was performed using the computing facilities of the CC LBBE/PRABI.
Funding: French National Research Agency, Grant ANR-15-CE12-0010-01 / DASIRE.

\bibliographystyle{natbib}%%%%Bibliography style file
\bibliography{references,introduction}
    
\end{document}
