\section{Summary statistics}

\subsection{Partial correlation coefficient}

The correlation coefficient $\rho_{\traiti, \traitj}$ give the total regression between two variable.
Partial-correlation coefficient account for the entire covariance matrix, and measure the correlation between $2$ traits, knowing the values of all the other traits:
\begin{equation}
    \rho_{\traiti, \traitj | c \in \traitInterval \setminus \{ a, b\} } = - \dfrac{\Precisionmatrix_{\traiti, \traitj}}{\sqrt{\Precisionmatrix_{\traiti, \traiti} \Precisionmatrix_{\traitj, \traitj}}},
\end{equation}
where the precision matrix $\PrecisionMatrix$ is the inverse of the covariance matrix:
\begin{equation}
    \PrecisionMatrix = \CovarianceMatrix^{-1}
\end{equation}

\subsection{Fitness profile entropy}
For a category $\cat$, the Shannon entropy ($\entropy$) of the fitness profile is defined as:
\begin{equation}
    \entropy\catexp = - \sum_{\aSetAa} \base\catexp_{\aminoacid} \log \left( \base\catexp_{\aminoacid} \right)
\end{equation}
The Shannon entropy measures the flatness of the fitness profile, with a value of $0$ corresponding to a single peak fitness landscape (only one amino-acid is present), and a value of $log(20)\simeq3$ corresponding to a \gls{neutral} landscape, where each amino-acid has the same fitness.

The Shannon entropy can be averaged over all sites as:
\begin{equation}
    \langle \entropy \rangle = \dfrac{1}{\Nsite}\sum_{\Setsite} \entropy\catsite
\end{equation}


\section{Simulations}

\subsection{Independent fitness profiles}
Under a simulation with site independent fitness profiles, a fitness profile give a fitness for each amino-acid (vector of size $20$).
Each site of the protein has a specific amino-acid fitness profile.
Overall, the protein fitness is computed as the sum of site-specific fitness, obtained by accessing the amino-acid present at each site of the protein.
For each possible mutant, we compute the selection coefficient of the mutant:
\begin{equation}
    s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right) = \sum_{1 \leq \site \leq \Nsite} \ln \left( \dfrac{\Base_{\site} \left(\mathbb{S}^{t+1}(\site) \right)}{\Base_{\site} \left(\mathbb{S}^{t}(\site) \right)} \right),
\end{equation}
where $\Base_{\site}$ is the fitness profile at site $\site$, obtained in empirical experiment \citep{Bloom2017}.

\begin{center}
    \includegraphics[width=\textwidth] {ModelSimuDiv}
\end{center}

The next change in the protein coding \acrshort{DNA} and the time to next the event is chosen using Gillespie algorithm, according to the rates of \gls{substitution} between \glspl{codon}:
\begin{equation}
{\submatrix_{\itoj}}
    = \mu_{\itoj} \dfrac{4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)} }},
\end{equation}
where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of \glspl{synonymous}.


\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
    \end{minipage} \hfill
    \caption[Inferred branch parameters for SimuDiv]{
    Inferred branch parameters under simulation accounting for long term fluctuation of $\Ne$, mutation rate per generation and generation time.
    }
\end{figure}


\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_ProfileCorrelation.png}
    \end{minipage}    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSel_ProfileCorrelation.png}
    \end{minipage}
    \caption[Inferred site amino-acid profiles for SimuDiv]{
    Inferred and simulated site specific amino-acid profiles under simulation accounting for long term fluctuation of $\Ne$, mutation rate per generation and generation time.
    Estimation with fluctuating branch $\Ne$ (left panel) or constant $\Ne$ (right panel).}
\end{figure}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        SimuDiv, chain 1 & $2.30 \pm 0.04$ & $2.45 \pm 0.02$\\ \hline
        SimuDiv, chain 2 & $2.30 \pm 0.04$ & $2.45 \pm 0.02$\\ \hline
    \end{tabu}}
    \caption[Inferred amino-acids entropy for SimuDiv]{
    Estimated amino-acids entropy.
    Simulation accounting for long term fluctuation of $\Ne$, mutation rate per generation and generation time.
    Estimated with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}

\subsection{Wright-Fisher with polymorphism}

The evolutionary dynamics was formalized as a Wright-Fisher model with mutation, selection and drift. The population is assumed to be panmictic, with with \gls{effective-population-size} $\Ne$ and with non-overlapping generations.

\begin{center}
    \includegraphics[width=\textwidth] {ModelSimuPoly}
\end{center}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
    \end{minipage} \hfill
    \caption[Inferred branch parameters for SimuPoly]{
    Inferred branch parameters under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
    }
\end{figure}


\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_ProfileCorrelation.png}
    \end{minipage}    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSel_ProfileCorrelation.png}
    \end{minipage}
    \caption[Inferred site amino-acid profiles for SimuPoly]{
    Inferred and simulated site specific amino-acid profiles under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
    Estimation with fluctuating branch $\Ne$ (left panel) or constant $\Ne$ (right panel).}
\end{figure}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        SimuPoly, chain 1 & $2.47 \pm 0.03$ & $2.37 \pm 0.02$\\ \hline
        SimuPoly, chain 2 & $2.47 \pm 0.03$ & $2.37 \pm 0.02$\\ \hline
    \end{tabu}}
    \caption[Inferred amino-acid entropy for SimuPoly]{
    Estimated amino-acids entropy under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
    Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}

\subsection{Fisher geometric landscape}

We simulated \glspl{substitution} in a protein using an adaptation of Fisher's geometric landscape.
In the original context, the \gls{Phenotype} is a vector ($\phenoGeo$) in a multidimensional space, where the number of dimensions is often termed complexity.
From a \gls{Phenotype}, the fitness is a monotonously decreasing function of the \gls{Phenotype} distance to $0$.
The exact functional phenotype-fitness map depend on $2$ external parameters controlling for strength ($\alpha$) and epistasis ($\beta$).
If the phenotype-fitness map is explicit, the genotype-phenotype map is more pervasive.
Mutations are seen has displacement of the \gls{Phenotype} in the multidimensional space.
Beneficial mutations are moving the \gls{Phenotype} closer to $0$, whereas deleterious mutations are moving the \gls{Phenotype} further away.
In such original context, the distribution of mutational effects is not dependent on the current genotype, but this can be relaxed using a genotype-phenotype map.\\

In a protein context, the genotype-phenotype map can be defined by assigning to each of the $20$ amino-acid a vector in the multidimensional space.
Since different sites of the protein do not have the same physico-chimical properties, we can define a specific genotype-phenotype map for each position of the sequence.
Overall, the protein \gls{Phenotype} is computed as the sum of site-specific multidimensional vectors, obtained by accessing the amino-acid present at each site of the protein.
From a \acrshort{DNA} sequence $\mathbb{S}^t$ after $t$ \glspl{substitution}, the protein's \gls{Phenotype} is given by:
\begin{equation}
    \phenoGeo\left(\mathbb{S}^{t}\right) = \sum_{1 \leq \site \leq \Nsite} \phenoGeo_{\site} \left(\mathbb{S}^t(\site) \right),
\end{equation}
where $\phenoGeo_{\site}$ is the genotype-phenotype map at site $\site$.\\

And the Wrightian fitness of $\mathbb{S}^t$ is :
\begin{equation}
    w\left( \phenoGeo\left(\mathbb{S}^{t}\right) \right) = e^{-\alpha \left| \phenoGeo\left(\mathbb{S}^{t}\right) \right|^{\beta}},
\end{equation}
where strength ($\alpha > 0$) and epistasis ($\beta$) are parameters of the fitness function.
\begin{center}
    \includegraphics[width=\textwidth] {ModelSimuGeo}
\end{center}
For each possible mutant ($t+1$ substitutions), we compute $\phenoGeo\left(\mathbb{S}^{t+1}\right)$ from the updated sequence $\mathbb{S}^{t+1}$, and subsequently the selection coefficient of the mutant:
\begin{equation}
    s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right) = \dfrac{ w\left( \phenoGeo\left(\mathbb{S}^{t+1}\right) \right) - w\left( \phenoGeo\left(\mathbb{S}^{t}\right) \right)}{w\left( \phenoGeo\left(\mathbb{S}^{t}\right) \right)}.
\end{equation}
The next change in the protein coding \acrshort{DNA} and the time to next the event is chosen using Gillespie algorithm, according to the rates of \gls{substitution} between \glspl{codon}:
\begin{equation}
{\submatrix_{\itoj}}
    = \mu_{\itoj} \dfrac{4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)} }},
\end{equation}
where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of \glspl{synonymous}.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
    \end{minipage} \hfill
    \caption[Inferred branch parameters for SimuGeo]{
    Inferred branch parameters under simulation accounting for site epistasis (Geometric landscape), thus fluctuation of the selection coefficient along the phylogeny.
    }
\end{figure}


\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        SimuGeo, chain 1 & $2.27 \pm 0.02$ & $2.46 \pm 0.02$\\ \hline
        SimuGeo, chain 2 & $2.23 \pm 0.04$ & $2.46 \pm 0.02$\\ \hline
    \end{tabu}}
    \caption[Entropy of amino-acids for SimuGeo]{
    Estimated amino-acids entropy under under simulation accounting for site epistasis (Geometric landscape), thus fluctuation of the selection coefficient along the phylogeny.
    Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}

\subsection{Protein folding probability}
We simulated \glspl{substitution} in the protein phosphatase ($\Nsite=300$ \gls{codon} sites) as in Goldstein \& Pollock (2017).
From a \acrshort{DNA} sequence $\mathbb{S}^t$ after $t$ \glspl{substitution}, we compute the free energy of the folded state $\Base_{\mathrm{F}}\left(\mathbb{S}^{t}\right)$, using the $3$-dimensional structure of the folded state and pair-wise contact energies between neighboring amino-acid residues:
\begin{equation}
    G_{\mathrm{F}}\left(\mathbb{S}^{t}\right) = \sum_{1 \leq \site \leq \Nsite} \sum_{r \in \mathcal{N}(\site)} I \left(\mathbb{S}^t(\site), \mathbb{S}^t(r) \right),
\end{equation}
where $I(a,b)$ is the pair-wise contact energies between amino-acid $a$ and $b$, using contact potentials estimated by Miya-zawa and Jernigan, and $\mathcal{N}(\site)$ are the neighbor residues of site $\site$ (closer than $7\angstrom$) in the $3$D structure.\\

The free energy of unfolded states $G_{\mathrm{U}}\left(\mathbb{S}^{t}\right)$ is approximated using $55$ decoy $3$D structures that supposedly represent a sample of possible unfolded states:
\begin{equation}
    G_{\mathrm{U}}\left(\mathbb{S}^{t}\right) = \langle G\left(\mathbb{S}^{t}\right) \rangle - kT \ln (1.0\mathrm{E}^{160}) - \dfrac{2 \left[ \langle G\left(\mathbb{S}^{t}\right)^2 \rangle - \langle G\left(\mathbb{S}^{t}\right) \rangle^2\right] }{kT}
\end{equation}
where the average $\langle . \rangle$ runs other the $55$ decoy $3$D structures, and $k$ is the Boltzmann constant and $T$ the temperature in Kelvin.\\

From the energy of folded and unfolded states, we can compute the difference in free energy between the states:
\begin{equation}
    \phenoFold\left(\mathbb{S}^{t}\right) = G_{\mathrm{F}}\left(\mathbb{S}^{t}\right) - G_{\mathrm{U}}\left(\mathbb{S}^{t}\right)
\end{equation}

Wrightian fitness is defined as the probability of our protein to be in the folded state:
\begin{equation}
    w(\phenoFold\left(\mathbb{S}^{t}\right)) = \dfrac{P_{\mathrm{F}}\left(\mathbb{S}^{t}\right)}{P_{\mathrm{F}}\left(\mathbb{S}^{t}\right) + P_{\mathrm{U}}\left(\mathbb{S}^{t}\right)} = \dfrac{e^{-\beta G_{\mathrm{F}}\left(\mathbb{S}^{t}\right) }}{e^{-\beta G_{\mathrm{F}} \left(\mathbb{S}^{t}\right) } + e^{-\beta G_{\mathrm{U}}\left(\mathbb{S}^{t}\right) }} = \dfrac{1}{1 + e^{\beta \phenoFold\left(\mathbb{S}^{t}\right) }},
\end{equation}
where $\beta$ is the inverse of the temperature ($\beta=1/kT$).
\begin{center}
    \includegraphics[width=\textwidth] {ModelSimuFold}
\end{center}
For each possible mutant ($t+1$ substitutions), we compute $\phenoFold^{t+1}$ from the updated sequence $\mathbb{S}^{t+1}$, and subsequently the selection coefficient of the mutant:
\begin{equation}
    s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right) = \dfrac{ w\left( \phenoFold\left(\mathbb{S}^{t+1}\right) \right) - w\left( \phenoFold\left(\mathbb{S}^{t}\right) \right)}{w\left( \phenoFold\left(\mathbb{S}^{t}\right) \right)}.
\end{equation}
The next change in the protein coding \acrshort{DNA} and the time to next the event is chosen using Gillespie algorithm, according to the rates of \gls{substitution} between \glspl{codon}:
\begin{equation}
{\submatrix_{\itoj}}
    = \mu_{\itoj} \dfrac{4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \mathbb{S}^{t},\mathbb{S}^{t+1}\right)} }},
\end{equation}
where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of \glspl{synonymous}.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
    \end{minipage} \hfill
    \caption[Inferred branch parameters for SimuFold]{
    Inferred branch parameters under simulation accounting for site epistasis (folding stability model), thus fluctuation of the selection coefficient along the phylogeny.
    }
\end{figure}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        SimuFold, chain 1 & $1.31 \pm 0.05$ & $1.61 \pm 0.03$\\ \hline
        SimuFold, chain 2 & $1.30 \pm 0.04$ & $1.60 \pm 0.03$\\ \hline
    \end{tabu}}
    \caption[Entropy of amino-acids in SimuFold]{
    Estimated amino-acids entropy under simulation accounting for site epistasis (folding stability model), thus fluctuation of the selection coefficient along the phylogeny.
    Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}


\section{Empirical data in mammals}

\subsection{Chain convergence}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_ProfileCorrelation.png}
    \end{minipage}    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogPopulationSizeCorrelation}
    \end{minipage}
    \caption[Chain convergence of site profiles and branche $\Ne$]{
    Chain convergence of site amino-acid preferences (left panel) and branch $\Ne$ (right panel).}
\end{figure}

\subsection{Traits estimation \& correlation (replicate 1, chain 1)}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{table}[H]
    \input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsCovariance.tex}
    \caption[Covariance matrix in mammals]{
    Covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in mammals]{
    Partial correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogMaximum_longevity}
    \caption[Maximum longevity estimation in mammals]{Maximum longevity estimation in mammals}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogAdult_weight}
    \caption[Adult weight estimation in mammals]{Adult weight estimation in mammals}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogFemale_maturity}
    \caption[Female maturity estimation in mammals]{Female maturity estimation in mammals}
\end{figure}

\subsection{Repeatability of experiments}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-4}
    \end{minipage}
    \caption[Repeatability of mutation rate estimation in mammals]{Repeatability of mutation rate ($\mu$) estimation in mammals}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-4}
    \end{minipage}
    \caption[Repeatability of branch time estimation in mammals]{Repeatability of branch time ($\Delta T$) estimation in mammals}
\end{figure}

\begin{table}[H]
    \input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsCorrelation.tex} \\
    \input{mammals/18CDS_SiteMutSelBranchNe_R2_TraitsCorrelation.tex} \\
    \input{mammals/18CDS_SiteMutSelBranchNe_R3_TraitsCorrelation.tex} \\
    \input{mammals/18CDS_SiteMutSelBranchNe_R4_TraitsCorrelation.tex}
    \caption[Covariance matrix repeatability in mammals]{
    In all four replicates, covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\subsection{Amino-acid preferences entropy}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        Mammals 18 CDS, replicate 1, Chain 1 & $1.07 \pm 0.10$ & $1.14 \pm 0.10$\\ \hline
        Mammals 18 CDS, replicate 2, Chain 2 & $1.07 \pm 0.09$ & $1.14 \pm 0.10$\\ \hline
        Mammals 18 CDS, replicate 2, Chain 1 & $1.06 \pm 0.10$ & $1.12 \pm 0.09$\\ \hline
        Mammals 18 CDS, replicate 2, Chain 2 & $1.06 \pm 0.09$ & $1.11 \pm 0.10$\\ \hline
        Mammals 18 CDS, replicate 3, Chain 1 & $1.08 \pm 0.12$ & $1.15 \pm 0.11$\\ \hline
        Mammals 18 CDS, replicate 3, Chain 2 & $1.04 \pm 0.10$ & $1.18 \pm 0.11$\\ \hline
        Mammals 18 CDS, replicate 4, Chain 1 & $0.94 \pm 0.11$ & $1.02 \pm 0.12$\\ \hline
        Mammals 18 CDS, replicate 4, Chain 2 & $0.89 \pm 0.11$ & $1.02 \pm 0.11$\\ \hline
        \hline
        Mammals 36 CDS, replicate 1, Chain 1 & $1.02 \pm 0.06$ & $1.07 \pm 0.10$\\ \hline
        Mammals 36 CDS, replicate 1, Chain 2 & $0.91 \pm 0.07$ & $1.03 \pm 0.07$\\ \hline
        Mammals 36 CDS, replicate 2, Chain 1 & $0.92 \pm 0.09$ & $0.96 \pm 0.09$\\ \hline
        Mammals 36 CDS, replicate 2, Chain 2 & $1.01 \pm 0.09$ & $1.02 \pm 0.11$\\ \hline
        Mammals 36 CDS, replicate 3, Chain 1 & $0.93 \pm 0.00$ & $1.05 \pm 0.09$\\ \hline
        Mammals 36 CDS, replicate 3, Chain 2 & $1.02 \pm 0.07$ & $1.05 \pm 0.11$\\ \hline
        Mammals 36 CDS, replicate 4, Chain 1 & $1.04 \pm 0.07$ & $1.10 \pm 0.08$\\ \hline
        Mammals 36 CDS, replicate 4, Chain 2 & $1.03 \pm 0.10$ & $1.08 \pm 0.08$\\ \hline
        Mammals 36 CDS, replicate 5, Chain 1 & $1.03 \pm 0.10$ & $1.03 \pm 0.08$\\ \hline
        Mammals 36 CDS, replicate 5, Chain 2 & $0.99 \pm 0.10$ & $1.04 \pm 0.08$\\ \hline
        Mammals 36 CDS, replicate 6, Chain 1 & $1.05 \pm 0.10$ & $1.10 \pm 0.08$\\ \hline
        Mammals 36 CDS, replicate 6, Chain 2 & $0.97 \pm 0.11$ & $1.10 \pm 0.10$\\ \hline
    \end{tabu}}
    \caption[Entropy of amino-acids in mammals]{Estimated amino-acids entropy in mammals.
    Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}

\subsection{Traits estimation with branch \texorpdfstring{$\omega$}{ω} (replicate 1, chain 1)}
Obtained with the model of branch fluctuation of $\dnds$ (without site selection for amino-acid).

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{mammals/18CDS_BranchOmega_R1_LogdNdS}
    \caption[$\dnds$ estimation in mammals]{{Non-synonymous substitution} rate ($\dnds$) estimation in mammals}
\end{figure}

\begin{table}[H]
    \input{mammals/18CDS_BranchOmega_R1_TraitsCorrelation.tex}
    \caption[Correlation coefficient matrix in mammals ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{mammals/18CDS_BranchOmega_R1_TraitsCovariance.tex}
    \caption[Covariance matrix in mammals ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{mammals/18CDS_BranchOmega_R1_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in mammals ($\dnds$)]{
    Partial correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}


\section{Isopods}

\subsection{Traits estimation (replicate 1, chain 1)}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_R1_LogPopulationSize}
    \caption[$\Ne$ estimation in isopods]{Effective population size ($\Ne$) estimation in isopods}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_R1_LogMutationRatePerTime}
    \caption[Mutation rate estimation in isopods]{Mutation rate ($\mu$) estimation in isopods}
\end{figure}

\subsection{Repeatability of experiments}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_Log10BranchLength}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_Log10BranchLength}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_Log10BranchLength}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_Log10BranchLength}
    \end{minipage}
    \caption[Repeatability of branch length estimation in isopods]{Repeatability of branch length ($\branchlength$) estimation in isopods}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_LogPopulationSize}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_LogPopulationSize}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_LogPopulationSize}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_LogPopulationSize}
    \end{minipage}
    \caption[Repeatability of $\Ne$ estimation in isopods]{Repeatability of {effective population size} ($\Ne$) estimation in isopods}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_LogMutationRatePerTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_LogMutationRatePerTime}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_LogMutationRatePerTime}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_LogMutationRatePerTime}
    \end{minipage}
    \caption[Repeatability of $\mu$ estimation in isopods]{Repeatability of mutation rate ($\mu$) estimation in isopods}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_BranchTime}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_BranchTime}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_BranchTime}
    \end{minipage}
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_BranchTime}
    \end{minipage}
    \caption[Repeatability of branch time estimation in isopods]{Repeatability of branch time ($\Delta T$) estimation in isopods}
\end{figure}

\subsection{Statistical test}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eco}
    \caption[$\Ne$ as a function of habitat in isopods]{$\Ne$ as a function of habitat in isopods.}
\end{figure}
\begin{verbatim}
Analysis of Variance Table

Response: PopulationSize
Df Sum Sq Mean Sq F value Pr(>F) 
Habitat 1 2.4899 2.48991 83.2450 < 2.2e-16 ***
Replicate 5 0.6446 0.12892 4.3102 0.0007908 ***
Residuals 389 11.6352 0.02991 
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
\end{verbatim}
\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_pig}
    \caption[$\Ne$ as a function of pigmentation in isopods]{$\Ne$ as a function of pigmentation in isopods}
\end{figure}
\begin{verbatim}
Analysis of Variance Table

Response: PopulationSize
Df Sum Sq Mean Sq F value Pr(>F) 
Pigmentation 2 2.8413 1.42067 48.850 < 2.2e-16 ***
Replicate 5 0.6446 0.12892 4.433 0.0006135 ***
Residuals 388 11.2838 0.02908 
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
\end{verbatim}
\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eye}
    \caption[$\Ne$ as a function of ocular structure in isopods]{$\Ne$ as a function of ocular structure in isopods}
\end{figure}
\begin{verbatim}
Analysis of Variance Table

Response: PopulationSize
Df Sum Sq Mean Sq F value Pr(>F) 
Ocular.structure 2 2.8463 1.42314 48.9571 < 2.2e-16 ***
Replicate 5 0.6446 0.12892 4.4349 0.000611 ***
Residuals 388 11.2789 0.02907 
---
Signif. codes: 0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
\end{verbatim}
\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eco_merged}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_pig_merged}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eye_merged}
    \end{minipage}
    \caption[$\Ne$ as a function of traits in isopods]{$\Ne$ as a function of traits in isopods}
\end{figure}


\section{Primates}

\subsection{Chain convergence}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_ProfileCorrelation.png}
    \end{minipage}    \hfill
    \begin{minipage}{0.49\linewidth}
        \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogPopulationSizeCorrelation}
    \end{minipage}
    \caption[Chain convergence of site profiles and branche $\Ne$]{
    Chain convergence of site amino-acid preferences (left panel) and branch $\Ne$ (right panel).}
\end{figure}

\subsection{Traits estimation (chain 1)}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{table}[H]
    \input{primates/SiteMutSelBranchNe_TraitsCorrelation.tex}
    \caption[Correlation coefficient matrix in primates ($\Ne$)]{
    Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{primates/SiteMutSelBranchNe_TraitsCovariance.tex}
    \caption[Covariance matrix in primates ($\Ne$)]{
    Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{primates/SiteMutSelBranchNe_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in primates ($\Ne$)]{
    Partial correlation coefficient between Neffective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogPopulationSize}
    \caption[$\Ne$ estimation in primates]{Effective population size ($\Ne$) estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogMutationRatePerTime}
    \caption[Mutation rate estimation in primates]{Mutation rate ($\mu$) estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Logmaturity}
    \caption[Female maturity estimation in primates]{Female maturity estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Logmass}
    \caption[Mass estimation in primates]{Mass estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Loglongevity}
    \caption[Longevity estimation in primates]{Longevity estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogpiS}
    \caption[$\pi_{S}$ estimation in primates]{$\pi_{S}$ estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogpiNpiS}
    \caption[$\pi_{N} / \pi_{S}$ estimation in primates]{$\pi_{N} / \pi_{S}$ estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Loggeneration_time}
    \caption[Generation time estimation in primates]{Generation time estimation in primates}
\end{figure}

\subsection{Amino-acid preferences entropy}

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Experiment} & $\langle \entropy \rangle$ (\textbf{branch} $\Ne$) &  $\langle \entropy \rangle$ (\textbf{constant} $\Ne$) \\ \hline
        \hline
        Primates, chain 1 & $1.41 \pm 0.10$ & $1.49 \pm 0.08$\\ \hline
        Primates, chain 2 & $1.40 \pm 0.10$ & $1.48 \pm 0.08$\\ \hline
    \end{tabu}}
    \caption[Entropy of amino-acids in primates]{Estimated amino-acids entropy in primates.
    Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$ (left column), or under the assumption of constant $\Ne$ (right column)}
\end{table}

\subsection{Traits estimation with branch \texorpdfstring{$\omega$}{ω} (chain 1)}
Obtained with the model of branch fluctuation of $\dnds$ (without site selection for amino-acid).

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/BranchOmega_LogOmega}
    \caption[$\dnds$ estimation in primates]{{Non-synonymous substitution} rate ($\dnds$) estimation in primates}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{primates/BranchOmega_LogMutationRatePerTime}
    \caption[$\mu$ estimation in primates]{Mutation rate ($\mu$) estimation in primates}
\end{figure}

\begin{table}[H]
    \input{primates/BranchOmega_TraitsCorrelation.tex}
    \caption[Correlation coefficient matrix in primates ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{primates/BranchOmega_TraitsCovariance.tex}
    \caption[Covariance matrix in primates ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{primates/BranchOmega_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in primates ($\dnds$)]{
    Partial correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in placental mammals.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}


\section{Empirical data in drosophila}

\subsection{Traits estimation \& correlation (replicate 1, chain 1)}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{table}[H]
    \input{drosophila/18CDS_SiteMutSelBranchNe_R1_TraitsCorrelation.tex}
    \caption[Correlation coefficient matrix in drosophila ($\dnds$)]{
    Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{drosophila/18CDS_SiteMutSelBranchNe_R1_TraitsCovariance.tex}
    \caption[Covariance matrix in drosophila]{
    Covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{drosophila/18CDS_SiteMutSelBranchNe_R1_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in drosophila]{
    Partial correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_R1_LogPopulationSize}
    \caption[Effective population size estimation in drosophila]{Effective population size ($\Ne$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_R1_LogMutationRatePerTime}
    \caption[Mutation rate estimation in drosophila]{Mutation rate ($\mu$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_R1_LogGenomeSize}
    \caption[Genome size estimation in drosophila]{Genome size estimation in drosophila}
\end{figure}

\subsection{Repeatability of experiments}
Obtained with the inference model of site selection for amino-acid, and branch fluctuation of $\Ne$.

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-4}
    \end{minipage}
    \caption[Repeatability of branch length estimation in drosophila]{Repeatability of branch length ($\branchlength$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-4}
    \end{minipage}
    \caption[Repeatability of \gls{effective-population-size} estimation in drosophila]{Repeatability of \gls{effective-population-size} ($\Ne$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-4}
    \end{minipage}
    \caption[Repeatability of mutation rate estimation in drosophila]{Repeatability of mutation rate ($\mu$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-2}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-3}
    \end{minipage}    \hfill
    \begin{minipage}{0.32\linewidth}
        \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-4}
    \end{minipage}
    \caption[Repeatability of branch time estimation in drosophila]{Repeatability of branch time ($\Delta T$) estimation in drosophila}
\end{figure}

\begin{table}[H]
    \input{drosophila/18CDS_SiteMutSelBranchNe_R1_TraitsCorrelation.tex} \\
    \input{drosophila/18CDS_SiteMutSelBranchNe_R2_TraitsCorrelation.tex} \\
    \input{drosophila/18CDS_SiteMutSelBranchNe_R3_TraitsCorrelation.tex} \\
    \input{drosophila/18CDS_SiteMutSelBranchNe_R4_TraitsCorrelation.tex}
    \caption[Covariance matrix repeatability in drosophila]{
    In all four replicates, covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and genome size drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\subsection{Traits estimation with branch \texorpdfstring{$\omega$}{ω} (replicate 1, chain 1)}
Obtained with the model of branch fluctuation of $\dnds$ (without site selection for amino-acid).

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_BranchOmega_R1_LogdNdS}
    \caption[Non-synonymous \gls{substitution} rate estimation in drosophila]{Non-synonymous \gls{substitution} rate ($\dnds$) estimation in drosophila}
\end{figure}

\begin{figure}[H]
    \centering
    \includegraphics[width=\linewidth, page=1]{drosophila/18CDS_BranchOmega_R1_LogMutationRatePerTime}
    \caption[Mutation rate estimation in drosophila]{Mutation rate ($\mu$) estimation in drosophila}
\end{figure}


\begin{table}[H]
    \input{drosophila/18CDS_BranchOmega_R1_TraitsCorrelation.tex}
    \caption[Correlation coefficient matrix in drosophila ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{drosophila/18CDS_BranchOmega_R1_TraitsCovariance.tex}
    \caption[Covariance matrix in drosophila ($\dnds$)]{
    Correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}

\begin{table}[H]
    \input{drosophila/18CDS_BranchOmega_R1_TraitsPartialCorrelation.tex}
    \caption[Partial correlation coefficient matrix in drosophila ($\dnds$)]{
    Partial correlation coefficient between Non-synonymous \gls{substitution} rate~($\dnds$), mutation rate per site per unit of time~($\mu$), and life-history traits (Maximum longevity, adult weight and female maturity) were computed in drosophila.
    Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
\end{table}


\section{Sufficient statistics}

A sequence of length $\Nsite$ evolves by point \glspl{substitution}, according to a random process defined by the \gls{substitution} matrices $\Submatrix\branchsiteexp$, over a phylogenetic tree.
A realization of the random process along a branch $\branch$, and at a particular site $\site$ results in a detailed \gls{substitution} history, which will be denoted by $\subhistory\branchsiteexp$.

\subsubsection{Path sufficient statistics}
All sites owning to the same category of fitness profile share the same \gls{substitution} rate matrix.
Hence, $\subhistory\branchsiteexp$ can be gathered across all sites owing to a specific category $\cat$, denoted $\subhistory\branchexp$.
If we express the probability of the \gls{substitution} mapping ($\subhistory\branchcatexp$) as a function of the \gls{codon} \gls{substitution} process for this category $\cat$, we get the following expression:
\begin{equation}
    \label{eq:PathSuffStat}
    p(\subhistory\branchcatexp | \branchlength\branchexp, \Submatrix\branchcatexp) \propto \prod_{\iSetCodon} \left[\subequi_{\ci}\branchcatexp\right]^{n_{\ci}\branchcatexp} \prod_{ \ijSetCodon} \left[\submatrix\branchcatexp_{\itoj}\right]^{m_{\ci, \cj}\branchcatexp} \prod_{\iSetCodon} \e^{- \left| \submatrix\branchcatexp_{\ci, \ci}\right| a_{\ci}\branchcatexp}
\end{equation}
where we define the sufficient statistics:
\begin{itemize}
    \setlength\itemsep{-0.25em}
    \item $m_{\ci, \cj}\branchcatexp$ is the total number of \glspl{substitution} from \gls{codon} $\ci$ to \gls{codon} $\cj$
    \item $n_{\ci}\branchcatexp$ is the number of sites starting with \gls{codon} $\ci$ at the tip of the branch.
    \item $a_{\ci}\branchcatexp$ is the total waiting time in \gls{codon} $\ci$.
\end{itemize}
Once these sufficient statistics have been computed, the parameters of the \gls{substitution} matrix $\Submatrix\branchcatexp$ can be resampled conditional on $\subhistory\branchcatexp$,
using equation~\ref{eq:PathSuffStat} each time the \gls{likelihood} needs to be recomputed. This leads to relatively fast \acrshort{MCMC} strategy.

\subsubsection{Length sufficient statistics}
$\subhistory\branchsiteexp$ can also be gathered across all sites along a specific branch, giving $\subhistory\branchexp$.
Then the probability of the \gls{substitution} history given the branch lengths ($\branchlength\branchexp = \mu\branchexp \branchtime\branchexp$), take a very simple form:
\begin{equation}
    \label{eq:LengthSuffStat}
    p(\subhistory\branchexp | L\branchexp) \propto \left[ L\branchexp\right]^{u\branchexp} \e^{- r\branchexp L\branchexp}
\end{equation}
where we define the sufficient statistics:
\begin{itemize}
    \setlength\itemsep{-0.25em}
    \item $u\branchexp$ is the total number of \glspl{substitution} over branch $\branch$, summed over all sites.
    \item $r\branchexp$ is the mean rate away from current \gls{codon} state (averaged over the entire \gls{substitution} history).
\end{itemize}
Thus, formally, the probability of the \gls{substitution} mapping can be summarized by saying that the total number of \glspl{substitution} along a given branch over all sites, $u\branchexp$, is Poisson distributed, of mean $r\branchexp L\branchexp$.

\subsubsection{Scatter sufficient statistics}
From the independent contrast $\contrast\branchexp$ of the Brownian process $\Brownian\nodeexp$, we can define the $2 \times 2$ scatter sufficient statistic matrix, $\Scattermatrix$ as:
\begin{equation}
    \Scattermatrix = \sum_{b=1}^{\Nbranch} \contrast\branchexp \cdot \left[\contrast\branchexp\right]^{\mathrm{T}}
\end{equation}
By Bayes theorem, the \gls{posterior} on $\Sigma$, conditional on a particular realization of $\brownian$ (and thus of $\contrast$) is an invert Wishart distribution, of parameter $\covariancekappa \Identitymatrix + \Scattermatrix$ and with $\Nbranch + 3$ degrees of freedom.
\begin{equation}
    \CovarianceMatrix \sim \mathrm{Wishart}^{-1}\left( \covariancekappa \Identitymatrix + \Scattermatrix, \Nbranch + 3\right)
\end{equation}
This invert Wishart distribution can be obtained by sampling $\Nbranch + 3$ independent and identically distributed multivariate normal random variables $\Multivariate\indiceexp$ defined by
\begin{equation}
    \Multivariate\indiceexp \sim \mathcal{N} \left( \bm{0}, \left[\covariancekappa \Identitymatrix + \Scattermatrix\right]^{-1} \right).
\end{equation}
And from these multivariate samples, $\CovarianceMatrix$ is Gibbs sampled as:
\begin{equation}
    \CovarianceMatrix = \left( \sum_{k=1}^{\Nbranch + 3} \Multivariate\indiceexp \cdot \left[\Multivariate\indiceexp \right] ^{\mathrm{T}} \right)^{-1}
\end{equation}
