\documentclass{article}
\usepackage{xcolor}
\definecolor{BLUELINK}{HTML}{0645AD}
\definecolor{DARKBLUELINK}{HTML}{0B0080}
\definecolor{LIGHTBLUELINK}{HTML}{3366BB}
\definecolor{PURPLELINK}{HTML}{663366}
\PassOptionsToPackage{hyphens}{url}
\usepackage[colorlinks=false ]{hyperref}
% for linking between references, figures, TOC, etc in the pdf document
\hypersetup{colorlinks,
linkcolor=DARKBLUELINK,
anchorcolor=DARKBLUELINK,
citecolor=DARKBLUELINK,
filecolor=DARKBLUELINK,
menucolor=DARKBLUELINK,
urlcolor=BLUELINK
} % Color citation links in purple
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{naturalnames}{hyperref}

\usepackage[margin=60pt]{geometry}
\usepackage{amssymb,amsfonts,amsmath,amsthm,mathtools}
\usepackage{lmodern}
\usepackage{bm,bbold}
\usepackage{verbatim}
\usepackage{float}
\usepackage{bm}
\usepackage{listings, enumerate, enumitem}
\usepackage[export]{adjustbox}
\usepackage{tabu}
\tabulinesep=0.6mm
\newcommand\cellwidth{\TX@col@width}
\usepackage{hhline}
\setlength{\arrayrulewidth}{1.2pt}
\usepackage{multicol,multirow,array}
\usepackage{etoolbox}
\AtBeginEnvironment{tabu}{\footnotesize}
\usepackage{booktabs}

\usepackage{graphicx}
\graphicspath{{artworks/}}
\makeatletter
\def\input@path{{artworks/}}
\makeatother
\pdfstringdefDisableCommands{%
\renewcommand*{\bm}[1]{#1}%
% any other necessary redefinitions
}
\usepackage{xfrac, nicefrac}
\usepackage{lmodern}
\usepackage{natbib}
\pdfinclusioncopyfonts=1
\input{notations}


\begin{document}
	\part*{Supplementary materials}
	\tableofcontents
	\section{Summary statistics}
	\label{sec:summary-statistics}

	\subsection{Partial correlation coefficient}
	\label{subsec:partial-correlation-coefficient}

	The correlation coefficient $\rho_{\traiti, \traitj}$ give the total regression between two variables.
	Partial-correlation coefficient account for the entire covariance matrix, and measure the correlation between $2$ traits, knowing the values of all the other traits:
	\begin{equation}
		\rho_{\traiti, \traitj | c \in \traitInterval \setminus \{ a, b\} } = - \dfrac{\Precisionmatrix_{\traiti, \traitj}}{\sqrt{\Precisionmatrix_{\traiti, \traiti} \Precisionmatrix_{\traitj, \traitj}}},
	\end{equation}
	where the precision matrix $\PrecisionMatrix$ is the inverse of the covariance matrix:
	\begin{equation}
		\PrecisionMatrix = \CovarianceMatrix^{-1}
	\end{equation}

	\subsection{Fitness profile entropy}
	\label{subsec:fitness-profile-entropy}

	For a category $\cat$, the Shannon's entropy ($\entropy$) of the fitness profile ($\Profile$) is defined as:
	\begin{equation}
		\entropy\catexp = - \sum\limits\sumSetAa \profile\catexp_{\aminoacid} \ln \left( \profile\catexp_{\aminoacid} \right)
	\end{equation}
	The Shannon's entropy measures the flatness of the fitness profile, with a value of $0$ corresponding to a single peak fitness landscape (only one amino acid is present), and a value of $log(20)\simeq3$ corresponding to a {neutral} landscape, where each amino acid has the same fitness.

	The Shannon's entropy can be averaged over all sites as:
	\begin{equation}
		\left\langle \entropy \right\rangle = \dfrac{1}{\Nsite}\sum\limits\sumSetsite \entropy^{\catsite}
	\end{equation}


	\section{Simulations}
	\label{sec:supp-mat-simulations}

	\subsection{Site-specific fitness profiles (\texttt{SimuDiv})}
	\label{subsec:independent-fitness-profiles}
	For simulations under a site-independent fitness landscape, with site-specific fitness profiles, the protein log-fitness is computed as the sum of amino-acid log-fitness coefficients along the sequence.
	In this model, for each {codon} site $\site$, we assign a fitness profile, denoted $\Profile\siteexp = \{ \profile\siteexp_{\aminoacid},\ 1 \leq \aminoacid \leq 20 \}$, a vector of 20 amino-acid scaled (Wrightian) fitness coefficients.
	Since $\Seqi[ \site ]$ is the {codon} at site $\site$, the encoded amino acid is $\aaMap \left( \Seqi[ \site ] \right)$, hence the fitness at site $\site$ is $\Profile\siteexp_{\aaMap \left( \Seqi[ \site ] \right)}$.
	Altogether, the selection coefficient of the mutant $\Seqj$ is:
	\begin{equation}
		s \left( \Seqi,\Seqj\right) = \sum\limits\sumSetsite \ln \left( \dfrac{\Profile\siteexp_{\aaMap \left( \Seqj[ \site ] \right)}}{\Profile\siteexp_{\aaMap \left( \Seqi[ \site ] \right)}} \right),
	\end{equation}
	The fitness vectors $\Profile\siteexp$ used in this study are extracted from \citet{Bloom2017}, which were experimentally determined by deep mutational scanning for 498 codon sites of the nucleoprotein in \textit{Influenza Virus} strains (as human host).
	Although the \textit{Influenza Virus} is phylogenetically far away from vertebrates, the experimentally determined profiles in these strains capture the structural constrains exerted on proteins.
	For each {codon} site $\site$ of our simulation, we assign randomly one the 498 fitness profile (sampling with replacement) experimentally determined, which altogether determines the selection coefficient for any non-synonymous mutation.
	\begin{center}
		\includegraphics[width=\textwidth] {ModelSimuDiv}
	\end{center}

	The next change in the protein coding {DNA} and the time to next the event is chosen using Gillespie's algorithm~\citep{Gillespie1977}, according to the rates of {substitution} between codons:
	\begin{equation}
	{\submatrix_{\itoj}}
		= \mu_{\itoj} \dfrac{4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)} }},
	\end{equation}
	where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of {synonymous} substitutions.


	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
		\end{minipage} \hfill
		\caption[Inferred branch parameters for \texttt{SimuDiv}]{
		Inferred branch parameters under simulations accounting for site-specific amino-acid profiles, long term fluctuation of $\Ne$, mutation rate per generation and generation time.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.
		}
	\end{figure}


	\begin{figure}[H]
		\centering
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSelBranchNe_ProfileCorrelation.png}
		\end{minipage} \hfill
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/BranchWise_SimuDiv_SiteMutSel_ProfileCorrelation.png}
		\end{minipage}
		\caption[Inferred site amino-acid profiles for \texttt{SimuDiv}]{
		Inferred and simulated site-specific amino-acid profiles under simulation accounting for long term fluctuation of $\Ne$, mutation rate per generation and generation time.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left panel), or under the assumption of constant $\Ne$ (in the right panel).
		}
	\end{figure}

	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			\texttt{SimuDiv}, chain~1 & $2.30 \pm 0.04$ & $2.45 \pm 0.02$\\ \hline
			\texttt{SimuDiv}, chain 2 & $2.30 \pm 0.04$ & $2.45 \pm 0.02$\\ \hline
		\end{tabu}}
		\caption[Inferred amino-acids entropy for \texttt{SimuDiv}]{
		Estimated amino-acid entropy under simulations accounting for long term fluctuation of $\Ne$, mutation rate per generation and generation time.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).
		}
	\end{table}

	\subsection{Wright-Fisher with polymorphism (\texttt{SimuPoly})}
	\label{subsec:wright-fisher-with-polymorphism}

	The evolutionary dynamics was formalized as a Wright-Fisher model with mutation, selection and drift
	The population is assumed to be panmictic, with {effective population size} $\Ne$ and with non-overlapping generations.

	\begin{center}
		\includegraphics[width=\textwidth] {ModelSimuPoly}
	\end{center}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
		\end{minipage} \hfill
		\caption[Inferred branch parameters for \texttt{SimuPoly}]{
		Inferred branch parameters under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.
		}
	\end{figure}


	\begin{figure}[H]
		\centering
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSelBranchNe_ProfileCorrelation.png}
		\end{minipage} \hfill
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuPoly_SiteMutSel_ProfileCorrelation.png}
		\end{minipage}
		\caption[Inferred site amino-acid profiles for \texttt{SimuPoly}]{
		Inferred and simulated site-specific amino-acid profiles under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left panel), or under the assumption of constant $\Ne$ (in the right panel).}
	\end{figure}

	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			\texttt{SimuPoly}, chain~1 & $2.47 \pm 0.03$ & $2.37 \pm 0.02$\\ \hline
			\texttt{SimuPoly}, chain 2 & $2.47 \pm 0.03$ & $2.37 \pm 0.02$\\ \hline
		\end{tabu}}
		\caption[Inferred amino-acid entropy for \texttt{SimuPoly}]{
		Estimated amino-acid entropy under simulation accounting for finite population effects, site linkage and short term fluctuation of $\Ne$.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).}
	\end{table}

	\subsection{Fisher geometric landscape (\texttt{SimuGeo})}
	\label{subsec:fisher-geometric-landscape}

	We simulated substitutions in a protein using an adaptation of Fisher's geometric landscape~\citep{Tenaillon2014,Blanquart2016}.
	In the original context, the {phenotype} is a vector ($\phenoGeo$) in a multidimensional space, where the number of dimensions is often termed complexity.
	From a {phenotype}, the fitness is a monotonously decreasing function of the {phenotype} distance to $0$.
	The exact functional phenotype-fitness map depends on $2$ external parameters controlling for strength ($\alpha$) and epistasis ($\beta$).
	If the phenotype-fitness map is explicit, the genotype-phenotype map is more pervasive.
	Mutations are seen has displacement of the {phenotype} in the multidimensional space.
	Beneficial mutations are moving the {phenotype} closer to $0$, whereas deleterious mutations are moving the {phenotype} further away.
	In such original context, the distribution of mutational effects is not dependent on the current genotype, but this can be relaxed using a genotype-phenotype map.\\

	In a protein context, the genotype-phenotype map can be defined by assigning to each of the $20$ amino acid a vector in the multidimensional space.
	Since different sites of the protein do not have the same physico-chimical properties, we can define a specific genotype-phenotype map for each position of the sequence.
	Overall, the protein {phenotype} is computed as the sum of site-specific multidimensional vectors, obtained by accessing the amino acid present at each site of the protein.
	From a {DNA} sequence $\Seqi^t$ after $t$ substitutions, the protein's {phenotype} is given by:
	\begin{equation}
		\phenoGeo\left(\Seqi^{t}\right) = \sum\limits\sumSetsite \phenoGeo_{\site} \left(\Seqi^t(\site) \right),
	\end{equation}
	where $\phenoGeo_{\site}$ is the genotype-phenotype map at site $\site$.\\

	And the Wrightian fitness of $\Seqi^t$ is :
	\begin{equation}
		\wrightfit\left( \phenoGeo\left(\Seqi^{t}\right) \right) = e^{-\alpha \left| \phenoGeo\left(\Seqi^{t}\right) \right|^{\beta}},
	\end{equation}
	where strength ($\alpha > 0$) and epistasis ($\beta$) are parameters of the fitness function.
	\begin{center}
		\includegraphics[width=\textwidth] {ModelSimuGeo}
	\end{center}
	For each possible mutant (at time $t+1$ substitutions), we compute $\phenoGeo\left(\Seqi^{t+1}\right)$ from the updated sequence $\Seqi^{t+1}$, and subsequently the selection coefficient of the mutant:
	\begin{equation}
		s \left( \Seqi^{t},\Seqi^{t+1}\right) = \dfrac{\wrightfit\left( \phenoGeo\left(\Seqi^{t+1}\right) \right) -\wrightfit\left( \phenoGeo\left(\Seqi^{t}\right) \right)}{\wrightfit\left( \phenoGeo\left(\Seqi^{t}\right) \right)}.
	\end{equation}
	The next change in the protein coding {DNA} and the time to next the event is chosen using Gillespie's algorithm~\citep{Gillespie1977}, according to the rates of {substitution} between codons:
	\begin{equation}
	{\submatrix_{\itoj}}
		= \mu_{\itoj} \dfrac{4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)} }},
	\end{equation}
	where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of {synonymous} substitutions.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuGeo_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
		\end{minipage} \hfill
		\caption[Inferred branch parameters for \texttt{SimuGeo}]{
		Inferred branch parameters under simulation accounting for site epistasis in geometric landscape, thus fluctuation of the selection coefficient along the phylogeny.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.
		}
	\end{figure}


	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			\texttt{SimuGeo}, chain~1 & $2.27 \pm 0.02$ & $2.46 \pm 0.02$\\ \hline
			\texttt{SimuGeo}, chain 2 & $2.23 \pm 0.04$ & $2.46 \pm 0.02$\\ \hline
		\end{tabu}}
		\caption[Amino-acid entropy for \texttt{SimuGeo}]{
		Estimated amino-acid entropy under simulation accounting for site epistasis (geometric landscape), thus fluctuation of the selection coefficient along the phylogeny.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).
		}
	\end{table}

	\subsection{Protein folding probability (\texttt{SimuFold})}
	\label{subsec:protein-folding-probability}

	We simulated substitutions in the protein phosphatase ($\Nsite=300$ {codon} sites) as in \citet{Goldstein2017}.
	From a {DNA} sequence $\Seqi^t$ after $t$ substitutions, we compute the free energy of the folded state $ \GFold \left(\Seqi^{t}\right)$, using the 3-dimensional structure of the folded state and pair-wise contact energies between neighboring amino-acid residues:
	\begin{equation}
		\GFold \left(\Seqi^{t}\right) = \sum\limits\sumSetsite \sum\limits_{r \in \mathcal{V}(\site)} I \left(\Seqi^t(\site), \Seqi^t(r) \right),
	\end{equation}
	where $I(a,b)$ is the pair-wise contact energies between amino acid $a$ and $b$, using contact potentials estimated by \citet{Miyazawa1985}, and $\mathcal{V}(\site)$ are the neighbor residues of site $\site$ (closer than $7\angstrom$) in the 3D structure.\\

	The free energy of unfolded states $\GUnfold \left(\Seqi^{t}\right)$ is approximated using $55$ decoy 3D structures that supposedly represent a sample of possible unfolded states:
	\begin{equation}
		\GUnfold \left(\Seqi^{t}\right) = \left\langle G\left(\Seqi^{t}\right) \right\rangle - kT \ln (1.0\text{E}^{160}) - \dfrac{2 \left[ \left\langle G\left(\Seqi^{t}\right)^2 \right\rangle - \left\langle G\left(\Seqi^{t}\right) \right\rangle^2\right] }{kT}
	\end{equation}
	where the average $\left\langle . \right\rangle$ runs other the $55$ decoy 3D structures, and $k$ is the Boltzmann constant and $T$ the temperature in Kelvin.\\

	From the energy of folded and unfolded states, we can compute the difference in free energy between the states:
	\begin{equation}
		\phenoFold\left(\Seqi^{t}\right) = \GFold \left(\Seqi^{t}\right) - \GUnfold \left(\Seqi^{t}\right)
	\end{equation}

	Wrightian fitness is defined as the probability of our protein to be in the folded state:
	\begin{equation}
		\wrightfit \left( \phenoFold\left(\Seqi^{t}\right) \right) = \probaFold \left(\Seqi^{t}\right) = \dfrac{e^{-\beta \GFold \left(\Seqi^{t}\right) }}{e^{-\beta \GFold  \left(\Seqi^{t}\right) } + e^{-\beta \GUnfold \left(\Seqi^{t}\right) }} = \dfrac{1}{1 + e^{\beta \phenoFold\left(\Seqi^{t}\right) }},
	\end{equation}
	where $\beta$ is the inverse of the temperature ($\beta=1/kT$).
	\begin{center}
		\includegraphics[width=\textwidth] {ModelSimuFold}
	\end{center}
	For each possible mutant (at time $t+1$ substitutions), we compute $\phenoFold^{t+1}$ from the updated sequence $\Seqi^{t+1}$, and subsequently the selection coefficient of the mutant:
	\begin{equation}
		s \left( \Seqi^{t},\Seqi^{t+1}\right) = \dfrac{\wrightfit\left( \phenoFold\left(\Seqi^{t+1}\right) \right) -\wrightfit\left( \phenoFold\left(\Seqi^{t}\right) \right)}{\wrightfit\left( \phenoFold\left(\Seqi^{t}\right) \right)}.
	\end{equation}
	The next change in the protein coding {DNA} and the time to next the event is chosen using Gillespie's algorithm~\citep{Gillespie1977}, according to the rates of {substitution} between codons:
	\begin{equation}
	{\submatrix_{\itoj}}
		= \mu_{\itoj} \dfrac{4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)}{{1 - \e^{-4 \Ne s \left( \Seqi^{t},\Seqi^{t+1}\right)} }},
	\end{equation}
	where ${\submatrix_{\itoj}} = \mu_{\itoj}$ in the case of {synonymous} substitutions.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{simulations/SimuFold_SiteMutSelBranchNe_BranchCorrelation_ContrastPopulationSize}
		\end{minipage} \hfill
		\caption[Inferred branch parameters for \texttt{SimuFold}]{
		Inferred branch parameters under simulation accounting for site epistasis (folding stability model), thus fluctuation of the selection coefficient along the phylogeny.
		Estimation is obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.
		}
	\end{figure}

	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			\texttt{SimuFold}, chain~1 & $1.31 \pm 0.05$ & $1.61 \pm 0.03$\\ \hline
			\texttt{SimuFold}, chain 2 & $1.30 \pm 0.04$ & $1.60 \pm 0.03$\\ \hline
		\end{tabu}}
		\caption[Amino-acid entropy in \texttt{SimuFold}]{
		Estimated amino-acid entropy under simulation accounting for site epistasis (folding stability model), thus fluctuation of the selection coefficient along the phylogeny.
		Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).}
	\end{table}

	\subsection{Identifiability of $\Ne$ and $\mu$}
	\label{subsec:simulation-identifiability}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{simulations/identifiability}
		\caption[Identifiability of $\Ne$ and $\mu$]{
			Some parameters might be strongly correlated or only weakly identifiable, which may make it hard to fit the model to realistically sized datasets.
			To test whether $\Ne$ and $\mu$ are identifiable, for each branch of the tree we draw a 2-D scatter plot for $\Ne$ and $\mu$, where each point is a step in the MCMC procedure.
			We subsequently fit a linear regression and compute the coefficient of determination ($r^2$) for each branch of the tree.
			For a given MCMC, the distribution of $r^2$ across all branches is then represented as a violin plot.
			The distribution of $r^2$ is relatively low, below 0.01 on average, providing confidence that $\Ne$ and $\mu$ are indeed identifiable.}
	\end{figure}


	\section{Empirical data in mammals}
	\label{sec:empirical-data-in-mammals}

	\subsection{Chain convergence}
	\label{subsec:chain-convergence}
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_ProfileCorrelation.png}
		\end{minipage} \hfill
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogPopulationSizeCorrelation}
		\end{minipage}
		\caption[Chain convergence of site profiles and branche $\Ne$]{
		Chain convergence of site amino-acid preferences (left panel) and branch $\Ne$ (right panel).}
	\end{figure}

	\subsection{Traits estimation \& correlation (replicate~1, chain~1)}
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.

	\begin{table}[H]
		\input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsCovariance.tex}
		\caption[Covariance matrix in mammals]{
		Covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsPartialCorrelation.tex}
		\caption[Partial correlation coefficient matrix in mammals]{
		Partial correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
		\label{tab:table-partcor-mammals}
	\end{table}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogPopulationSize}
		\caption[$\Ne$ estimation in mammals]{Effective population size ($\Ne$) estimation in mammals}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogMutationRatePerTime}
		\caption[Mutation rate estimation in mammals]{Mutation rate ($\mu$) estimation in mammals}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogMaximum_longevity}
		\caption[Maximum longevity estimation in mammals]{Maximum longevity estimation in mammals}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogAdult_weight}
		\caption[Adult weight estimation in mammals]{Adult weight estimation in mammals}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_R1_LogFemale_maturity}
		\caption[Female maturity estimation in mammals]{Female maturity estimation in mammals}
	\end{figure}

	\subsection{Repeatability of experiments}
	\label{subsec:repeatability-of-experiments-mammals}

	$4$ independent inferences were performed on a randomly chosen set of $18$ coding sequences ({CDS}) out of $226$.
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.
	Each plot is a correlation between a pair of experiments for a given parameter.
	For each node (or branch) of the tree, the mean {posterior} of the parameter over the {MCMC} (after burn-in) is represented in blue dots, green solid lines are the $90\%$ confidence interval of the {MCMC}.
	Solid red line is the regression line between replicates.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-2}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-3}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_Log10BranchLength-1-4}
		\end{minipage}
		\caption[Repeatability of branch length estimation in mammals]{Repeatability of branch length ($\branchlength$) estimation in mammals}
	\end{figure}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-2}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-3}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize-1-4}
		\end{minipage}
		\caption[Repeatability of $\Ne$ estimation in mammals]{Repeatability of {effective population size} ($\Ne$) estimation in mammals}
	\end{figure}

	\begin{table}[htbp]
		\renewcommand{\arraystretch}{0.5}
		\scriptsize \centering
		\input{mammals/18CDS_SiteMutSelBranchNe_Rep_LogPopulationSize.tex}
		\caption[Repeatability of $\Ne$ estimation in mammals]{Repeatability of {effective population size} ($\Ne$) estimation in mammals, for the extant taxa.}
		\label{tab:Ne-estimated-mammals}
	\end{table}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-2}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-3}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime-1-4}
		\end{minipage}
		\caption[Repeatability of mutation rate estimation in mammals]{Repeatability of mutation rate ($\mu$) estimation in mammals}
	\end{figure}

	\begin{table}[htbp]
		\renewcommand{\arraystretch}{0.5}
		\scriptsize \centering
		\input{mammals/18CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime.tex}
		\caption[Repeatability of mutation rate estimation in mammals]{Repeatability of mutation rate ($\mu$) estimation in mammals, for the extant taxa.}
	\end{table}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-2}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-3}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{mammals/18CDS_SiteMutSelBranchNe_Rep_BranchTime-1-4}
		\end{minipage}
		\caption[Repeatability of branch time estimation in mammals]{Repeatability of branch time ($\Delta T$) estimation in mammals}
	\end{figure}

	\begin{table}[H]
		\input{mammals/18CDS_SiteMutSelBranchNe_R1_TraitsCorrelation.tex} \\
		\input{mammals/18CDS_SiteMutSelBranchNe_R2_TraitsCorrelation.tex} \\
		\input{mammals/18CDS_SiteMutSelBranchNe_R3_TraitsCorrelation.tex} \\
		\input{mammals/18CDS_SiteMutSelBranchNe_R4_TraitsCorrelation.tex}
		\caption[Covariance matrix repeatability in mammals]{
		In all four replicates, covariance coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\subsection{Amino-acid preferences entropy}

	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			Mammals 18 {CDS}, replicate~1, Chain 1 & $1.07 \pm 0.10$ & $1.14 \pm 0.10$\\ \hline
			Mammals 18 {CDS}, replicate 2, Chain 2 & $1.07 \pm 0.09$ & $1.14 \pm 0.10$\\ \hline
			Mammals 18 {CDS}, replicate 2, Chain 1 & $1.06 \pm 0.10$ & $1.12 \pm 0.09$\\ \hline
			Mammals 18 {CDS}, replicate 2, Chain 2 & $1.06 \pm 0.09$ & $1.11 \pm 0.10$\\ \hline
			Mammals 18 {CDS}, replicate 3, Chain 1 & $1.08 \pm 0.12$ & $1.15 \pm 0.11$\\ \hline
			Mammals 18 {CDS}, replicate 3, Chain 2 & $1.04 \pm 0.10$ & $1.18 \pm 0.11$\\ \hline
			Mammals 18 {CDS}, replicate 4, Chain 1 & $0.94 \pm 0.11$ & $1.02 \pm 0.12$\\ \hline
			Mammals 18 {CDS}, replicate 4, Chain 2 & $0.89 \pm 0.11$ & $1.02 \pm 0.11$\\ \hline
			\hline
			Mammals 36 {CDS}, replicate~1, Chain 1 & $1.02 \pm 0.06$ & $1.07 \pm 0.10$\\ \hline
			Mammals 36 {CDS}, replicate~1, Chain 2 & $0.91 \pm 0.07$ & $1.03 \pm 0.07$\\ \hline
			Mammals 36 {CDS}, replicate 2, Chain 1 & $0.92 \pm 0.09$ & $0.96 \pm 0.09$\\ \hline
			Mammals 36 {CDS}, replicate 2, Chain 2 & $1.01 \pm 0.09$ & $1.02 \pm 0.11$\\ \hline
			Mammals 36 {CDS}, replicate 3, Chain 1 & $0.93 \pm 0.00$ & $1.05 \pm 0.09$\\ \hline
			Mammals 36 {CDS}, replicate 3, Chain 2 & $1.02 \pm 0.07$ & $1.05 \pm 0.11$\\ \hline
			Mammals 36 {CDS}, replicate 4, Chain 1 & $1.04 \pm 0.07$ & $1.10 \pm 0.08$\\ \hline
			Mammals 36 {CDS}, replicate 4, Chain 2 & $1.03 \pm 0.10$ & $1.08 \pm 0.08$\\ \hline
			Mammals 36 {CDS}, replicate 5, Chain 1 & $1.03 \pm 0.10$ & $1.03 \pm 0.08$\\ \hline
			Mammals 36 {CDS}, replicate 5, Chain 2 & $0.99 \pm 0.10$ & $1.04 \pm 0.08$\\ \hline
			Mammals 36 {CDS}, replicate 6, Chain 1 & $1.05 \pm 0.10$ & $1.10 \pm 0.08$\\ \hline
			Mammals 36 {CDS}, replicate 6, Chain 2 & $0.97 \pm 0.11$ & $1.10 \pm 0.10$\\ \hline
		\end{tabu}}
		\caption[Entropy of amino acids in mammals]{Estimated amino-acid entropy in mammals.
		Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).}
		\label{tab:table-entropy-aa-mutselne}
	\end{table}

	\subsection{Identifiability of $\Ne$ and $\mu$}
	\label{subsec:mammals-identifiability}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/identifiability}
		\caption[Identibiblatexfiability of $\Ne$ and $\mu$]{
			Some parameters might be strongly correlated or only weakly identifiable, which may make it hard to fit the model to realistically sized datasets.
			To test whether $\Ne$ and $\mu$ are identifiable, for each branch of the tree we draw a 2-D scatter plot for $\Ne$ and $\mu$, where each point is a step in the MCMC procedure.
			We subsequently fit a linear regression and compute the coefficient of determination ($r^2$) for each branch of the tree.
			For a given MCMC, the distribution of $r^2$ across all branches is then represented as a violin plot.
			The distribution of $r^2$ is relatively low, below 0.05 on average, providing confidence that $\Ne$ and $\mu$ are indeed identifiable.}
	\end{figure}

	\subsection{Traits estimation with branch \texorpdfstring{$\omega$}{ω} (replicate~1, chain~1)}
	Obtained with the phenomenological inference model of log-Brownian process for the $\mu$ and the relative {non-synonymous} {substitution} rate ($\omega$), as in \citet{Lartillot2011}.

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{mammals/18CDS_BranchOmega_R1_LogdNdS}
		\caption[$\omega$ estimation in mammals]{{Non-synonymous substitution} rate ($\omega$) estimation in mammals}
	\end{figure}

	\begin{table}[H]
		\input{mammals/18CDS_BranchOmega_R1_TraitsCorrelation.tex}
		\caption[Correlation coefficient matrix in mammals ($\omega$)]{
		Correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{mammals/18CDS_BranchOmega_R1_TraitsCovariance.tex}
		\caption[Covariance matrix in mammals ($\omega$)]{
		Correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{mammals/18CDS_BranchOmega_R1_TraitsPartialCorrelation.tex}
		\caption[Partial correlation coefficient matrix in mammals ($\omega$)]{
		Partial correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in placental mammals.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}


	\section{Empirical data in Isopods}
	\label{sec:empirical-data-in-isopods}

	\subsection{Traits estimation (replicate~1, chain~1)}
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_R1_LogPopulationSize}
		\caption[$\Ne$ estimation in isopods]{Effective population size ($\Ne$) estimation in isopods}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_R1_LogMutationRatePerTime}
		\caption[Mutation rate estimation in isopods]{Mutation rate ($\mu$) estimation in isopods}
	\end{figure}

	\subsection{Repeatability of experiments}
	\label{subsec:repeatability-of-experiments-isopods}
	$6$ independent inferences were performed on a randomly chosen set of $12$ coding sequences ({CDS}) out of $135$.
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$.
	Each plot is a correlation between a pair of experiments for a given parameter.
	For each node (or branch) of the tree, the mean {posterior} of the parameter over the {MCMC} (after burn-in) is represented in blue dots, green solid lines are the $90\%$ confidence interval of the {MCMC}.
	Solid red line is the regression line between replicates.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_Log10BranchLength}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_Log10BranchLength}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_Log10BranchLength}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_Log10BranchLength}
		\end{minipage}
		\caption[Repeatability of branch length estimation in isopods]{Repeatability of branch length ($\branchlength$) estimation in isopods}
	\end{figure}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_LogPopulationSize}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_LogPopulationSize}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_LogPopulationSize}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_LogPopulationSize}
		\end{minipage}
		\caption[Repeatability of $\Ne$ estimation in isopods]{Repeatability of {effective population size} ($\Ne$) estimation in isopods}
	\end{figure}

	\begin{table}[htbp]
		\renewcommand{\arraystretch}{0.5}
		\scriptsize
		\noindent\adjustbox{max width=\textwidth}{%
		\input{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize.tex}
		}
		\caption[Repeatability of $\Ne$ estimation in isopods]{Repeatability of {effective population size} ($\Ne$) estimation in isopods, for the extant taxa.}
		\label{tab:Ne-estimated-isopods}
	\end{table}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_LogMutationRatePerTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_LogMutationRatePerTime}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_LogMutationRatePerTime}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_LogMutationRatePerTime}
		\end{minipage}
		\caption[Repeatability of $\mu$ estimation in isopods]{Repeatability of mutation rate ($\mu$) estimation in isopods}
	\end{figure}

	\begin{table}[htbp]
		\renewcommand{\arraystretch}{0.5}
		\scriptsize
		\noindent\adjustbox{max width=\textwidth}{%
		\input{isopods/12CDS_SiteMutSelBranchNe_Rep_LogMutationRatePerTime.tex}
		}
		\caption[Repeatability of mutation rate estimation in isopods]{Repeatability of mutation rate ($\mu$) estimation in isopods, for the extant taxa.}
	\end{table}

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-2_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-3_BranchTime}
		\end{minipage} \hfill
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-4_BranchTime}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-5_BranchTime}
		\end{minipage}
		\begin{minipage}{0.32\linewidth}
			\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep-1-6_BranchTime}
		\end{minipage}
		\caption[Repeatability of branch time estimation in isopods]{Repeatability of branch time ($\Delta T$) estimation in isopods}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eco}
		\caption[$\Ne$ as a function of habitat in isopods]{$\Ne$ as a function of habitat in isopods.}
	\end{figure}
	\verbatiminput{isopods/12CDS_SiteMutSelBranchNe_anova_habitat.txt}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_pig}
		\caption[$\Ne$ as a function of pigmentation in isopods]{$\Ne$ as a function of pigmentation in isopods}
	\end{figure}
	\verbatiminput{isopods/12CDS_SiteMutSelBranchNe_anova_pigmentation.txt}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/12CDS_SiteMutSelBranchNe_Rep_LogPopulationSize_eye}
		\caption[$\Ne$ as a function of ocular structure in isopods]{$\Ne$ as a function of ocular structure in isopods}
	\end{figure}
	\verbatiminput{isopods/12CDS_SiteMutSelBranchNe_anova_ocular_structure.txt}

	\subsection{Identifiability of $\Ne$ and $\mu$}
	\label{subsec:isopods-identifiability}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{isopods/identifiability}
		\caption[Identifiability of $\Ne$ and $\mu$]{
			Some parameters might be strongly correlated or only weakly identifiable, which may make it hard to fit the model to realistically sized datasets.
			To test whether $\Ne$ and $\mu$ are identifiable, for each branch of the tree we draw a 2-D scatter plot for $\Ne$ and $\mu$, where each point is a step in the MCMC procedure.
			We subsequently fit a linear regression and compute the coefficient of determination ($r^2$) for each branch of the tree.
			For a given MCMC, the distribution of $r^2$ across all branches is then represented as a violin plot.
			The distribution of $r^2$ is relatively low, below 0.1 on average, providing confidence that $\Ne$ and $\mu$ are indeed identifiable.}
	\end{figure}

	\section{Empirical data in Primates}
	\label{sec:empirical-data-in-primates}

	\subsection{Chain convergence}
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.

	\begin{figure}[H]
		\centering
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_ProfileCorrelation.png}
		\end{minipage} \hfill
		\begin{minipage}{0.49\linewidth}
			\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogPopulationSizeCorrelation}
		\end{minipage}
		\caption[Chain convergence of site profiles and branche $\Ne$]{
		Chain convergence of site amino-acid preferences (left panel) and branch $\Ne$ (right panel).}
	\end{figure}

	\subsection{Traits estimation (chain~1)}
	Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits.

	\begin{table}[H]
		\input{primates/SiteMutSelBranchNe_TraitsCorrelation.tex}
		\caption[Correlation coefficient matrix in primates ($\Ne$)]{
		Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{primates/SiteMutSelBranchNe_TraitsCovariance.tex}
		\caption[Covariance matrix in primates ($\Ne$)]{
		Correlation coefficient between effective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{primates/SiteMutSelBranchNe_TraitsPartialCorrelation.tex}
		\caption[Partial correlation coefficient matrix in primates ($\Ne$)]{
		Partial correlation coefficient between Neffective population size~($\Ne$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogPopulationSize}
		\caption[$\Ne$ estimation in primates]{Effective population size ($\Ne$) estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogMutationRatePerTime}
		\caption[Mutation rate estimation in primates]{Mutation rate ($\mu$) estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Logmaturity}
		\caption[Female maturity estimation in primates]{Female maturity estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Logmass}
		\caption[Mass estimation in primates]{Mass estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Loglongevity}
		\caption[Longevity estimation in primates]{Longevity estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogpiS}
		\caption[$\ps$ estimation in primates]{$\ps$ estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_LogpiNpiS}
		\caption[$\pnps$ estimation in primates]{$\pnps$ estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/SiteMutSelBranchNe_Loggeneration_time}
		\caption[Generation time estimation in primates]{Generation time estimation in primates}
	\end{figure}

	\subsection{Amino-acid preferences entropy}

	\begin{table}[H]
		\centering
		\noindent\adjustbox{max width=\textwidth}{%
		\begin{tabu}{|c|c|c|}
			\hline
			\textbf{Experiment} & $\left\langle \entropy \right\rangle$ (\textbf{branch} $\Ne$) & $\left\langle \entropy \right\rangle$ (\textbf{constant} $\Ne$) \\ \hline
			\hline
			Primates, chain~1 & $1.41 \pm 0.10$ & $1.49 \pm 0.08$\\ \hline
			Primates, chain 2 & $1.40 \pm 0.10$ & $1.48 \pm 0.08$\\ \hline
		\end{tabu}}
		\caption[Amino-acid entropy in primates]{Estimated amino-acid entropy in primates.
		Obtained with the mechanistic inference model developed in this paper of site-specific amino-acid fitness profiles and log-Brownian process for $\Ne$, $\mu$ and life-history traits (in the left column), or under the assumption of constant $\Ne$ (in the right column).}
	\end{table}

	\subsection{Traits estimation with branch \texorpdfstring{$\omega$}{ω} (chain~1)}
	Obtained with the phenomenological inference model of log-Brownian process for the $\mu$ and the relative {non-synonymous} {substitution} rate ($\omega$), as in \citet{Lartillot2011}.

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/BranchOmega_LogOmega}
		\caption[$\omega$ estimation in primates]{{Non-synonymous substitution} rate ($\omega$) estimation in primates}
	\end{figure}

	\begin{figure}[H]
		\centering
		\includegraphics[width=\linewidth, page=1]{primates/BranchOmega_LogMutationRatePerTime}
		\caption[$\mu$ estimation in primates]{Mutation rate ($\mu$) estimation in primates}
	\end{figure}

	\begin{table}[H]
		\input{primates/BranchOmega_TraitsCorrelation.tex}
		\caption[Correlation coefficient matrix in primates ($\omega$)]{
		Correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{primates/BranchOmega_TraitsCovariance.tex}
		\caption[Covariance matrix in primates ($\omega$)]{
		Correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\begin{table}[H]
		\input{primates/BranchOmega_TraitsPartialCorrelation.tex}
		\caption[Partial correlation coefficient matrix in primates ($\omega$)]{
		Partial correlation coefficient between {non-synonymous} {substitution} rate~($\omega$), mutation rate per site per unit of time~($\mu$), and life-history traits (maximum longevity, adult weight and female maturity) were computed in primates.
		Asterisks indicate strength of support ($\smash{^{*}} pp > 0.95$, $\smash{^{**}} pp > 0.975$).}
	\end{table}

	\section{Sufficient statistics}
	\label{sec:sufficient-statistics-mutselne}

	A sequence of length $\Nsite$ evolves by point substitutions, according to a random process defined by the {substitution} matrices $\Submatrix\branchsiteexp$, over a phylogenetic tree.
	A realization of the random process along a branch $\branch$, and at a particular site $\site$ results in a detailed {substitution} history, which will be denoted by $\subhistory\branchsiteexp$.

	\subsection{Path sufficient statistics}
	All sites owning to the same category of fitness profile share the same {substitution} rate matrix.
	Hence, $\subhistory\branchsiteexp$ can be gathered across all sites owing to a specific category $\cat$, denoted $\subhistory\branchexp$.
	If we express the probability of the {substitution} mapping ($\subhistory\branchcatexp$) as a function of the {codon} {substitution} process for this category $\cat$, we get the following expression:
	\begin{equation}
		\label{eq:PathSuffStat}
		\proba (\subhistory\branchcatexp | \branchlength\branchexp, \Submatrix\branchcatexp) \propto \left[  \prod\limits\sumOverCodons \left[\subequi_{\ci}\branchcatexp\right]^{n_{\ci}\branchcatexp} \right] \cdot \left[ \prod\limits_{ \ijSetCodon} \left[\submatrix\branchcatexp_{\itoj}\right]^{m_{\ci, \cj}\branchcatexp} \right] \cdot \left[ \prod\limits\sumOverCodons \e^{- \left| \submatrix\branchcatexp_{\ci, \ci}\right| a_{\ci}\branchcatexp}\right],
	\end{equation}
	where we define the sufficient statistics:
	\begin{itemize}
		\setlength\itemsep{-0.25em}
		\item $m_{\ci, \cj}\branchcatexp$ is the total number of substitutions from {codon} $\ci$ to {codon} $\cj$
		\item $n_{\ci}\branchcatexp$ is the number of sites starting with {codon} $\ci$ at the tip of the branch.
		\item $a_{\ci}\branchcatexp$ is the total waiting time in {codon} $\ci$.
	\end{itemize}
	Once these sufficient statistics have been computed, the parameters of the {substitution} matrix $\Submatrix\branchcatexp$ can be resampled conditional on $\subhistory\branchcatexp$,
	using equation~\ref{eq:PathSuffStat} each time the {likelihood} needs to be recomputed. This leads to relatively fast {MCMC} strategy.

	\subsection{Length sufficient statistics}
	$\subhistory\branchsiteexp$ can also be gathered across all sites along a specific branch, giving $\subhistory\branchexp$.
	Then the probability of the {substitution} history given the branch lengths ($\branchlength\branchexp = \mu\branchexp \branchtime\branchexp$), takes a very simple form:
	\begin{equation}
		\label{eq:LengthSuffStat}
		\proba (\subhistory\branchexp | L\branchexp) \propto \left[ L\branchexp\right]^{u\branchexp} \e^{- r\branchexp L\branchexp},
	\end{equation}
	where we define the sufficient statistics:
	\begin{itemize}
		\setlength\itemsep{-0.25em}
		\item $u\branchexp$ is the total number of substitutions over branch $\branch$, summed over all sites.
		\item $r\branchexp$ is the mean rate away from current {codon} state (averaged over the entire {substitution} history).
	\end{itemize}
	Thus, formally, the probability of the {substitution} mapping can be summarized by saying that the total number of substitutions along a given branch over all sites, $u\branchexp$, is Poisson distributed, of mean $r\branchexp L\branchexp$.

	\subsection{Scatter sufficient statistics}
	From the independent contrast $\contrast\branchexp$ of the Brownian process $\Brownian\nodeexp$, we can define the $2 \times 2$ scatter sufficient statistic matrix, $\Scattermatrix$ as:
	\begin{equation}
		\Scattermatrix = \sum\limits_{b=1}^{\Nbranch} \contrast\branchexp \cdot \left[\contrast\branchexp\right]^{\text{T}}
	\end{equation}
	By Bayes theorem, the {posterior} on $\CovarianceMatrix$, conditional on a particular realization of $\brownian$ (and thus of $\contrast$) is an invert Wishart distribution, of parameter $\covariancekappa \Identitymatrix + \Scattermatrix$ and with $\Nbranch + 3$ degrees of freedom.
	\begin{equation}
		\CovarianceMatrix \sim \text{Wishart}^{-1}\left( \covariancekappa \Identitymatrix + \Scattermatrix, \Nbranch + 3\right)
	\end{equation}
	This invert Wishart distribution can be obtained by sampling $\Nbranch + 3$ independent and identically distributed multivariate normal random variables $\Multivariate\indiceexp$ defined by
	\begin{equation}
		\Multivariate\indiceexp \sim \mathcal{N} \left( \vecZero, \left[\covariancekappa \Identitymatrix + \Scattermatrix\right]^{-1} \right).
	\end{equation}
	And from these multivariate samples, $\CovarianceMatrix$ is Gibbs sampled as:
	\begin{equation}
		\CovarianceMatrix = \left( \sum\limits_{k=1}^{\Nbranch + 3} \Multivariate\indiceexp \cdot \left[\Multivariate\indiceexp \right] ^{\text{T}} \right)^{-1}
	\end{equation}

	\bibliographystyle{natbib}%%%%Bibliography style file
	\bibliography{references}

\end{document}