\thispagestyle{empty}
\vspace*{\stretch{1}}


\begin{center}
	\LARGE \textbf{Modeling the articulation of adaptive and neutral mechanisms in the evolution of protein coding DNA sequences}
\end{center}

\section*{Abstract}
Selection in protein-coding sequences can be detected based on multiple sequence alignments using phylogenetic codon models.
Mechanistic approaches, grounded on population-genetics first principles, explicitly formalize the interplay between mutation, selection and random drift, and return an estimate of the amino-acid fitness landscape.
However, these recently developed models rely on the assumption of constant effective population size.
We propose an extended mutation-selection model reconstructing site-dependent fitness landscape, long-term trends in effective population size and mutation rate along the phylogeny, from an alignment of DNA coding sequences.
Independently, ancestral life-history traits are reconstructed along the phylogeny from observation in present-day species.
Together, we estimate the correlation between reconstructed life-history traits, mutation rate and effective population size, intrinsically including phylogenetic inertia.
Our framework has been tested against simulated data, and in empirical data in mammals and primates.
Finally, our work also points to important theoretical questions about how coding sequences respond to changes in effective population size and to fluctuating selection.\\

\textbf{keywords: }{Phylogenetic, codon models, mutation-selection models, population genetic, population size, mutation rate, life history traits.}

\vspace*{\stretch{3}}
\newpage