\newacronym{AA}{AA}{Amino-Acid}
\newacronym{AIC}{AIC}{Akaike information criterion}
\newacronym{A}{A}{Adenine}
\newacronym{HB}{HB}{Halpern \& Bruno}
\newacronym{gBGC}{gBGC}{GC Biased Gene Conversion}
\newacronym{dBGC}{dBGC}{DSB driven Biased Gene Conversion}
\newacronym{BGC}{BGC}{Biased Gene Conversion}
\newacronym{C}{C}{Cytosine}
\newacronym{CUB}{CUB}{Codon Usage Bias}
\newacronym{CDS}{CDS}{Coding DNA Sequence}
\newacronym{DSB}{DSB}{Double Strand Break}
\newacronym{DFE}{DFE}{Distribution of Fitness Effects}
\newacronym{DNA}{DNA}{DeoxyriboNucleic Acid}
\newacronym{G}{G}{Guanine}
\newacronym{GTR}{GTR}{Generalized Time Reversible}
\newacronym{GY}{GY}{Goldman \& Yang}
\newacronym{LHT}{LHT}{Life-History Trait}
\newacronym{LRT}{LRT}{Likelihood Ratio Test}
\newacronym{MCMC}{MCMC}{Markov Chain Monte Carlo}
\newacronym{MC}{MC}{Markov Chain}
\newacronym{MG}{MG}{Muse \& Gaut}
\newacronym{MK}{MK}{McDonald \& Kreitman}
\newacronym{MF}{MF}{Mean-Field}
\newacronym{ML}{ML}{Maximum Likelihood}
\newacronym{Ne}{\ensuremath{\Ne}}{effective population size}
\newacronym{PCR}{PCR}{Polymerase Chain Reaction}
\newacronym{RNA}{RNA}{RiboNucleic Acid}
\newacronym{S}{S}{Strong nucleotide (G or C)}
\newacronym{SFS}{SFS}{Site Frequency Spectrum}
\newacronym{T}{T}{Thymine}
\newacronym{tRNA}{tRNA}{transfer RNA}
\newacronym{W}{W}{Weak nucleotide (A or T)}
\newacronym{WS}{WS}{Mutation from a ‘weak’ (W) to ‘strong’ (S) nucleotide}

\newglossaryentry{allele}{name={allele},description={A variant form of a given gene}}
\newglossaryentry{GC-biased gene conversion}{name={GC-biased gene conversion},description={The process by which the GC-content increases because of biased gene conversion}}
\newglossaryentry{codon usage bias}{name={codon usage bias},description={Unequal frequency of the alternative codons that specify the same amino acid}}
\newglossaryentry{codon}{name={codon},description={Sequence of three nucleotides coding for a given amino acid}}
\newglossaryentry{effective population size}{name={effective population size},description={The number of individuals in a population who contribute to the next generation}}
\newglossaryentry{GC-content}{name={GC-content},description={The percentage of G or C nucleotides in a DNA sequence}}
\newglossaryentry{genetic drift}{name={genetic drift},description={The random fluctuation in allele frequencies due to random sampling of individuals}}
\newglossaryentry{genetic interference}{name={genetic interference},description={The fact that the formation of a recombination event can affect that of others in adjacent regions}}
\newglossaryentry{genetic linkage}{name={genetic linkage},description={Non-independent assortment of genes}}
\newglossaryentry{non-synonymous}{name={non-synonymous},description={Transition that modifies the encoded amino acid}}
\newglossaryentry{phenotype}{name={phenotype},description={The composite of observable traits}}
\newglossaryentry{ploidy}{name={ploidy},description={The number of complete sets of chromosomes in a cell}}
\newglossaryentry{polymorphic}{name={polymorphic},description={Which presents several forms. Subject to inter-individual variability}}
\newglossaryentry{recombination}{name={recombination},description={Exchange of DNA sequence information}}
\newglossaryentry{synonymous}{name={synonymous},description={substitution that does not modify the encoded amino acid}}
\newglossaryentry{Akaike information criterion}{name={Akaike information criterion},description={Measure of the relative quality of a maximum likelihood estimated given the data, penalizing for too many parameters in the model}}
\newglossaryentry{Bayes factor}{name={Bayes factor},description={Ratio of the posterior probability of two competing hypotheses, usually a null and an alternative}}
\newglossaryentry{Dirichlet process}{name={Dirichlet process},description={Family of stochastic processes whose realizations are probability distributions}}
\newglossaryentry{Likelihood ratio test}{name={Likelihood ratio test},description={Statistical test used to compare the goodness of fit of two models, one of which (the null model) is a special case of the other (the alternative model)}}
\newglossaryentry{likelihood}{name={likelihood},description={Probability of observing the data given the parameters of the statistical model. This is a function of the parameters of the model, the observed data are fixed}}
\newglossaryentry{Markov chain Monte Carlo}{name={Markov chain Monte Carlo},description={Class of algorithms for sampling from a probability distribution based on constructing a Markov chain that has the desired distribution as its equilibrium distribution}}
\newglossaryentry{Markov chain}{name={Markov chain},description={Stochastic process with property that the next state of the process depends only on the present state of the process and not on its past}}
\newglossaryentry{maximum likelihood}{name={maximum likelihood},description={Method of estimating the parameters of a model given the data, by finding the parameter value that maximize the likelihood function}}
\newglossaryentry{nearly-neutral}{name={nearly-neutral},description={Slightly deleterious or advantageous mutations are effectively neutral when their selection coefficient are lower than one divided by the effective population size}}
\newglossaryentry{neutral}{name={neutral},description={Mutations are nor deleterious neither advantageous, their probability of fixation is one divided by twice the effective population size (for diploids)}}
\newglossaryentry{prior}{name={prior},description={Probability distribution that would express one's beliefs about a parameter of the model before the data is taken into account}}
\newglossaryentry{posterior}{name={posterior},description={Probability distribution of a parameter of the model conditioned on randomly observed data and the prior distribution}}
\newglossaryentry{substitution}{name={substitution},description={Point mutation that appeared in only one individual in the population, and subsequently reached fixation in the population}}

\printglossary[type=\acronymtype]
\mtcaddchapter
\printglossary
\mtcaddchapter