\thispagestyle{empty}
\chapter{Historical perspective on molecular evolution}
{
	\hypersetup{linkcolor=GREYDARK}
	\minitoc
}

\label{sec:intro-historical}

From the discovery of evolution to today knowledge, the understanding of the mechanisms from which the diversity of life and complexity emerges has seen dramatic changes, and revolution.
One such revolution is called molecular evolution, a recent scientific fields, emerging at the crossroad of evolutionary biology which had seen tremendous theoretical development in the nineteenth and twentieth century, and molecular biology which recruited advances in chemistry and had seen many technical revolutions.
Being both empirical and theoretical biology, molecular evolution borrows strength from the amount of empirical data available in molecular biology, and the predictive power of evolutionary biology.
From the difference of molecular sequences observable between individuals of the same population, or difference of sequences between species, we can wonder what are the processes generating such diversity?
What are the forces governing such evolutionary mechanisms?
Can we quantify the relative strength of these forces, shaping both nowadays populations but also ancient and sometimes extinct lineages?
In a nutshell, molecular evolution leverages the patterns of sequences distribution carried by individuals in order to uncover evolutionary mechanisms shaping organisms evolution and their ancestral lineages, while at the same time shining light on cellular and molecular processes allowing organism to live and reproduce.

This section will recall theoretical frameworks, assumptions and limitations on which molecular evolution is based.
It will emphasis the milestones and their respective contribution, while doing its best to highlight the undertaken paths which have been forgotten with time.
It is a modest attempt neither exhaustive nor accurate, imprinted with ideology of our current society on how we perceive and interpret past discoveries.
Moreover, this introduction will highlight a few names, while the bulk of the rigorous assessment and develpoment of molecular evolution has been done by unmentioned and sometimes forgotten scientists.

\section{Population-genetic}
Molecular evolution is theoretically build upon the framework of population-genetic, which in turn historically emerged as an unifying theory between Mendelian inheritance and quantitative genetic, in the early twentieth century.
Originally,
Johann Gregor Mendel% (1822-1884)
 established the statistical laws governing heredity of discrete characters through  hybridisation experiments on the garden pea plant \textit{Pisum sativum} between 1857 and 1864.
This model of inheritance was rediscovered and confirmed in the early twentieth century independently by botanists
Hugo de Vries% (1848-1935)
, Carl Correns% (1864-1933)
 and Erich von Tschermak% (1871-1962)
 ~\citep{dunn2003gregor}.

Models of Mendelian inheritance where deemed incompatible with models of biometricians, while the crux of the argument revolved around the evolution of continuous characters\footnote{Incompatibility between continuous and discrete evolution can actually be traced back to debates between Jean-Baptiste de Lamarck (1744-1829) defending gradual changes and Georges Cuvier (1869-1932) supporting punctual catastrophic changes, in the late eighteenth century.}.
Broadly speaking, supports of Mendelian genetic believed that evolution was driven by mutations transmitted by the discrete segregation of \glspl{allele}, which biometricians rejected on the basis that it would necessarily imply discontinuous evolutionary leaps~\citep{bowler2003evolution}.
On the other hand, biometricians claimed that variation was continuous, which mendelian geneticist rejected on the basis variations measured by biometricians were too small to be subject to selection~\citep{provine2001origins}.

Statistician Ronald A.\ Fisher reconciled both theories, first by proving mathematically that mutiple discrete loci could result in a continuous variation~\citep{fisher1919xv}.
Secondly, \citet{fisher1930genetical} and \citet{haldane1932causes} proved that natural selection could change \gls{allele} frequencies in a population.

Fisher and Haldane hence articulated selection on continuous traits with discrete underlying genetic inheritance, completed by the work of \citet{wright1932roles} on combinations of interacting genes.
Altogether, they laid the foundations of population genetics, a discipline which basically integrated Mendelism, Darwinism and biometry, easing the debate between continous and gradual evolution\footnote{This debate was revived by paleontologists \citet*{Gould1972}.
As of today it is admitted that both macroevolutive patterns of ponctual and gradual changes can be found.}.

The emergence of this new field of study was the first step towards the development of a unified theory of evolution named the ‘modern synthesis’~\citep{huxley1942evolution}, defined on the basis that natural selection acts on the heritable variation supplied by mutations~\citep{mayr1959where,stebbins1966processes,dobzhansky1974chance}.

\section{Central dogma of molecular biology}
During the theoretical development of population-genetic, the support of heredity was largely unknown, the terminology of gene, \glspl{allele} and loci where theoretical and not grounded on chemical first principles.
The first evidence that deoxyribonucleic acid (\acrshort{DNA}) carries genetic information is in the work of \citet{Avery1944}, after bacteria treated with a deoxyribonuclease enzyme failed to transform, while otherwise transforming, even when treated by protease.
The chemical composition of \acrshort{DNA} was later refined by \citet{Chargaff1950}, whom found that the amounts of adenine (A) and thymine (T) in \acrshort{DNA} were roughly the same as the amounts of cytosine (C) and guanine (G).
Most importantly, the relative amounts of guanine, cytosine, adenine and thymine bases were found to vary from one species to another, which provided evidence that \acrshort{DNA} could encodes genetic information, via a four letter molecular alphabet.

Ultimately, the double-helix structure of \acrshort{DNA} was deciphered by \citet{franklin1953molecular}, \citet{watson1953molecular} and \citet{wilkins1953molecular}.
Chemically, \acrshort{DNA} consists of two interwoven strands of nucleotides, each of which contains an identical phosphate group, an identical $5$-carbon sugar (deoxyribose) and a variable base which define the four nucleotide: T, C, A and G.
Apart from being different molecules, bases are also hybridizing via hydrogen bonds, which are weak attractive forces between hydrogen and either nitrogen or oxygen.
This hybridization explains that the two strands of \acrshort{DNA} are interwoven due to base pair complementarity, where a specific base on one strand is aligned with its complement on the other strand.
More precisely, the complementary bases are:
\begin{itemize}
	\item A with T, through two hydrogen bonds.
	\item G with C, through three hydrogen bonds which are stronger than A-T bonds.

\end{itemize}
It is important to note the two strands are oriented, a constrained imposed by the asymmetry of the deoxyribose, where one end is called the $5'$ end and the other $3'$ end\footnote{The naming come from the numbering of carbon atoms of the asymmetric sugar molecule ($5$ of them) on which the end is attached.}.

This means that two strands will only hybridize if they are reverse complement, such that that the sequence of one strand when read from $5'$ to $3'$ is complementary to the sequence of the other strand read from $3'$ to $5'$.

Ultimately, the information contained by each strand of \acrshort{DNA} is redundant, and this redundancy is leveraged during replication of the \acrshort{DNA}.
During the cell cycle, the \acrshort{DNA} double strand is split into its two separate strands, each of them used as a template to synthesize its complementary strand, resulting in two copies of the original double-stranded \acrshort{DNA}.

Once understood that molecular structure of \acrshort{DNA} and its role has a support of heredity, the guest to understand the transfer of information between \acrshort{DNA} and protein~\citep{Crick1958} resulted in the determination of the genetic code, the translation table from triplet of nucleotides (\gls{codon}) to amino-acids, and of the central dogma of molecular biology detailing the process of protein synthesis~\citep{Crick1970}.
Proteins are synthesized in a two-step process called gene expression.
First, an messenger ribonucleic acid (mRNA) transcript is synthesized from \acrshort{DNA}, containing the same information since \acrshort{RNA} is also formed by 4 bases, where thymin (T) is replaced by uracil (U), though the three other bases are the same.

Second, mRNA is matured and spliced to form a mature \acrshort{RNA}, which is read and interpreted by ribosomes to synthesize a protein in a process called translation.

More broadly, the central dogma of molecular biology states that the determination of sequence from nucleic acid to nucleic acid, or from nucleic acid to protein may be possible, but transfer from protein to protein, or from protein to nucleic acid is impossible.
It is worth noting that the same \acrshort{DNA} sequence can produce many different proteins through a process called alternative splicing.

As the support of heredity, \acrshort{DNA} gained a central role in evolutionary biology, and development of polymerase chain reaction (PCR), Sanger sequencing and their subsequent refinement revolutionized the availability of empirical data on which to test the theoretical prediction and development of population genetics.

\section{Neutral theory}

Although an unifying theory, population-genetics remained rather theoretical for some time because it deals with the concept of gene frequencies and has no direct way to connect unambiguously to conventional dataset obtained at the phenotypic level.
With the advent of molecular genetics, it became possible to study the variability of nucleic and protein sequences within a species, as well as in related organism such as to estimate the rate at which allelic genes are substituted.

From protein sequences in related species, it was then observed that a given protein rate of \gls{substitution} is about the same in many diverse lineages, where the \glspl{substitution} seemed to be random rather than having a specific pattern.
Additionally, from \acrshort{DNA} sequences in related species, it was observed that the overall rate of \acrshort{DNA} \glspl{substitution} is very high, of least one nucleotide base per genome every two years in a mammalian lineage.
On the other hand, from the variability of protein sequences in the same population, electrophoretic methods suddenly unveiled a wealth of genetic variability, such that the proteins produced by a large fraction of the genes in diverse organisms were found to be \gls{polymorphic}, and in many cases the protein polymorphism had no visible phenotypic effects and no obvious correlation.
Altogether, these observation lead Motoo Kimura to propose the \gls{neutral} theory of molecular evolution~\citep{kimura1968evolutionary,kimura1991neutral,kimura1986dna}.
Neutral theory claimed that most mutations are adaptively \gls{neutral}, thus explaining the high protein variability observed in polymorphism dataset, where the diversity is supplied by a high mutational input.
Subsequently, this selectively \gls{neutral} diversity is reduced by random extinction of \glspl{allele}, via the cumulative effect of genetic random sampling of \glspl{allele} at each generation.
Although the likely outcome of a \gls{neutral} \gls{allele} in a population is it ultimate extinction, it is also possible that the random drift leads to a fixation of this \gls{allele} in the population.
In this context, the frequency of the \gls{neutral} \gls{allele} fluctuates through generations, increasing or decreasing fortuitously over time, because only a relatively small number of \glspl{Gamete} are randomly sampled out of the vast number of male and female \glspl{Gamete} produced in each generation.
As a consequence, effect of \gls{drift} at the level of a population results into divergence between lineages, where the majority of the nucleotide \glspl{substitution} in the course of evolution must then be the result of the random fixation of \gls{neutral} or mutants rather than the result of positive Darwinian selection.
Tomoko Ohta later incorporated weakly selected mutation into the \gls{nearly-neutral} theory~\citep{ohta1973slightly}, which posits that selective effect in the order of inverse population size are negligible and behaves neutrally.

This theory sparked controversy between neutralist and selectionists.
Selectionists maintain that a mutant \gls{allele} must have some selective advantage to spread through a species, although admitting that a \gls{neutral} \gls{allele} may occasionally be carried along by hitchhiking on a closely linked gene that is positively selected.
Neutralists, on the other hand, argued that some mutants might spread through a population without having any selective advantage by random sampling, such that if a mutant is selectively equivalent to preexisting resident \glspl{allele}, its fate is thus left to chance.
As of today, it is widely accepted that both \gls{drift} and natural selection participate in the evolution of genomes.
The controversy is no longer strictly dichotomous but rather concerns the quantitative contributions of adaptive and of non-adaptive evolutionary processes, and their articulation with regards to mutation, selection, drift, migrations, \gls{GeneConv}, and other evolutionary processes.

\section{Molecular evolution}

The \gls{nearly-neutral} theory had broad implications, in modelling of selective landscape, in phylogenetic context and in population-genetic.
It formally predict the rate of evolution of molecular sequences, measurable in the divergence between molecular sequences in different organisms, as well as predicting diversity within species.
Importantly, \gls{nearly-neutral} evolution of molecular sequences is resulting from the interplay between mutation, selection and drift.
As such this theory allowed to posit assumption on the underlying process and test them against empirical sequences.
Questions ranged from structure of fitness landscapes, to the causes of mutational rate variations, and wondering whether drift is an observable prominent process.

It fostered our understanding of selection:

$\bullet$ Which part of the genome are under selection?
Non-synonymous mutation are selected while synonymous are more \gls{neutral} \cite{Muse1994,Goldman1994}, although this is not strict (\gls{Codon}).

$\bullet$ How to disentangle purifying selection, \gls{neutral} evolution and adaptive evolution?
Amino-acid sequences are mainly under purifying selection, but traces of adaptation for specific genes and sites can be detected~\citep{McDonald1991, enard_viruses_2016}.

Our understanding of mutation:

$\bullet$ Is the mutation rate constant along the branches?
Substitution rate is proportional to mutation rate.
Molecular clock \citep{Thorne1998, Lanfear2010a}.
The rate of mutation multiplied the genome size is approximately constant \citep{Drake1991}.

$\bullet$ Is mutation biased? \citep{Singer2000}.

$\bullet$ Is the mutation rate constant along the genome?
Ref needed.

Our understanding of drift:

$\bullet$ How to access independent estimation of effective population size?
Synonymous polymorphism as a proxy of $\Ne$~\citep{Galtier2016}.
Strength of selection as proxy in phylogenetic context~\citep{Seo2004}.
Focus of chapter~\ref{chap:MutSelDrift}.

$\bullet$ Is the evolution rate determined by drift?
Depends on the assumption of the fitness landscape~\citep{Cherry1998, Goldstein2013}.
Focus of chapter~\ref{chap:GenoPhenoFit}.

$\bullet$ Importance of \gls{recombination}, hitchhiking if mutations are spatially close.
Some patterns are inconsistent, and thus lead to uncovering new forces such as \gls{bgc} which mimics selection but are fundamentally a mutational process.
Duret \& Mouchiroud.

Evolution is resulting from the interplay between mutation, selection and drift, where this formalism is developed in chapter \ref{sec:intro-formalism}.
Selection is the most evasive component, which can be nailed down in protein-coding \acrshort{DNA} sequences, for which selection can be related to bio-chemical and bio-physical constrains (chapter \ref{sec:selection}).
Consequently from the patterns of differences between sequences, how can we estimate and learn about the process generating them, more specifically the interplay between mutation, selection and drift (chapter \ref{sec:phylo_codon_models}).
