\thispagestyle{empty}
\chapter{Historical perspective on molecular evolution}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:intro-historical}

From the discovery of evolution to today knowledge, the understanding of the mechanisms from which the diversity of life and complexity emerges has seen dramatic changes, and revolution.
One such revolution is called molecular evolution, a recent scientific endeavor emerging at the crossroad of two fields, on one hand evolutionary biology which had seen tremendous theoretical development in the nineteenth and twentieth century, and on the other hand molecular biology which recruited advances in chemistry and had seen many technical revolutions.
Being both empirical and theoretical biology, molecular evolution borrows strength from the amount of empirical data available in molecular biology, and the predictive power of evolutionary biology.
From the difference of molecular sequences observable between individuals of the same population, or difference of sequences between species, evolutionary biologist can uncover the processes generating such diversity, and unravel the forces governing such evolutionary mechanisms.
Can we quantify the relative strength of these forces, shaping both nowadays populations but also ancient and sometimes extinct lineages?
In a nutshell, molecular evolution leverages the patterns of sequences distribution carried by individuals in order to uncover evolutionary mechanisms shaping organisms evolution and their ancestral lineages, while at the same time shining light on cellular and molecular processes allowing organism to live and reproduce.

This section will recall theoretical frameworks, assumptions and limitations on which molecular evolution is based.
It is a modest attempt neither exhaustive nor accurate, imprinted with ideology of our current society on how we perceive and interpret past discoveries.
Moreover, this introduction will highlight a few names, while the bulk of the rigorous assessment and develpoment of molecular evolution has been done by unmentioned and sometimes forgotten scientists.

\section{Population-genetic}
Molecular evolution is theoretically build upon the framework of population-genetic, which in turn historically emerged as an unifying theory between Mendelian inheritance and quantitative genetic, in the early twentieth century.
Originally,
Johann Gregor Mendel% (1822-1884)
 established the statistical laws governing heredity of discrete characters through hybridization experiments on the garden pea plant \textit{Pisum sativum} between 1857 and 1864.
This model of inheritance was rediscovered and confirmed in the early twentieth century independently by botanists
Hugo de Vries% (1848-1935)
, Carl Correns% (1864-1933)
 and Erich von Tschermak% (1871-1962)
 ~\citep{dunn2003gregor}.

Models of Mendelian inheritance where deemed incompatible with models of biometricians, while the crux of the argument revolved around the evolution of continuous characters\footnote{Incompatibility between continuous and discrete evolution can actually be traced back to debates between Jean-Baptiste de Lamarck (1744-1829) defending gradual changes and Georges Cuvier (1869-1932) supporting punctual catastrophic changes, in the late eighteenth century.}.
Broadly speaking, supports of Mendelian genetic believed that evolution was driven by mutations transmitted by the discrete segregation of \glspl{allele}, which biometricians rejected on the basis that it would necessarily imply discontinuous evolutionary leaps~\citep{bowler2003evolution}.
On the other hand, biometricians claimed that variation was continuous, which mendelian geneticist rejected on the basis variations measured by biometricians were too small to be subject to selection~\citep{provine2001origins}.

Statistician Ronald A.\ Fisher reconciled both theories, first by proving mathematically that mutiple discrete loci could result in a continuous variation~\citep{fisher1919xv}.
Secondly, \citet{fisher1930genetical} and \citet{haldane1932causes} proved that natural selection could change \gls{allele} frequencies in a population.

Fisher and Haldane hence articulated selection on continuous traits with discrete underlying genetic inheritance, completed by the work of \citet{wright1932roles} on combinations of interacting genes.
Altogether, they laid the foundations of population genetics, a discipline which basically integrated Mendelism, Darwinism and biometry, easing the debate between continous and gradual evolution\footnote{This debate was revived by paleontologists \citet*{Gould1972}.
As of today it is admitted that both macroevolutive patterns of ponctual and gradual changes can be found.}.

The emergence of this new field of study was the first step towards the development of a unified theory of evolution named the ‘modern synthesis’~\citep{huxley1942evolution}, defined on the basis that natural selection acts on the heritable variation supplied by mutations~\citep{mayr1959where,stebbins1966processes,dobzhansky1974chance}.

\section{Central dogma of molecular biology}

During the theoretical development of population-genetic, the support of heredity was largely unknown, the terminology of gene, \glspl{allele} and loci where theoretical and not grounded on chemical first principles.
The first evidence that deoxyribonucleic acid (\acrshort{DNA}) carries genetic information is in the work of \citet{Avery1944}, after bacteria treated with a deoxyribonuclease enzyme failed to transform, while otherwise transforming even when treated by protease.
The chemical composition of \acrshort{DNA} was later refined by \citet{Chargaff1950}, whom found that the amounts of adenine (A) and thymine (T) in \acrshort{DNA} were roughly the same as the amounts of cytosine (C) and guanine (G).
Most importantly, the relative amounts of guanine, cytosine, adenine and thymine bases were found to vary from one species to another, which provided evidence that \acrshort{DNA} could encodes genetic information, via a four letter molecular alphabet.

Ultimately, the interwoven double-helix structure of \acrshort{DNA} was deciphered by \citet{franklin1953molecular}, \citet{watson1953molecular} and \citet{wilkins1953molecular}.
Once understood that molecular structure of \acrshort{DNA} and its role has a support of heredity, the guest to understand the transfer of information between \acrshort{DNA} and protein~\citep{Crick1958} resulted in the determination of the genetic code, the translation table from triplet of nucleotides (\gls{codon}) to amino-acids, and of the central dogma of molecular biology detailing the process of protein synthesis~\citep{Crick1970}.
Broadly, the central dogma of molecular biology states that the determination of sequence from nucleic acid to nucleic acid, or from nucleic acid to protein may be possible, but transfer from protein to protein, or from protein to nucleic acid is impossible.
It is worth noting that the same \acrshort{DNA} sequence can produce many different proteins through a process called alternative splicing.

As the support of heredity, \acrshort{DNA} gained a central role in evolutionary biology, and development of polymerase chain reaction (PCR), Sanger sequencing and their subsequent refinement revolutionized the availability of empirical data on which to test the theoretical prediction and development of population genetics.

\section{Neutral theory}

Although an unifying theory, population-genetics remained rather theoretical for some time because it deals with the concept of gene frequencies and has no direct way to connect unambiguously to conventional dataset obtained at the phenotypic level.
With the advent of molecular genetics, it became possible to study the variability of nucleic and protein sequences within a species, as well as in related organism such as to estimate the rate at which allelic genes are substituted.

From protein sequences in related species, it was then observed that a given protein rate of \gls{substitution} is similar in many diverse species (Zuckerkandl \& Pauling, 1965; Salser et al, 1976), where \glspl{substitution} seemed to be random rather than having a specific pattern.
Additionally, from \acrshort{DNA} sequences in related species, it was observed that the overall rate of \acrshort{DNA} \glspl{substitution} is very high, of least one nucleotide base per genome every two years in a mammalian lineage.
On the other hand, from the variability of protein sequences in the same population, electrophoretic methods suddenly unveiled a wealth of genetic variability, such that the proteins produced by a large fraction of the genes in diverse organisms were found to be \gls{polymorphic}, and in many cases the protein polymorphism had no visible phenotypic effects and no obvious correlation (Harris 1966, Lewontin and Hubby 1966).
Altogether, these observation led Motoo Kimura to propose the \gls{neutral} theory of molecular evolution~\citep{kimura1968evolutionary,kimura1986dna, kimura1991neutral}.
Neutral theory claimed that most mutations are adaptively \gls{neutral}, thus explaining the high protein variability observed in polymorphism dataset, where the diversity is supplied by a high mutational input.
Subsequently to origination by mutation, this selectively \gls{neutral} diversity is reduced by random extinction of \glspl{allele}, via the cumulative effect of genetic random sampling of \glspl{allele} at each generation.
Although the likely outcome of a \gls{neutral} \gls{allele} in a population is it ultimate extinction, it is also possible that the random drift leads to a fixation of this \gls{allele} in the population.
In this context, the frequency of the \gls{neutral} \gls{allele} fluctuates through generations, increasing or decreasing fortuitously over time, because only a relatively small number of \glspl{Gamete} are randomly sampled out of the vast number of male and female \glspl{Gamete} produced in each generation.
As a consequence, effect of \gls{drift} at the level of a population results into divergence between lineages, where the majority of the nucleotide \glspl{substitution} in the course of evolution must then be the result of the random fixation of \gls{neutral} or mutants rather than the result of positive Darwinian selection.

Kimura \& Ohta (1971) precised that mutation can have an effect on fitness but still behave neutrally and have their fate dictated solely by drift.
Indeed, \citet{ohta1973slightly} later incorporated weakly selected mutation to form the \gls{nearly-neutral} theory, which posits that selective effect lower than the inverse of effective population size are negligible and behaves neutrally.
In this regard, effective population size is a quantitative measure of drift, where genetic drift decreases with increased effective population size.

This theory sparked controversy between neutralist and selectionists.
Selectionists maintain that a mutant \gls{allele} must have some selective advantage to spread through a species, although admitting that a \gls{neutral} \gls{allele} may occasionally be carried along by hitchhiking on a closely linked gene that is positively selected.
Neutralists, on the other hand, argued that some mutants might spread through a population without having any selective advantage by random sampling, such that if a mutant is selectively equivalent to preexisting resident \glspl{allele}, its fate is thus left to chance.
As of today, it is widely accepted that both \gls{drift} and natural selection participate in the evolution of genomes.
The controversy is no longer strictly dichotomous but rather concerns the quantitative contributions of adaptive and of non-adaptive evolutionary processes, and their articulation with regards to mutation, selection, drift, migrations, \gls{GeneConv}, and other evolutionary processes.

\section{Legacy of nearly-neutral theory}

The \gls{nearly-neutral} theory had broad implications in evolutionary biology, from modeling of selective landscape to phylogenetic reconstruction, as well as detection of adaptation in population-genetic.
Indeed the theory formally predicts the rate of evolution of molecular sequences, measurable in the divergence between molecular sequences in different organisms, as well as predicting diversity within species.
Importantly, \gls{nearly-neutral} evolution of molecular sequences formalize the interplay between mutation, selection and drift.
As such this theory allowed to posit assumptions on the underlying process and test them against empirical sequences, which became and are still increasingly available.
Questions within this framework ranged from causes of mutational rate variations, to structures of fitness landscapes, and wondering whether drift is an observable prominent process.

Although nearly-neutral theory introduced the force of drift balancing the force of selection, the theory also shed light on the primal ingredient of evolution, namely the component of mutation.
Originally, neutral theory was developed in the context of theoretically scrutinizing the clock hypothesis of Zuckerkandl and Pauling (1965), which posit that rate of sequence evolution is the same in different evolutionary species.
Theoretical development, available empirical data and computing resources altogether led to development of methods modeling fluctuation of the substitution rate along a phylogeny \citep{Thorne1998}, which subsequently showed that the molecular clock approximation does not hold empirically.
Relaxation of strict molecular clock also fostered methods of comparative evolution, where variation of substitution rate is related to strength of selection \citep{Seo2004}, or variation of life-history traits, such as body size or generation time \citep{Lartillot2012}, most importantly taking into account non-independence of lineages due to the underlying phylogeny \citep{Lanfear2010a}.
Notably, nearly-neutral theory and more especially the drift barrier hypothesis has been invoked to explain the observation that product of genome size and mutation rate is constant~\citep{Lynch2016a} (see end of chapter \ref{chap:intro-formalism}).
Finally, because mutation, selection and drift are untangled, it is important to note that composition of genome does not reflect the underlying mutational process \citep{Singer2000}, but its filtering by selection and drift.
As a result a careful modeling of selection and drift is necessary to uncover mutational process, an articulation of this thesis studied in chapter~\ref{chap:NucleotideBias}.

Naturally, nearly-neutral theory had profound effect on our understanding of selection. 
Along with adoption of nearly-neutral theory by evolutionary biologists, the nature of selection shifted from a driver of changes mediated by adaptive mutations to a mainly purifying force discarding and filtering out strongly deleterious mutations (Lynch and Walsh 2007).
As a result, proteins are considered relatively optimized, such that proposed mutations are likely disrupting their functions.
This effect can be observed in protein coding DNA sequences, since the substitution rate of DNA is reduced whenever the translated DNA in protein is also modified \cite{Muse1994,Goldman1994}.
This effect observed between species could also be replicated inside population, where in many populations alleles segregate at lower frequencies whenever the allelic change result in change of the protein, a phenomenon explained by purification of deleterious alleles which cannot reach high frequencies (Akashi 1999; Cargill et al. 1999; Hughes 2005).
Altogether, is was found that amino-acid sequences are mainly under purifying selection, but traces of adaptation for specific genes and sites has been detected in a phylogenetic context, detectable by local acceleration of substitution rate~\citep{enard_viruses_2016}
Most notably, because nearly-neutral theory prominently modeled purifying selection, it has been leveraged as a null model where violation of this model can be interpreted as traces of adaptations in polymorphism context~\citep{McDonald1991, Galtier2016} or phylogenetic context \cite{Rodrigue2016}.

Tempering the effect of selection, drift mediated by effective population size had been invoked numerous time to explain relaxation of selective strength.
First, it has been observed that within populations, relative diversity of selected site is more reduced for species with smaller effective population size (Piganeau and Eyre-Walker 2009; Elyashiv et al. 2010; \citep{Galtier2016}; Chen et al. 2017; James et al. 2017).
In a phylogenetic context between species, strength of selection measured as a relative substitution rate of selected site is lower along lineages with small effective population size~\citep{Popadin2007}.
It is important to note that in most cases the effective population is not directly measured but a surrogate measure is used instead, for example diversity of neutral markers in polymorphism context, or body size in a phylogenetic context.
Leveraging nearly-neutral theory in order to quantitatively measure drift in a phylogenetic context is a main articulation of this thesis, studied in chapter~\ref{chap:MutSelDrift}.
In a phylogenetic context, studies relating substitution rate of selected sites under selection to effective population size posit that nearly-neutral theory predicts this negative relationship, however it has been found that this relationship can be null for some fitness landscapes~\citep{Cherry1998, Goldstein2013}.
This relationship between the rate of evolution and effective population size is also a main articulation of this thesis, studied in chapter~\ref{chap:GenoPhenoFit}.

Finally, some patterns have been found inconsistent within the general framework of mutation, selection and drift, thus leading to uncovering new forces such as \gls{bgc} which mimics selection but are fundamentally segregation distortion during recombination (Marais, 2003).
Such forces are altering the composition of genomes and must be carefully accounted for in models of evolution \citep{Galtier2009, Figuet2014}.
However, even though forces such as \gls{bgc} are not within the scope of this thesis, some assumptions and design of our models had been taken such as to implement these forces subsequently. 

Altogether, evolution of sequences result from the interplay between mutation, selection and drift, where this formalism is developed in chapter \ref{chap:intro-formalism}.
Of all these components, selection is the most pervasive, which can be approximated and observed in protein-coding \acrshort{DNA} sequences in a phylogenetic context between lineages, presented in chapter \ref{chap:intro-codon-models}).
Consequently, models are applied to empirical data, and the methodology of Bayesian inference from an alignment of DNA sequences is presented in chapter \ref{chap:intro-inference}.
Finally, selection of protein coding DNA sequences is related to bio-chemical and bio-physical constrains (chapter \ref{chap:intro-inference}).
