\thispagestyle{empty}
\chapter{The mathematics of molecular evolution}
{\hypersetup{linkcolor=GREYDARK}\minitoc}
\label{chap:intro-formalism}

In molecular evolution, the information contained in empirically observed sequences is leveraged to reconstruct ancestral lineages and to unveil the evolutionary mechanisms having generated this diversity of sequences.
In other words, the task is to reconstruct the ancestral path followed by lineages using the knowledge available today, by working backward in time.
To do so, however, requires a theoretical model of the generating process forward in time.
One can then play this model forward and derive how likely it would be for this model to generate the sequence patterns that are empirically observed.

Working out the long-term molecular evolutionary process first requires to formalize what happens in a short time period within populations.
Population genetics, with its assumptions and limitations, provides the theoretical framework for this.
The first section thus recalls the basics of mathematical population genetics, and more specifically, the Wright-Fisher model and its assumptions.
This will allow me to relate parameters of evolution such as mutation, selection and drift to observable patterns in molecular sequences such as the probability of fixation of a mutant \gls{allele}, as well as the expected number of copies of the derived \gls{allele} we should observe in a population.
This will allow me to relate the parameters of the evolutionary process, such as mutation rates, selection coefficients and effective population sizes, to observable properties of molecular sequences, such as the probability of fixation of a mutant allele or the expected number of copies of the derived allele we should observe in a population.
These relationships between the underlying evolutionary forces and the observable patterns will subsequently be leveraged and recruited in the next section to derive an approximation of the long-term process of sequence substitution, again, parametrized directly in terms of mutation, selection and drift.

Although the mathematical proof for most of the results presented here are out of the scope of this manuscript, an effort was made to state all definitions and assumptions.
Such an effort is meant to clearly define the models, their assumptions and their parametrization from the ground up.


\section{Population-genetics of sequences}

\subsection{The Wright-Fisher model}

The Wright-Fisher model describe the change in frequency of a \gls{polymorphic} gene with two \glspl{allele} in a \gls{diploid} population over time.
The population is assumed to consist of fixed number of diploid individuals $N \gg 1$.
It is also assumed to be panmictic, with non-overlapping generations.
The number of copies of the derived allele $B$ present at the current generation is denoted $i$ and the frequency of this mutant allele $B$ is denoted $p = 1/2N$, while the frequency of the resident $A$ alleles is $1 - p$.

The ability to survive and produce offspring differs between the three diploid genotypes ($AA$, $AB$, $BB$).
Here, selection is assumed to occur between the zygotic and the adult stage, called post-zygotic selection.
Quantitatively, selection is captured by a measure called Wrightian fitness ($w$), which, for a given diploid genotype, is defined as the expected number of offspring produced by an individual having this genotype.
Since the population is regulated in size, only the relative fitness matters, which is usually set to $1$ for the reference (wild-type) genotype.
It is convenient to define the fitness of the other two genotypes, relative to the wild-type, in terms of a selection coefficient.
Furthermore, in the following, we will assume additive effects (co-dominance), such that the heterozygote has an intermediate fitness between the two homozygotes.
Altogether, fitness of the three diploid genotypes are defined as:
\begin{equation}
    \begin{dcases}
        w_{AA} & = 1 \\
        w_{AB} & = 1 + s \\
        w_{BB} & = 1 + 2s \\
    \end{dcases}
\end{equation}

More generally than the previous equations, under the assumption that selection is weak $|s| \ll 1$, the selection coefficient can be approximated by the difference in Wrightian fitness of the mutant and the resident \gls{allele} as:
\begin{align}
    s & = \dfrac{w_{B} - w_{A}}{w_{A}}, \\
    & = \dfrac{w_{B}}{w_{A}} - 1, \\
    & \simeq \ln\left( \dfrac{w_{B}}{w_{A}} \right), \\
    & \simeq \ln(w_{B}) - \ln(w_{A}), \\
    & \simeq f_{B} - f_{A},
\end{align}
where $f = \ln(w)$ is often referred to as the Malthusian fitness, relative fitness or also log-fitness.

\subsection{Frequency changes across successive generations}

Under the Hardy-Weinberg equilibrium of the population, the diploid genotype frequencies in the current generation are distributed as given in table~\ref{table:fitnesses}.

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|l||c|c|c|}
        \hline
        \textbf{Genotype} & $AA$ & $AB$ & $BB$ \\
        \hline
        \textbf{Wrightian fitness ($\bm{w}$)}  & $1+s$ & $1$ & $1+2s$ \\
        \hline \textbf{Hardy-Weinberg frequency}  & $2p(1-p)$ & $(1-p)^2$ & $p^2$\\
        \hline \textbf{Relative Wrightian fitness}  & $\dfrac{1+s}{1+2ps}$ & $\dfrac{1}{1+2ps}$ & $\dfrac{1+2s}{1+2ps}$\\
        \hline
    \end{tabu}}
    \caption[Fitnesses of the different genotypes]{Fitnesses of the different genotypes}\label{table:fitnesses}
\end{table}

As a result, the mean fitness in the population is a function of the selection coefficient and the frequency of two alleles as:
\begin{align}
    \overline{w} &= (1+2s)p^2 + (1+s)2pq + q^2 \\
    &= 1 + 2ps,
\end{align}
And the relative fitness\footnote{Analogous explanation between absolute and relative fitness can be found in economics, where absolute fitness correspond to wealth generation, while relative fitness is just transfer of wealth from one to another~\citep{Masel2016}.} of the three different genotypes are also shown in table~\ref{table:fitnesses}.

Reproduction proceeds in two steps.
In a first step, a very large pool of gametes is produced, in which adults contribute proportionally to the fitness of their genotype.
Altogether, the frequency $p'$ of gametes bearing the $B$ allele is:
\begin{align}
    p' & = p^2 \dfrac{1+2s}{1+2ps} + p(1-p)\dfrac{1+s}{1+2ps}\\
    & = p\dfrac{1+s(1+p)}{1 + 2ps}
\end{align}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth, page=1] {figures.pdf}
    \caption[Frequency of derived {allele} after a generation]{
    Frequency of derived \gls{allele} after a generation.
    Positive selection coefficient ($s > 0$) result in increased derived \gls{allele} frequency at the next generation, which is intuitively expected.
    The effect is stronger when the derived \gls{allele} frequency is close to $0.5$, intuitively because the poll of both \glspl{allele} must be sufficiently large such that they can be replaced.
    It is worth noting that even for strong selection coefficient ($s=0.2$), completely unrealistic in real population, the difference in frequency from one generation to the next is subtle.}
\end{figure}

Under the assumption that selection is weak $|s| \ll 1$,  $p'$ of gametes bearing the $A$ allele reduces to:
\begin{align}
    p' & \simeq p (1 + s + ps - 2ps) \\
    & = p + sp(1-p)
    & = p + \Delta p
\end{align}
where, $\Delta p = sp(1-p)$

In a second step, the $N$ individuals of the next generation are obtained by randomly sampling from the pool of gametes.
As a result, the probability $\mathcal{P}_{ij}$, that there are $j$ copies of the derived allele $B$ present at the next generation, given that there were $i$ copies in the current generation is given by the binomial distribution, with mean given by $p'$:
\begin{align}
    \mathcal{P}_{ij} & = \binom{2 N}{j} \left( p' \right)^j \left(1 - p' \right)^{2 N -j} \\
    & = \binom{2 N}{j} \left( p\dfrac{1+s(1+p)}{1 + 2ps} \right)^j \left(1 - p\dfrac{1+s(1+p)}{1 + 2ps} \right)^{2 N -j} \label{eq:binomial-WF}
\end{align}

These {transition} probabilities define a discrete-space and discrete-time Markov process.
However, it has been shown to be extremely difficult to explicitly derive formulas for several quantities of evolutionary interest.
However, as the size of the population approaches infinity (i.e. $ N \to \infty$), and assuming that the scaled selection pressure ($N s $) remain constant, the discrete Markov process given above can be closely approximated by continuous-time and continuous-space diffusion process.\\

Of note, the fluctuations induced by the binomial sampling (equation~\ref{eq:binomial-WF}) are the original cause of random drift.
Quantitatively, the expected frequency change from one adult generation to the next adult generation is:
\begin{equation}
    \E[\Delta p] = s p (1-p).
\end{equation}
The variance of this binomial distribution is given by:
\begin{equation}
    \Var[\Delta p] = \frac{p' (1-p')}{2N}
\end{equation}
Since the change in frequency between two generations is small ($p \simeq p'$), the variance is very close to:
\begin{equation}
    \Var[\Delta p] \simeq \frac{p(1-p)}{2N}
\end{equation}
Thus, the variance induced by random drift is inversely proportional to the population size $N$.
Also, if $s \gg 1/2N$, then $E[\Delta p] \gg Var[\Delta p]$, or in other words the systematic trend imprinted by selection dominates over drift, describing the strong selection regime.
In contrast, if $s \ll 1 / 2N$, drift dominates over selection, describing the effectively neutral regime.

\subsection{Effective population size}

The notion of effective population size, called $\Ne$ only appears when we apply a panmictic model to a population that is not, or to a real population.
$\Ne$ was originally defined as \textit{the number of breeding individuals in an idealised population that would show the same amount of dispersion of allele frequencies under random genetic drift or the same amount of inbreeding as the population under consideration}.
For most quantities of interest and most real populations, the census population size $N$ of a real population is usually larger than the effective population size $\Ne$.
The same population may have multiple effective population sizes, for different properties of interest, including for different genetic loci.

\subsection{Probability of fixation}

Starting from an initial frequency, the Wright-Fisher process eventually reaches absorption: the derived \gls{allele} either invade the population or dies out.
Under the continuous-time and continuous-space diffusion approximation, a partial differential equation know as the Kolmogorov backward equation can be used to derive the probability of these two alternative events.
If the selection coefficient is small (or if selection is weak, $|s| \ll 1$), the probability of fixation ($P_{\mathrm{fix}}$) of a derived \gls{allele} with selection coefficient $s$ and initial frequency $p$ was derived by \citet{Kimura1962}:
\begin{align}
    P_{\mathrm{fix}}(s, \Ne, p) & = \dfrac{1 - \e^{-4 \Ne p s }}{1 - \e^{-4 \Ne s}},\\
    & = \dfrac{1 - \e^{- p S }}{1 - \e^{-S}},
\end{align}
where $S = 4 \Ne s$ is the scaled selection coefficient.
\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth, page=3] {figures.pdf}
    \caption[Probability of fixation]{
    Probability of fixation.
    The selection coefficient of the derived \gls{allele} is scaled by population size $S=4 \Ne s$.
    In contrast to changes of frequency during a generation, the probability of fixation is sensitive to very weak selection coefficient ($|s| \ll 1$), as long as the scaled selection coefficient is not negligible ($|S| > 1$).
    Intuitively, selective effects are magnified by population size because the fixation is the inherently the resultant of the trajectory overall, integrating small effects throughout the lifespan of the \gls{allele}. }
\end{figure}

An interesting special case is obtained for a new mutation appearing in the population.
Because it is a single mutant the derived \gls{allele} initial frequency is $p = 1 / 2 \Ne$, and the probability of fixation ($P_{\mathrm{fix}}$) is given by:
\begin{align}
    P_{\mathrm{fix}}(s, \Ne) & = \dfrac{1 - \e^{-2 s}}{1 - \e^{-4 \Ne s}} \\
    & \simeq  \dfrac{2 s }{1 - \e^{-4 \Ne s}}
\end{align}
The special case of a \gls{neutral} \gls{allele} can be obtained by taking the limit when $s$ goes to $0$.
\begin{align}
    P_{\mathrm{fix}}(0, \Ne) & = \dfrac{1}{2 \Ne}
\end{align}
Altogether, the fixation probability of a selected \gls{allele} relative to a \gls{neutral} \gls{allele} is solely dependent on the scaled selection coefficient:
\begin{align}
    \dfrac{P_{\mathrm{fix}}(s, \Ne)}{P_{\mathrm{fix}}(0, \Ne)} & \simeq 2 \Ne \dfrac{2s}{1 - \e^{-S}} \\
    & \simeq  \dfrac{S}{1 - \e^{-S}} \label{eq:p-fix}
\end{align}
\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth, page=2] {figures.pdf}
    \caption[Relative fixation probability]{Fixation probability of a selected {allele} relative to a {neutral} {allele}.
    For a substantive negative scaled selection coefficient ($s \leq -1/\Ne$, red filled area), the probability of fixation is greatly reduced (by an exponential factor), and \gls{allele} can not likely reach fixation. On the other hand for a positive scaled selection coefficient ($s \geq 1 / \Ne$, green filled area), the ratio is approximately linear w.r.t $S$. In between, whenever the absolute value of is close to $1 / \Ne$ (yellow filled area), the \gls{allele} behave approximately neutrally.}
\end{figure}
This essential result has important consequences, random \gls{drift} and selection are intrinsically confounded factors.
As a an example, increasing population size by a factor $2$ while reducing the selection coefficient by the same amount lead to the exact same equation, such that they are indistinguishable.

\subsection{Site frequency spectrum}
The probability of fixation of an \gls{allele} can be empirically observable, and in the context of Wright-Fisher processes is related to selection and drift.
However, this absorbing fate is not the sole characteristic of the process that relates empirical observable quantities to parameters of the process.
Along the whole trajectory of an \gls{allele}, before fixation or extinction, the probability of this \gls{allele} to be at a certain frequency can be related to its selection coefficient and to the \gls{effective-population-size}.
More precisely, $g(x) \der x $ is the expected time for which the population frequency of derived \gls{allele} is in the range $(x, x+\der x)$ before eventual absorption, and can be derived using Kolmogorov forward equation:
\begin{align}
    g(x, S) & = \dfrac{\left( 1 - \e^{- 2 s }\right) \left( 1 - \e^{-4 \Ne s(1-x)}\right)}{ s (1 - \e^{-4 \Ne s})x(1-x)} \\
    & \approx \dfrac{2 \left[ 1 - \e^{-S(1-x)}\right]}{(1 - \e^{-S})x(1-x)} \label{eq:expected_time}
\end{align}

\begin{figure}[H]
    \centering
    \includegraphics[width=0.8\textwidth, page=4] {figures.pdf}
    \caption[Expected time at a derived frequency]{
    Expected time at a derived frequency.
    Allele with positive selection coefficient can be observed at high frequency, while \glspl{allele} with negative selection coefficient are unlikely to be observed at high frequency.}
\end{figure}

This equation is solely valid for a gene with two \glspl{allele}, a configuration which is rarely observed in empirical data since more than two variants of a gene are usually present in the population.
However, it is frequent to observe sites inside a gene sequence for which only two \glspl{allele} are segregating.
This observation led to the development of a sitewise Wright-Fisher process, assuming that each site follows an independent process~\citep{Sawyer1992}.
Strictly speaking, this model considers a collection of independently evolving loci, meaning without linkage.
It provides a good approximation if there is free-recombination between sites.
Moreover, the collection is considered infinite whereas the total mutation rate across this infinite collection is considered finite.
Assumption of an infinite number of sites is necessary to ensure that each mutation arise at a new site, with a Poisson distribution of total rate $u$ per generation for the whole sequence.

From an empirical perceptive, for a sample of $n$ sequences taken in the population, the expected number of sites with $i$ copies of the derived allele (with $i$ ranging from $1$ to $n - 1$) denoted $G(i, n)$.
The collection of all $G(i, n)$ generate what is called a site frequency spectrum (\acrshort{SFS}), which can intuitively be interpreted as the discrete version of the expected time at a derived frequency (equation~\ref{eq:expected_time}), readily available from a sample of sequences from a population.
Given the scaled selection pressure ($S=4 \Ne s$), and the scaled mutation rate per generation for the whole sequence ($\theta = 4 \Ne u $), each entry of the \acrshort{SFS} is:
\begin{align}
    G(i, n) & = \int_{0}^{1}  2 \Ne u g(x) \binom{n}{i} x^{i} (1-x)^{n-i} \der x \\
    & = \theta \int_{0}^{1} \dfrac{1 - \e^{-S(1-x)}}{(1 - \e^{-S})x(1-x)} \binom{n}{i} x^{i} (1-x)^{n-i} \der x \\
    & =  \dfrac{\theta }{1 - \e^{-S}} \binom{n}{i} \int_{0}^{1} \left( 1 - \e^{-S(1-x)} \right) x^{i-1} (1-x)^{n-i-1} \der x
\end{align}

This site frequency spectrum can be confronted to empirical \gls{polymorphic} data in order to estimate the scaled selection coefficient of new mutations.
However, a single selection coefficient for all sites and all mutations is biologically not realistic.
Accordingly, a distribution of selection coefficients across sites is assumed, which is usually modeled as a continuous distribution, known as the distribution of fitness effects of mutations (DFE).
Mixing over this distribution, the SFS can then be computed as a function of the underlying DFE, and can thus be estimated based on empirical data~\citep{eyre-walker_distribution_2006, eyre-walker_estimating_2009}.

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|l|c|c|c|}
        \hline
        \textbf{Parameter} & \textbf{Symbol} & \textbf{Range} \\
        \hline \hline
        Population size & $N$ & $ [10^2, 10^6]$ \\
        Effective population size & $\Ne$ & $ [10^2, 10^6]$ \\
        \hline Absolute Wrightian fitness & $w$ & $ \simeq 1 $ \\
        \hline Relative fitness & $f=\ln(w)$ & $ \ll 1 $ \\
        \hline Selection coefficient & $s$ & $ |s| \ll 1 $ \\
        \hline Scaled selection coefficient & $S=4 \Ne s$ & Finite (negative or positive) \\
        \hline Mutation rate per generation & $u$ & $[10^{-10}, 10^{-7}]$ per site \\
        \hline Scaled mutation rate & $\theta = 4 \Ne u$ & $[10^{-8}, 10^{-1}]$ per site \\
        \hline
    \end{tabu}}
    \caption[Parameter of population-genetics]{Parameter of population-genetics}\label{table:params-popgen}
\end{table}


\section{Mutation-selection process}
The previous section recalled the Wright-Fisher process of evolution inside a population, relating selection and drift to the diversity of sequences, which empirically requires gene sequences for at least several individuals.
However, modeling sequence evolution between different species along lineages is a different endeavor, in which species are often simplified with a single representative sequence, collapsing the intra-specific diversity.
Under this simplification, the inter-specific variability and the evolutionary trajectory of sequences is described by the past history of point \glspl{substitution} along lineages.
The rate at which such \glspl{substitution} occur can nonetheless be decomposed into two mechanisms, their origination through mutation and their final fate of fixation or loss, a modeling approach broadly known as the origin-fixation approximation~\citep{McCandlish2014}, illustrated in figure~\ref{fig:point-process}.
Most importantly this decomposition of \gls{substitution} events into mutation and fixation events is able to conciliate population genetics and inter-specific molecular evolution, where the \gls{substitution} history is parameterized by mutation, selection and drift.
In the field of phylogenetic, the origin-fixation framework is more commonly known as the mutation-selection paradigm, where fixation of an \gls{allele} encompass the confounded effect of natural selection and drift (which are confounded factors, see equation~\ref{eq:p-fix}), and origination corresponds to mutation.
Since the scope of this manuscript emanates from phylogenetic, I will use the convention mutation-selection terminology hereafter.
Of note, a more general mathematical description of the mutation-selection framework recruiting tools from statistical physics can be found in \citet{Sella2005} and \citet{Mustonen2009}.

\begin{figure}[H]
    \centering
    \includegraphics[width=0.75\textwidth]{figures/point-process.pdf}
    \caption[Mutation-selection point substitutions]{
    Mutation-selection point \glspl{substitution} models.
    The trajectory of \glspl{allele} inside a population is collapsed into a single point \gls{substitution} process.
    This approximation is valid under low mutation rate such that a mutation originate uniquely whenever the gene is monomorphic (with a single allele).}
    \label{fig:point-process}
\end{figure}

\subsection{Mutation-limited process}
\label{subsec:mutation-limited-assumption}

Mutation-selection probabilistic models are usually Markovian with respect to time, such that the next \gls{substitution} event depends on the current representative sequence but not on earlier sequences visited in the history of a lineage.
This continuous-time Markovian process is valid if the mutation rate is sufficiently low, such that the event of a new mutation reaching fixation is completed before the next one occurs.
Since the rate of \gls{substitution} is equal to $u$ (per generation) and that each \gls{allele} ultimately reaching fixation is segregating for an average of $4 \Ne$ generations~\citep{Kimura1969}, this assumption is broadly applicable whenever the product of population size and mutation rate per generation for the sequence is lower than $1$ ($4 \Ne u \ll 1$).
More strictly, the model would require not only that new mutations reaching fixation do so before the next \gls{substitution} occurs, but before any mutation occurs, even the one that ultimately become extinct.
Since at each generation during the process an average of $2\Ne$ of mutations are produced, the point \gls{substitution} is valid under the condition that $8\Ne^2 u \ll 1$.
In practice, the assumptions that $4 \Ne u \ll 1$ is a sufficient condition for the process to be well approximated.
Throughout this development, it is important to note that $u$ is the mutation rate for the whole sequence under consideration.

For large sequences this approximation is usually not valid, and the sequence is then decomposed into each individual sites, forming a collection of independently evolving continuous-time \glspl{mc}.
For such a decomposition to be valid, these models have to assume free \gls{recombination} between sites.
The mutation rate $u$ in the condition then refers to the mutation rate for each independent site, rather than the total mutation rate over the collection as a whole.
For example, \citet{Halpern1998} construct a model for the evolution of coding sequences where each \gls{codon} site is modeled as an independent \gls{mc}.

\subsection{Substitution rate}
The continuous-time \gls{mc} is defined by the instantaneous rate at which transitions occur between pairs of states.
Given the current state of the process is \gls{allele} $A$, the rate of {transition} to other states can be derived using the population-genetic equations introduced above.
At each generation, the expectation for the number of possible mutants is $2\Ne u$, and each of these mutant have a probability $P_{\mathrm{fix}}$ to result in a \gls{substitution}.
Altogether, the instantaneous rate of \gls{substitution} from \gls{allele} $A$ to $B$, denoted $Q_{A \to B}$, is equal to the rate of mutation ($\mu_{A \to B}$) multiplied by the probability of fixation of the mutation $P_{\mathrm{fix}}(s_{A \to B}, \Ne)$ and scaled by the number of possible mutants at each generation ($2\Ne$):
\begin{align}
    Q_{A \to B} & = 2 \Ne \mu_{A \to B}  P_{\mathrm{fix}}(s_{A \to B}, \Ne)
\end{align}
It is important to note that the \gls{substitution} rate and the mutation rate are in the same unit, such that this equation is valid whether the rate is measured either in unit of chronological time or per generation (or in branch length, which will matter later on).
As a convention, in what follows, mutation rate is denoted $u$ when measured in unit of generation, and denoted $\mu$ when measured in unit of time.
As a consequence, $Q$ is measured in unit of time in this section.

In the case of \gls{neutral} mutations, the probability of fixation is independent of the original and target sequence, and equal $1/2 \Ne$.
As a consequence, the \gls{substitution} rate $Q_{A \to B}$ simplifies to:
\begin{align}
    Q_{A \to B} & = 2 \Ne \mu_{A \to B}  P_{\mathrm{fix}}(0, \Ne) \\
    Q_{A \to B} & = 2 \Ne \mu_{A \to B} \dfrac{1}{2\Ne} \\
    Q_{A \to B} & =  \mu_{A \to B} \label{eq:sub-equal-mut}
\end{align}
In the case of selected mutations, the probability of fixation depends on the difference in log-fitness ($f_A$ and $f_B$) between the two \glspl{allele}:
\begin{align}
    Q_{A \to B} & = 2 \Ne \mu_{A \to B} P_{\mathrm{fix}}(s_{A \to B}, \Ne) \\
    & = 2 \Ne \mu_{A \to B}  \dfrac{2(f_{B} - f_{A})}{1 - \e^{4\Ne(f_{A} - f_{B})} } \\
    & = \mu_{A \to B} \dfrac{F_{B} - F_{A}}{1 - \e^{F_{A} - F_{B}} }\text{, where } F = 4\Ne f \label{eq:sub-transion-rates}
\end{align}

If the difference of log-fitness tends to $0$, the \gls{substitution} rate is equal to the mutation rate, retrieving equation~\ref{eq:sub-equal-mut}:
\begin{align}
    \lim_{|F_{B} - F_{A}| \to 0} Q_{A \to B} & = \mu_{A \to B}   \dfrac{F_{B} - F_{A}}{{1 - (1 + {F_{A} - F_{B}}) }} \nonumber \\
    & =  \mu_{A \to B}
\end{align}

Taken together, the {transition} rates which generates the \gls{substitution} history and ultimately the inter-specific diversity is parameterized solely by mutation, selection and drift.
Consequently, from a particular history of \glspl{substitution}, one can theoretically estimate the parameter of selection, mutation and drift, although it is important to keep in mind that selection and drift are confounded.

\subsection{Reversibility of the process}
The continuous-time \gls{mc} as so far been defined for $2$ \glspl{allele} but can be generalized to any number of \glspl{allele}, when the number of \glspl{allele} must be discrete and is must be possible to undergo {transition} from any \gls{allele} to any other possible \gls{allele}, possibly in several \glspl{substitution}.
In this configuration, the {transition} rates between all possible pairs of \glspl{allele} is defined by equation~\ref{eq:sub-transion-rates}, and equal to $0$ whenever single step {transitions} are not possible.
Because any state is ultimately connected to any other state, the continuous-time \gls{mc} is irreducible.
Moreover, this \gls{substitution} process is positive recurrent and aperiodic since any strictly positive {transition} rate is matched by a stricly positive {transition} for the reverse \gls{substitution}.
More precisely, the \gls{substitution} rate between two \glspl{allele} is null only if the underlying mutation rate is null, in which case the {transition} rate for the reverse mutation is also null, hence the {transition} rate for the reverse \gls{substitution} is also null.

Theoretically, for an irreducible, positive recurrent and aperiodic continuous-time \gls{mc}, a necessary and sufficient condition to be reversible is given by Kolmogorov's criterion.
Kolmogorov's criterion implies that the product of {transition} rates through any closed loop is the same whenever the traversing is done forward or in reverse.
As an example for a \gls{mc} composed of $3$ \glspl{allele} ($A$, $B$ and $C$) the {transitions} rate must satisfies the equality:
\begin{equation}
    Q_{A \to B}Q_{B \to C}Q_{C \to A} = Q_{A \to C}Q_{C \to B}Q_{B \to A}
\end{equation}

\begin{figure}[htb!]
    \centering
    \begin{tikzpicture}[->,>=stealth',auto,node distance=1.2cm and 1.6cm,semithick]
        \tikzstyle{every state}=[]

        \node[] (0) {};
        \node[state] (A) [above=of 0] {$A$};
        \node[state] (B) [below left=of 0] {$B$};
        \node[state] (C) [below right=of 0] {$C$};

        \path[->]
        (A) edge [BLUE,bend right=45] node [left] {$Q_{A \to B}$} (B)
        (B) edge [BLUE,bend right=45] node [below] {$Q_{B \to C}$} (C)
        (C) edge [BLUE,bend right=45] node [right] {$Q_{C \to A}$} (A)
        (B) edge [RED,bend right=15] node [left] {$Q_{B \to A}$} (A)
        (C) edge [RED,bend right=15] node [below] {$Q_{C \to B}$} (B)
        (A) edge [RED,bend right=15] node [right] {$Q_{A \to C}$} (C);
    \end{tikzpicture}
    \caption[Kolmogorov's criterion]{
    The continuous-time \gls{mc} is reversible if the process fulfills Kolmogorov's criterion, such that the product of {transition} rate for a closed loop is equal whether traversed in one sens (blue arrows) or the other (red arrows).}
    \label{fig:reversible-circuit}%
\end{figure}

From {transition} rates of the \gls{substitution} process (\ref{eq:sub-transion-rates}), Kolmogorov's criterion can be derived as:
\begin{align}
    \dfrac{Q_{A \to B}Q_{B \to C}Q_{C \to A}}{Q_{A \to C}Q_{C \to B}Q_{B \to A}}
    ={}& \dfrac{\mu_{A \to B}\mu_{B \to C}\mu_{C \to A}}{\mu_{A \to C}\mu_{C \to B}\mu_{B \to A}}
    \times \dfrac{(F_{B} - F_{A})(F_{C} - F_{B})(F_{A} - F_{C})}{(F_{C} - F_{A})(F_{B} - F_{C})(F_{A} - F_{B})} \notag \\
    &\ \times \dfrac{( 1 - \e^{F_{A} - F_{C}} )( 1 - \e^{F_{C} - F_{B}} )( 1 - \e^{F_{B} - F_{A}} )}{( 1 - \e^{F_{A} - F_{B}} )( 1 - \e^{F_{B} - F_{C}} )( 1 - \e^{F_{C} - F_{A}} )}, \\
    ={}& - \dfrac{\mu_{A \to B}\mu_{B \to C}\mu_{C \to A}}{\mu_{A \to C}\mu_{C \to B}\mu_{B \to A}} \notag \\
    &\ \times \dfrac{\e^{F_{A}}( \e^{-F_{A}} - \e^{ - F_{C}} )\e^{F_{C}}( \e^{-F_{C}} - \e^{ - F_{B}} )\e^{F_{B}}( \e^{-F_{B}} - \e^{ - F_{A}} )}{\e^{F_{A}}( \e^{-F_{A}} - \e^{ - F_{B}} )\e^{F_{B}}( \e^{-F_{B}} - \e^{ - F_{C}} )\e^{F_{C}}( \e^{-F_{C}} - \e^{ - F_{A}} )}, \\
    ={}& \dfrac{\mu_{A \to B}\mu_{B \to C}\mu_{C \to A}}{\mu_{A \to C}\mu_{C \to B}\mu_{B \to A}}.
\end{align}
As a result, Kolmogorov's criterion for the \gls{substitution} process is satisfied only if the mutation process is also reversible, in which case Kolmogorov's criterion is also fulfilled:
\begin{equation}
    \mu_{A \to B}\mu_{B \to C}\mu_{C \to A}=\mu_{A \to C}\mu_{C \to B}\mu_{B \to A}.
\end{equation}
This example can be generalized for any close loop, such that the reversibility of the \gls{substitution} process is conditioned on the reversibility of the underlying mutation process, which is often assumed.

\subsection{Stationary distribution}

A realization of the \gls{mc} for a long period of time results in a given proportion of the time for which the process is fixed for a specific \gls{allele}, where this proportion depends of the \gls{allele} fitnesses, the mutational process and $\Ne$.
Because the continuous-time \gls{mc} is irreducible, positive recurrent and aperiodic, it has a unique stationary distribution $\bm{\pi}$, where $\pi_{A}$ corresponds to the proportion of time spent in \gls{allele} $A$ after the \gls{mc} has run for an infinite amount of time.

Moreover, under the condition that the \gls{mc} is time-reversible, the detailed balanced for the stationary distribution is satisfied:
\begin{align}
    \dfrac{\pi_{A}}{\pi_{B}} & = \dfrac{Q_{B \to A}}{Q_{A \to B}} \\
    & = \dfrac{\mu_{B \to A}}{\mu_{A \to B}}  \dfrac{F_{A}-F_{B}}{ 1 - \e^{F_{B}-F_{B}}}  \dfrac{1 - \e^{F_{A} - F_{B}} }{F_{B} - F_{A}}\\
    & = - \dfrac{\mu_{B \to A}}{\mu_{A \to B}}  \dfrac{ \e^{F_{A}}(\e^{-F_{A}} - \e^{- F_{B}}) }{ \e^{F_{B}}(\e^{-F_{B}} - \e^{- F_{A}})}  \\
    & = \dfrac{\mu_{B \to A}}{\mu_{A \to B}} \dfrac{\e^{F_{A}}}{\e^{F_{B}}} \\
\end{align}
Under the assumption that the mutational process is also reversible, the detailed balanced for the stationary distribution of the mutation process ($\bm{\sigma}$) is satisfied:
\begin{align}
    \dfrac{\mu_{B \to A}}{\mu_{A \to B}} & = \dfrac{\sigma_{A}}{\sigma_{B}}
\end{align}
Altogether, the probability $\pi_{A}$ to find the population in \gls{allele} $A$ is proportional to a function (also called a Boltzmann factor) that depends only on the fitness of \gls{allele} $A$, the population size, and details of the mutation process~\citep{Sella2005,Mustonen2005}:
\begin{align}
    \dfrac{\pi_{A}}{\pi_{B}} & = \dfrac{\sigma_{A}\e^{F_{A}}}{\sigma_{B}\e^{F_{B}}} \text{ and } \sum\limits_{A}\pi_{A} = 1, \\
    \iff \pi_{A} & = \dfrac{\sigma_{A}\e^{F_{A}}}{\sum\limits_{B} \sigma_{B}\e^{F_{B}} }, \label{eq:equilibrium-mut-sel}
\end{align}
where the denominator is a normalizing constant such that the sum of probabilities is equal to $1$.
By analogy with thermodynamic systems, the evolutionary system thus reaches a Boltzmann-like distribution with $\Ne^{-1}$ playing the role of evolutionary temperature, and the log-fitness $f$ the role of energy\footnote{At high mutation rates, the quasi-species theory provides another analogy with statistical mechanics, in which the mutation rate plays the role of temperature instead of \gls{drift}.}.

\subsection{Mean scaled fixation probability}
\label{subsec:mean-scaled-fixation-probability}

Occurrence probabilities given by the stationary distribution allows one to calculate all observable quantities of interest, such as the mean fitness, or the mean mutation and \gls{substitution} rates, using standard probability theory.
One quantity of interest is the ratio of the mean \gls{substitution} rate over the mean mutation rate, called $ \nu $:
\begin{align}
    \nu & = \dfrac{ \langle Q \rangle }{\langle \mu \rangle},
    \label{eq:relative-sub-rate} \\
    & = \dfrac{ \sum\limits_{A} \pi_{A} \sum\limits_{B} Q_{A \to B}}{\sum\limits_{A} \pi_{A}  \sum\limits_{B} \mu_{A \to B}},
\end{align}
where the notation $\langle \cdot \rangle$ denotes the statistical average.
In other words, $ \nu $ represents the flow of substitutions at equilibrium, normalized by the mutational flow (or mutational opportunities).

Moreover, the mean probability fixation, scaled by $2 \Ne$ can be approximated as:
\begin{align}
    \langle  2\Ne P_{\mathrm{fix}} \rangle & = \left\langle 2\Ne \dfrac{ 2 s }{1 - \e^{- S}} \right\rangle, \\
    & = \left\langle  \dfrac{ S }{1 - \e^{- S}} \right\rangle, \\
    & = \left\langle \dfrac{ Q }{\mu} \right\rangle, \\
    & \simeq \dfrac{ \langle Q \rangle }{\langle \mu \rangle}.
\end{align}
Because of this relation, $\nu$ can thus be seen as an approximation of the mean scaled probability of fixation of mutations.

As a result of this definition, $\nu=1$ for genes or sites under \gls{neutral} evolution.
Most importantly, departure from $1$ would be interpreted as a signature of selection on sequences.
First, $\nu>1$ is interpreted as a signal of adaptive recurrent evolution, where $P_{\mathrm{fix}} > 1/2 \Ne$ on average.
On the other hand, $\nu<1$ is a signal of underlying purifying selection, such that the sequence is constrained.
Of note, $ \nu > 1$ (or $ < 1$) does not necessarily mean that the selection coefficients are positive (or negative) on average.
Moreover, under the assumption of a static fitness landscape, a mutation-selection point \gls{substitution} process at equilibrium results in $\nu \leq 1$ \footnote{However, I haven't found a proof for this claim in the bibliography.}.

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|l|c|c|c|}
        \hline
        \textbf{Parameter} & \textbf{Symbol} & \textbf{Range} \\
        \hline\hline
        Scaled fitness & $F=4 \Ne f$ & finite, positive or negative \\
        \hline Mutation rate per time & $\mu$ & $[10^{-11}, 10^{-8}]$ per site per year \\
        \hline \Gls{substitution} rate per time & $Q$ & $[10^{-11}, 10^{-8}]$ per site per year \\
        \hline Equilibrium frequency & $\pi$ & $[0, 1]$ \\
        \hline Equilibrium frequency under mutation & $\sigma$ & $[0, 1]$ \\
        \hline Mean scaled fixation probability & $\nu$ & $[0, 1]$ for purifying selection \\
        \hline
    \end{tabu}}
    \caption[Parameter of mutation-selection processes]{Parameter of mutation-selection processes}
    \label{table:params-mutsel}
\end{table}


\section{Mutation-selection analogy}
Presented in the context of phylogenetic evolution of genetic sequences, the mutation-selection process bears many similarities and analogies between other processes present in a variety of scientific fields outside of evolutionary biology, displaying the same underlying mechanism and emerging properties, though with different name and aspiration.
This section is an attempt to describe analogous processes and their emerging properties.
This effort is made in the aim of giving another view of the mutation-selection process, such as to better appreciate and conceptualize its assumptions, its limits, and the respective role of the different components.
Such attempts requires to boil down the mutation-selection mechanism into its core components, while at the same time rephrasing the description using lexicography outside of population-genetics such as to open new perceiving angles.

At the bottom, mutation is a process creating diversity, changing and moving the current viable state to a novel and unknown position, fundamentally allowing exploration of the state space.
On the other hand, selection is the criteria on which a new state is deemed a disrupting innovation or a nonviable alteration, and allow to determine which changes to exploit and which to filter out and discard based on its fitness.
Fundamentally, mutation creates diversity and selection reduce this diversity by selecting the fittest ones.
Finally, drift arbitrate between the creation and reduction of the two processes, it dictates how much exploration of novelty is permitted, and conversely how much exploitation of only the fittest states is granted.

I argue that this creation and reduction process is found at the core of several research disciplines, while the link between them is scarcely made~\citep{Baeck1994, Eiben1998}.

\subsection{Metropolis-Hasting sampling}
Obtaining a sequence of random samples from a probability distribution can be difficult, especially when the number of dimensions is high.
However, the Metropolis-Hasting procedure based on a Monte Carlo \gls{mc} can sample from any probability distribution, provided that we know how to compute the probability density, or even less restrictively any function proportional to the density~\citep{Hastings1970}.
This stochastic procedure which is based on three steps bears many similarities with the mutation-selection process:
\begin{itemize}
    \item Generate a stochastic candidate from the current state, analogous to the mutation.
    \item Calculate the acceptance ratio as the ratio of the two densities, analogous to the selection coefficient of the mutated state.
    \item Stochastic acceptance or rejection based on the acceptance ratio, a process analogous to drift.
\end{itemize}
Inherently, Metropolis-Hasting procedure is based on creating and subsequently reducing diversity, which allows to obtain a random sequence of samples from any distribution with a straightforward recipe, and is a critical tools in statistic and statistical physics.

\subsection{The exploration-exploitation dilemma}
Many mathematical, engineering and day-life problems are not about sampling a state space, but rather finding the optimal and best state given a criteria or a function to maximize.
Naturally, we would prefer deterministic (strictly reproducible) rather than stochastic optimizing strategies to search for an optimal state.
Unfortunately, whenever the state space is too large, often due to the curse of dimensionality, a greedy or heuristic search of an optimal state can performs atrociously~\citep{Bellman1966}.
In high dimensional space, stochastic optimization tools have been deemed very valuable, such as stochastic gradient decent or so called evolutionary algorithm~\citep{Russell2010,Vikhar2017}.
Inherently, they are based on two processes, one is stochastically creating diversity and exploring the state space, while the other is filtering the explored states and thus reducing the diversity.

In the constrained case of a finite number of time or attempts to find the best outcome overall, the problem is best described by the multi-armed bandit problem.
The name comes from imagining a gambler at a row of slot machines (sometimes known as one-armed bandits), where each slot machine provides a random reward from a probability distribution specific to that machine. The player has to decide which machines to play, how many times to play each machine and in which order to play them, and whether to continue with the current machine or try a different machine, such as to maximize the sum of rewards earned through a sequence of trials.
The gambler faces a dilemma at each trial, either reducing his regret by exploiting the best arm, or gaining information through exploration of other arms.
The best strategy to solved this dilemma can be mathematically derived in numerous cases, and encompass a mixing strategies with a defined ratio of exploration and exploitation~\citep{Auer2002,Kocsis2006,Furnkranz2006}.
This problem is far from be only theoretical, and has be used to explain a multitude of phenomena, such as the movement of animals in novel landscapes, the most efficient resource allocation for a start-up company, the effects of age on knowledge acquisition in humans, and in search of the most efficient treatment in clinical trials~\citep{Berger-Tal2014, March}.
An other application of the exploration-exploitation dilemma is AlphaGo, the first computational program mastering the board game go at the professional 9-dan level in 2017, and out-compete Ke Jie, the world nÂ°1 ranked player at the time~\citep{Silver2017, Silver2018}.
AlphaGo has often been publicized and hyped in various media outlets that this feat was possible due to machine learning, more specifically due to convolutionnal neural networks.
However, it is more scarcely mentioned that AlphaGo neural network is combined with an exploration-exploitation algorithm, or more specifically Monte Carlo tree search.
In practice, the convolutionnal neural network is used as a criteria to measure the advantage of a board configuration\footnote{Convolutionnal neural networks also use a stochastic gradient descend to reach convergence, inherently leveraging stochastic exploration and exploitation procedure to optimize the parameters of the neural network.}, but the different moves and path probed and trimmed is done via an exploration-exploitation procedure.


Altogether, exploration and exploitation, creation and reduction, mutation and selection, are different names that encompass the inherently same process efficiently sampling and optimizing whenever the state is too large to be traversed in a finite amount of time.
I argue that scientific research endeavor is also a exploration-exploitation dilemma, which is arguably externally pressured to pursue exploitation, through funding of impactful research and a \textit{publish-or-perish} systemic culture in early career stage.

\begin{table}[H]
    \centering
    \noindent\adjustbox{max width=\textwidth}{%
    \begin{tabu}{|c|c|c|}
        \hline
        \textbf{Mutation} & \textbf{Selection} & \textbf{Drift} \\ \hline
        \hline Exploration & Exploitation & Trade-off \\
        \hline Creation & Reduction & Arbitration \\
        \hline Candidate generation & Acceptance & Hasting ratio \\
        \hline
    \end{tabu}}
    \caption[Mutation, selection and drift analogy]{
    Mutation, selection and drift lexicographic rephrasing in different fields.
    }
    \label{table:intro-mutsel-analogy}
\end{table}

% As a side note, it appears that drift and selection are actually confounded, they are both on the side of exploitation, not on exploration.
As a side note, mutation is a necessary process of evolution, while sexe is not mandatory but increase the diversity of genomes between generations.
Arguably, studying evolution while disregarding mutation is a sufficient approximation whenever mutation rate and number of generations is low, which is developed thoroughly through the whole field of quantitative genetics.
Sex and mutation are both generating new states and are part of the more general exploration facet.
This explains why sex is favored in fluctuating environments.

I argue that evolutionary biologists, studying and leveraging the pervasive process of mutation and selection, can gain knowledge by recruiting insight and developments from other fields, much like there as been many crossover between economics and evolution in the context of game theory.\footnote{Game theory had originally been developed to model economic actors behavior and strategies~\citep{VonNeumann1947}, while latter being emerging as the framework of evolutionary dynamics, which explains the emergence of altruistic behaviours in Darwinian evolution~\citep{Smith1973, Smith1982, Nowak2006}.}.

As an example of such insight from exploration-exploitation dilemma is on the understanding of relationship between mutation rate and chromosome size.
It has been observed that organism with a low mutation rate (per site per generation) tend also to have a long chromosome size, such that the product of genome size and mutation rate is constant~\citep{Drake1991}.
An explanation for this pervasive negative correlation between mutation rate and genome size invoke the drift barrier hypothesis, where the correlation is inherently due to an underlying changes of \gls{drift}~\citep{Lynch2016a}.
The drift barrier hypothesis posits the selection operates to minimize the mutation rate, with the efficiency of such improvement eventually being overcome by random \gls{drift}, such that population with higher population size have smaller mutation rate.
On the other hand, higher population size induces stronger purifying selection which reduce genome size by expelling transposable elements.
Altogether, both high mutation rate and smaller genome are consequences of higher population size.
An alternative explanation for the negative correlation between mutation rate and genome size is that for each new generation, the genetic diversity of offsprings must balance the trade-off between exploration of new genotype and the reliable transmission of the current one.
The genetic difference between parents and offsprings is the product of mutation rate and genome size, such that individuals whom are not robust enough undergo deleterious mutations and disappear, whereas individuals whose genotypes are not variable enough are outcompeted by those that were able to discover innovations~\citep{Knibbe2007, Beslon2010, Hindre2012, Batut2014, Biller2016}.
